// thirdweb/dist/esm/extensions/erc721/__generated__/ISignatureDropERC721/read/totalMinted.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xa2309ff8";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the totalMinted function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeTotalMintedResult } from "thirdweb/extensions/erc721";
 * const result = decodeTotalMintedResult("...");
 * ```
 */
export function decodeTotalMintedResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "totalMinted" function on the contract.
 * @param options - The options for the totalMinted function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { totalMinted } from "thirdweb/extensions/erc721";
 *
 * const result = await totalMinted();
 *
 * ```
 */
export async function totalMinted(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
