// thirdweb/dist/cjs/extensions/uniswap/__generated__/ISwapRouter/write/exactOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.exactOutput = exports.encodeExactOutputParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xf28c0498";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "params",
        components: [
            {
                type: "bytes",
                name: "path",
            },
            {
                type: "address",
                name: "recipient",
            },
            {
                type: "uint256",
                name: "deadline",
            },
            {
                type: "uint256",
                name: "amountOut",
            },
            {
                type: "uint256",
                name: "amountInMaximum",
            },
        ],
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "amountIn",
    },
];
/**
 * Encodes the parameters for the "exactOutput" function.
 * @param options - The options for the exactOutput function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeExactOutputParams } "thirdweb/extensions/uniswap";
 * const result = encodeExactOutputParams({
 *  params: ...,
 * });
 * ```
 */
function encodeExactOutputParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.params]);
}
exports.encodeExactOutputParams = encodeExactOutputParams;
/**
 * Calls the "exactOutput" function on the contract.
 * @param options - The options for the "exactOutput" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```
 * import { exactOutput } from "thirdweb/extensions/uniswap";
 *
 * const transaction = exactOutput({
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function exactOutput(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
exports.exactOutput = exactOutput;
