// thirdweb/dist/cjs/extensions/uniswap/__generated__/ISwapRouter/write/exactOutputSingle.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.exactOutputSingle = exports.encodeExactOutputSingleParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xdb3e2198";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "params",
        components: [
            {
                type: "address",
                name: "tokenIn",
            },
            {
                type: "address",
                name: "tokenOut",
            },
            {
                type: "uint24",
                name: "fee",
            },
            {
                type: "address",
                name: "recipient",
            },
            {
                type: "uint256",
                name: "deadline",
            },
            {
                type: "uint256",
                name: "amountOut",
            },
            {
                type: "uint256",
                name: "amountInMaximum",
            },
            {
                type: "uint160",
                name: "sqrtPriceLimitX96",
            },
        ],
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "amountIn",
    },
];
/**
 * Encodes the parameters for the "exactOutputSingle" function.
 * @param options - The options for the exactOutputSingle function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeExactOutputSingleParams } "thirdweb/extensions/uniswap";
 * const result = encodeExactOutputSingleParams({
 *  params: ...,
 * });
 * ```
 */
function encodeExactOutputSingleParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.params]);
}
exports.encodeExactOutputSingleParams = encodeExactOutputSingleParams;
/**
 * Calls the "exactOutputSingle" function on the contract.
 * @param options - The options for the "exactOutputSingle" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```
 * import { exactOutputSingle } from "thirdweb/extensions/uniswap";
 *
 * const transaction = exactOutputSingle({
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function exactOutputSingle(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
exports.exactOutputSingle = exactOutputSingle;
