// thirdweb/dist/cjs/extensions/farcaster/__generated__/IStorageRegistry/read/maxUnits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.maxUnits = exports.decodeMaxUnitsResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x06517a29";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the maxUnits function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeMaxUnitsResult } from "thirdweb/extensions/farcaster";
 * const result = decodeMaxUnitsResult("...");
 * ```
 */
function decodeMaxUnitsResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeMaxUnitsResult = decodeMaxUnitsResult;
/**
 * Calls the "maxUnits" function on the contract.
 * @param options - The options for the maxUnits function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { maxUnits } from "thirdweb/extensions/farcaster";
 *
 * const result = await maxUnits();
 *
 * ```
 */
async function maxUnits(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.maxUnits = maxUnits;
