// thirdweb/dist/cjs/extensions/farcaster/__generated__/IStorageRegistry/read/rentedUnits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.rentedUnits = exports.decodeRentedUnitsResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x2751c4fd";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the rentedUnits function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeRentedUnitsResult } from "thirdweb/extensions/farcaster";
 * const result = decodeRentedUnitsResult("...");
 * ```
 */
function decodeRentedUnitsResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeRentedUnitsResult = decodeRentedUnitsResult;
/**
 * Calls the "rentedUnits" function on the contract.
 * @param options - The options for the rentedUnits function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { rentedUnits } from "thirdweb/extensions/farcaster";
 *
 * const result = await rentedUnits();
 *
 * ```
 */
async function rentedUnits(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.rentedUnits = rentedUnits;
