// thirdweb/dist/cjs/extensions/farcaster/__generated__/IStorageRegistry/read/unitPrice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unitPrice = exports.decodeUnitPriceResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xe73faa2d";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the unitPrice function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeUnitPriceResult } from "thirdweb/extensions/farcaster";
 * const result = decodeUnitPriceResult("...");
 * ```
 */
function decodeUnitPriceResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeUnitPriceResult = decodeUnitPriceResult;
/**
 * Calls the "unitPrice" function on the contract.
 * @param options - The options for the unitPrice function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { unitPrice } from "thirdweb/extensions/farcaster";
 *
 * const result = await unitPrice();
 *
 * ```
 */
async function unitPrice(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.unitPrice = unitPrice;
