// thirdweb/dist/cjs/extensions/erc721/__generated__/IMintableERC721/write/mintTo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mintTo = exports.encodeMintToParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x0075a317";
const FN_INPUTS = [
    {
        type: "address",
        name: "to",
    },
    {
        type: "string",
        name: "uri",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Encodes the parameters for the "mintTo" function.
 * @param options - The options for the mintTo function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeMintToParams } "thirdweb/extensions/erc721";
 * const result = encodeMintToParams({
 *  to: ...,
 *  uri: ...,
 * });
 * ```
 */
function encodeMintToParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.to, options.uri]);
}
exports.encodeMintToParams = encodeMintToParams;
/**
 * Calls the "mintTo" function on the contract.
 * @param options - The options for the "mintTo" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```ts
 * import { mintTo } from "thirdweb/extensions/erc721";
 *
 * const transaction = mintTo({
 *  contract,
 *  to: ...,
 *  uri: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function mintTo(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.to, resolvedParams.uri];
            }
            : [options.to, options.uri],
    });
}
exports.mintTo = mintTo;
