// thirdweb/dist/esm/extensions/farcaster/__generated__/IStorageRegistry/read/rentedUnits.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x2751c4fd";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the rentedUnits function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeRentedUnitsResult } from "thirdweb/extensions/farcaster";
 * const result = decodeRentedUnitsResult("...");
 * ```
 */
export function decodeRentedUnitsResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "rentedUnits" function on the contract.
 * @param options - The options for the rentedUnits function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { rentedUnits } from "thirdweb/extensions/farcaster";
 *
 * const result = await rentedUnits();
 *
 * ```
 */
export async function rentedUnits(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
