// thirdweb/dist/esm/extensions/farcaster/__generated__/IStorageRegistry/read/unitPrice.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xe73faa2d";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the unitPrice function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeUnitPriceResult } from "thirdweb/extensions/farcaster";
 * const result = decodeUnitPriceResult("...");
 * ```
 */
export function decodeUnitPriceResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "unitPrice" function on the contract.
 * @param options - The options for the unitPrice function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { unitPrice } from "thirdweb/extensions/farcaster";
 *
 * const result = await unitPrice();
 *
 * ```
 */
export async function unitPrice(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
