// thirdweb/dist/cjs/extensions/erc1155/__generated__/IPack/write/createPack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPack = exports.encodeCreatePackParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x092e6075";
const FN_INPUTS = [
    {
        type: "tuple[]",
        name: "contents",
        components: [
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "uint8",
                name: "tokenType",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "totalAmount",
            },
        ],
    },
    {
        type: "uint256[]",
        name: "numOfRewardUnits",
    },
    {
        type: "string",
        name: "packUri",
    },
    {
        type: "uint128",
        name: "openStartTimestamp",
    },
    {
        type: "uint128",
        name: "amountDistributedPerOpen",
    },
    {
        type: "address",
        name: "recipient",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "packId",
    },
    {
        type: "uint256",
        name: "packTotalSupply",
    },
];
/**
 * Encodes the parameters for the "createPack" function.
 * @param options - The options for the createPack function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```
 * import { encodeCreatePackParams } "thirdweb/extensions/erc1155";
 * const result = encodeCreatePackParams({
 *  contents: ...,
 *  numOfRewardUnits: ...,
 *  packUri: ...,
 *  openStartTimestamp: ...,
 *  amountDistributedPerOpen: ...,
 *  recipient: ...,
 * });
 * ```
 */
function encodeCreatePackParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.contents,
        options.numOfRewardUnits,
        options.packUri,
        options.openStartTimestamp,
        options.amountDistributedPerOpen,
        options.recipient,
    ]);
}
exports.encodeCreatePackParams = encodeCreatePackParams;
/**
 * Calls the "createPack" function on the contract.
 * @param options - The options for the "createPack" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```
 * import { createPack } from "thirdweb/extensions/erc1155";
 *
 * const transaction = createPack({
 *  contents: ...,
 *  numOfRewardUnits: ...,
 *  packUri: ...,
 *  openStartTimestamp: ...,
 *  amountDistributedPerOpen: ...,
 *  recipient: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function createPack(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.contents,
                    resolvedParams.numOfRewardUnits,
                    resolvedParams.packUri,
                    resolvedParams.openStartTimestamp,
                    resolvedParams.amountDistributedPerOpen,
                    resolvedParams.recipient,
                ];
            }
            : [
                options.contents,
                options.numOfRewardUnits,
                options.packUri,
                options.openStartTimestamp,
                options.amountDistributedPerOpen,
                options.recipient,
            ],
    });
}
exports.createPack = createPack;
