// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IRulesEngine/read/getRulesEngineOverride.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRulesEngineOverride = exports.decodeGetRulesEngineOverrideResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xa7145eb4";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "rulesEngineAddress",
    },
];
/**
 * Decodes the result of the getRulesEngineOverride function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { decodeGetRulesEngineOverrideResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeGetRulesEngineOverrideResult("...");
 * ```
 */
function decodeGetRulesEngineOverrideResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetRulesEngineOverrideResult = decodeGetRulesEngineOverrideResult;
/**
 * Calls the "getRulesEngineOverride" function on the contract.
 * @param options - The options for the getRulesEngineOverride function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { getRulesEngineOverride } from "thirdweb/extensions/thirdweb";
 *
 * const result = await getRulesEngineOverride();
 *
 * ```
 */
async function getRulesEngineOverride(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.getRulesEngineOverride = getRulesEngineOverride;
