// thirdweb/dist/cjs/extensions/farcaster/__generated__/IIdRegistry/write/changeRecoveryAddress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.changeRecoveryAddress = exports.encodeChangeRecoveryAddressParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xf1f0b224";
const FN_INPUTS = [
    {
        type: "address",
        name: "recovery",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "changeRecoveryAddress" function.
 * @param options - The options for the changeRecoveryAddress function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeChangeRecoveryAddressParams } "thirdweb/extensions/farcaster";
 * const result = encodeChangeRecoveryAddressParams({
 *  recovery: ...,
 * });
 * ```
 */
function encodeChangeRecoveryAddressParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.recovery]);
}
exports.encodeChangeRecoveryAddressParams = encodeChangeRecoveryAddressParams;
/**
 * Calls the "changeRecoveryAddress" function on the contract.
 * @param options - The options for the "changeRecoveryAddress" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { changeRecoveryAddress } from "thirdweb/extensions/farcaster";
 *
 * const transaction = changeRecoveryAddress({
 *  contract,
 *  recovery: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function changeRecoveryAddress(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.recovery];
            }
            : [options.recovery],
    });
}
exports.changeRecoveryAddress = changeRecoveryAddress;
