// thirdweb/dist/cjs/react/web/wallets/walletConnect/locale.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.walletConnectLocales = void 0;
exports.walletConnectLocales = {
    en_US: {
        scanInstruction: "Scan this with your wallet app to connect",
        openWCModal: "Open Official WalletConnect Modal",
    },
    ja_JP: {
        scanInstruction: '接続するためにウォレットアプリでこちらをスキャンしてください"',
        openWCModal: "公式のWalletConnectモーダルを開く",
    },
    tl_PH: {
        scanInstruction: "I-scan ito gamit ang iyong wallet app para makakonekta",
        openWCModal: "Buksan ang Opisyal na WalletConnect Modal",
    },
    es_ES: {
        scanInstruction: "Escanea esto con tu aplicación de cartera para conectar",
        openWCModal: "Abre el Modal Oficial de WalletConnect",
    },
    vi_VN: {
        scanInstruction: "Quét mã QR bằng ứng dụng ví để kết nối",
        openWCModal: "Mở giao diện WalletConnect",
    },
    de_DE: {
        scanInstruction: "Scanne dies mit deiner Wallet-App, um zu verbinden",
        openWCModal: "Offizielles WalletConnect-Modal öffnen",
    },
};
