// thirdweb/dist/cjs/extensions/common/__generated__/IPrimarySale/write/setPrimarySaleRecipient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setPrimarySaleRecipient = exports.encodeSetPrimarySaleRecipientParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x6f4f2837";
const FN_INPUTS = [
    {
        type: "address",
        name: "_saleRecipient",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setPrimarySaleRecipient" function.
 * @param options - The options for the setPrimarySaleRecipient function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeSetPrimarySaleRecipientParams } "thirdweb/extensions/common";
 * const result = encodeSetPrimarySaleRecipientParams({
 *  saleRecipient: ...,
 * });
 * ```
 */
function encodeSetPrimarySaleRecipientParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.saleRecipient]);
}
exports.encodeSetPrimarySaleRecipientParams = encodeSetPrimarySaleRecipientParams;
/**
 * Calls the "setPrimarySaleRecipient" function on the contract.
 * @param options - The options for the "setPrimarySaleRecipient" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```
 * import { setPrimarySaleRecipient } from "thirdweb/extensions/common";
 *
 * const transaction = setPrimarySaleRecipient({
 *  saleRecipient: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setPrimarySaleRecipient(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.saleRecipient];
            }
            : [options.saleRecipient],
    });
}
exports.setPrimarySaleRecipient = setPrimarySaleRecipient;
