// thirdweb/dist/cjs/extensions/erc1155/__generated__/IERC1155/write/safeTransferFrom.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.safeTransferFrom = exports.encodeSafeTransferFromParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xf242432a";
const FN_INPUTS = [
    {
        type: "address",
        name: "_from",
    },
    {
        type: "address",
        name: "_to",
    },
    {
        type: "uint256",
        name: "tokenId",
    },
    {
        type: "uint256",
        name: "_value",
    },
    {
        type: "bytes",
        name: "_data",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "safeTransferFrom" function.
 * @param options - The options for the safeTransferFrom function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeSafeTransferFromParams } "thirdweb/extensions/erc1155";
 * const result = encodeSafeTransferFromParams({
 *  from: ...,
 *  to: ...,
 *  tokenId: ...,
 *  value: ...,
 *  data: ...,
 * });
 * ```
 */
function encodeSafeTransferFromParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.from,
        options.to,
        options.tokenId,
        options.value,
        options.data,
    ]);
}
exports.encodeSafeTransferFromParams = encodeSafeTransferFromParams;
/**
 * Calls the "safeTransferFrom" function on the contract.
 * @param options - The options for the "safeTransferFrom" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```ts
 * import { safeTransferFrom } from "thirdweb/extensions/erc1155";
 *
 * const transaction = safeTransferFrom({
 *  contract,
 *  from: ...,
 *  to: ...,
 *  tokenId: ...,
 *  value: ...,
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function safeTransferFrom(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.from,
                    resolvedParams.to,
                    resolvedParams.tokenId,
                    resolvedParams.value,
                    resolvedParams.data,
                ];
            }
            : [
                options.from,
                options.to,
                options.tokenId,
                options.value,
                options.data,
            ],
    });
}
exports.safeTransferFrom = safeTransferFrom;
