// thirdweb/dist/cjs/extensions/erc1155/__generated__/IERC1155/write/safeBatchTransferFrom.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.safeBatchTransferFrom = exports.encodeSafeBatchTransferFromParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x2eb2c2d6";
const FN_INPUTS = [
    {
        type: "address",
        name: "_from",
    },
    {
        type: "address",
        name: "_to",
    },
    {
        type: "uint256[]",
        name: "tokenIds",
    },
    {
        type: "uint256[]",
        name: "_values",
    },
    {
        type: "bytes",
        name: "_data",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "safeBatchTransferFrom" function.
 * @param options - The options for the safeBatchTransferFrom function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeSafeBatchTransferFromParams } "thirdweb/extensions/erc1155";
 * const result = encodeSafeBatchTransferFromParams({
 *  from: ...,
 *  to: ...,
 *  tokenIds: ...,
 *  values: ...,
 *  data: ...,
 * });
 * ```
 */
function encodeSafeBatchTransferFromParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.from,
        options.to,
        options.tokenIds,
        options.values,
        options.data,
    ]);
}
exports.encodeSafeBatchTransferFromParams = encodeSafeBatchTransferFromParams;
/**
 * Calls the "safeBatchTransferFrom" function on the contract.
 * @param options - The options for the "safeBatchTransferFrom" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```ts
 * import { safeBatchTransferFrom } from "thirdweb/extensions/erc1155";
 *
 * const transaction = safeBatchTransferFrom({
 *  contract,
 *  from: ...,
 *  to: ...,
 *  tokenIds: ...,
 *  values: ...,
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function safeBatchTransferFrom(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.from,
                    resolvedParams.to,
                    resolvedParams.tokenIds,
                    resolvedParams.values,
                    resolvedParams.data,
                ];
            }
            : [
                options.from,
                options.to,
                options.tokenIds,
                options.values,
                options.data,
            ],
    });
}
exports.safeBatchTransferFrom = safeBatchTransferFrom;
