// thirdweb/dist/esm/extensions/ens/__generated__/AddressResolver/read/addr.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x3b3b57de";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "name",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Encodes the parameters for the "addr" function.
 * @param options - The options for the addr function.
 * @returns The encoded ABI parameters.
 * @extension ENS
 * @example
 * ```
 * import { encodeAddrParams } "thirdweb/extensions/ens";
 * const result = encodeAddrParams({
 *  name: ...,
 * });
 * ```
 */
export function encodeAddrParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.name]);
}
/**
 * Decodes the result of the addr function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ENS
 * @example
 * ```
 * import { decodeAddrResult } from "thirdweb/extensions/ens";
 * const result = decodeAddrResult("...");
 * ```
 */
export function decodeAddrResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "addr" function on the contract.
 * @param options - The options for the addr function.
 * @returns The parsed result of the function call.
 * @extension ENS
 * @example
 * ```
 * import { addr } from "thirdweb/extensions/ens";
 *
 * const result = await addr({
 *  name: ...,
 * });
 *
 * ```
 */
export async function addr(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.name],
    });
}
