// thirdweb/dist/cjs/extensions/marketplace/__generated__/IOffers/write/makeOffer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.makeOffer = exports.encodeMakeOfferParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x016767fa";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "_params",
        components: [
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "quantity",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "uint256",
                name: "totalPrice",
            },
            {
                type: "uint256",
                name: "expirationTimestamp",
            },
        ],
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "offerId",
    },
];
/**
 * Encodes the parameters for the "makeOffer" function.
 * @param options - The options for the makeOffer function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { encodeMakeOfferParams } "thirdweb/extensions/marketplace";
 * const result = encodeMakeOfferParams({
 *  params: ...,
 * });
 * ```
 */
function encodeMakeOfferParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.params]);
}
exports.encodeMakeOfferParams = encodeMakeOfferParams;
/**
 * Calls the "makeOffer" function on the contract.
 * @param options - The options for the "makeOffer" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { makeOffer } from "thirdweb/extensions/marketplace";
 *
 * const transaction = makeOffer({
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function makeOffer(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
exports.makeOffer = makeOffer;
