// thirdweb/dist/cjs/extensions/common/__generated__/IRoyalty/write/setDefaultRoyaltyInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setDefaultRoyaltyInfo = exports.encodeSetDefaultRoyaltyInfoParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x600dd5ea";
const FN_INPUTS = [
    {
        type: "address",
        name: "_royaltyRecipient",
    },
    {
        type: "uint256",
        name: "_royaltyBps",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setDefaultRoyaltyInfo" function.
 * @param options - The options for the setDefaultRoyaltyInfo function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```ts
 * import { encodeSetDefaultRoyaltyInfoParams } "thirdweb/extensions/common";
 * const result = encodeSetDefaultRoyaltyInfoParams({
 *  royaltyRecipient: ...,
 *  royaltyBps: ...,
 * });
 * ```
 */
function encodeSetDefaultRoyaltyInfoParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.royaltyRecipient,
        options.royaltyBps,
    ]);
}
exports.encodeSetDefaultRoyaltyInfoParams = encodeSetDefaultRoyaltyInfoParams;
/**
 * Calls the "setDefaultRoyaltyInfo" function on the contract.
 * @param options - The options for the "setDefaultRoyaltyInfo" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```ts
 * import { setDefaultRoyaltyInfo } from "thirdweb/extensions/common";
 *
 * const transaction = setDefaultRoyaltyInfo({
 *  contract,
 *  royaltyRecipient: ...,
 *  royaltyBps: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setDefaultRoyaltyInfo(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.royaltyRecipient,
                    resolvedParams.royaltyBps,
                ];
            }
            : [options.royaltyRecipient, options.royaltyBps],
    });
}
exports.setDefaultRoyaltyInfo = setDefaultRoyaltyInfo;
