// thirdweb/dist/cjs/extensions/common/__generated__/IRoyalty/write/setRoyaltyInfoForToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setRoyaltyInfoForToken = exports.encodeSetRoyaltyInfoForTokenParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x9bcf7a15";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "tokenId",
    },
    {
        type: "address",
        name: "recipient",
    },
    {
        type: "uint256",
        name: "bps",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setRoyaltyInfoForToken" function.
 * @param options - The options for the setRoyaltyInfoForToken function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeSetRoyaltyInfoForTokenParams } "thirdweb/extensions/common";
 * const result = encodeSetRoyaltyInfoForTokenParams({
 *  tokenId: ...,
 *  recipient: ...,
 *  bps: ...,
 * });
 * ```
 */
function encodeSetRoyaltyInfoForTokenParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.tokenId,
        options.recipient,
        options.bps,
    ]);
}
exports.encodeSetRoyaltyInfoForTokenParams = encodeSetRoyaltyInfoForTokenParams;
/**
 * Calls the "setRoyaltyInfoForToken" function on the contract.
 * @param options - The options for the "setRoyaltyInfoForToken" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```
 * import { setRoyaltyInfoForToken } from "thirdweb/extensions/common";
 *
 * const transaction = setRoyaltyInfoForToken({
 *  tokenId: ...,
 *  recipient: ...,
 *  bps: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setRoyaltyInfoForToken(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.tokenId,
                    resolvedParams.recipient,
                    resolvedParams.bps,
                ];
            }
            : [options.tokenId, options.recipient, options.bps],
    });
}
exports.setRoyaltyInfoForToken = setRoyaltyInfoForToken;
