// thirdweb/dist/cjs/extensions/erc1155/__generated__/IDrop1155/read/getClaimConditionById.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getClaimConditionById = exports.decodeGetClaimConditionByIdResult = exports.encodeGetClaimConditionByIdParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xd45b28d7";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_tokenId",
    },
    {
        type: "uint256",
        name: "_conditionId",
    },
];
const FN_OUTPUTS = [
    {
        type: "tuple",
        name: "condition",
        components: [
            {
                type: "uint256",
                name: "startTimestamp",
            },
            {
                type: "uint256",
                name: "maxClaimableSupply",
            },
            {
                type: "uint256",
                name: "supplyClaimed",
            },
            {
                type: "uint256",
                name: "quantityLimitPerWallet",
            },
            {
                type: "bytes32",
                name: "merkleRoot",
            },
            {
                type: "uint256",
                name: "pricePerToken",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "string",
                name: "metadata",
            },
        ],
    },
];
/**
 * Encodes the parameters for the "getClaimConditionById" function.
 * @param options - The options for the getClaimConditionById function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeGetClaimConditionByIdParams } "thirdweb/extensions/erc1155";
 * const result = encodeGetClaimConditionByIdParams({
 *  tokenId: ...,
 *  conditionId: ...,
 * });
 * ```
 */
function encodeGetClaimConditionByIdParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.tokenId, options.conditionId]);
}
exports.encodeGetClaimConditionByIdParams = encodeGetClaimConditionByIdParams;
/**
 * Decodes the result of the getClaimConditionById function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC1155
 * @example
 * ```ts
 * import { decodeGetClaimConditionByIdResult } from "thirdweb/extensions/erc1155";
 * const result = decodeGetClaimConditionByIdResult("...");
 * ```
 */
function decodeGetClaimConditionByIdResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetClaimConditionByIdResult = decodeGetClaimConditionByIdResult;
/**
 * Calls the "getClaimConditionById" function on the contract.
 * @param options - The options for the getClaimConditionById function.
 * @returns The parsed result of the function call.
 * @extension ERC1155
 * @example
 * ```ts
 * import { getClaimConditionById } from "thirdweb/extensions/erc1155";
 *
 * const result = await getClaimConditionById({
 *  tokenId: ...,
 *  conditionId: ...,
 * });
 *
 * ```
 */
async function getClaimConditionById(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.tokenId, options.conditionId],
    });
}
exports.getClaimConditionById = getClaimConditionById;
