// thirdweb/dist/cjs/extensions/erc1155/__generated__/IERC1155Receiver/read/supportsInterface.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.supportsInterface = exports.decodeSupportsInterfaceResult = exports.encodeSupportsInterfaceParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x01ffc9a7";
const FN_INPUTS = [
    {
        type: "bytes4",
        name: "interfaceId",
    },
];
const FN_OUTPUTS = [
    {
        type: "bool",
    },
];
/**
 * Encodes the parameters for the "supportsInterface" function.
 * @param options - The options for the supportsInterface function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeSupportsInterfaceParams } "thirdweb/extensions/erc1155";
 * const result = encodeSupportsInterfaceParams({
 *  interfaceId: ...,
 * });
 * ```
 */
function encodeSupportsInterfaceParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.interfaceId]);
}
exports.encodeSupportsInterfaceParams = encodeSupportsInterfaceParams;
/**
 * Decodes the result of the supportsInterface function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC1155
 * @example
 * ```ts
 * import { decodeSupportsInterfaceResult } from "thirdweb/extensions/erc1155";
 * const result = decodeSupportsInterfaceResult("...");
 * ```
 */
function decodeSupportsInterfaceResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeSupportsInterfaceResult = decodeSupportsInterfaceResult;
/**
 * Calls the "supportsInterface" function on the contract.
 * @param options - The options for the supportsInterface function.
 * @returns The parsed result of the function call.
 * @extension ERC1155
 * @example
 * ```ts
 * import { supportsInterface } from "thirdweb/extensions/erc1155";
 *
 * const result = await supportsInterface({
 *  interfaceId: ...,
 * });
 *
 * ```
 */
async function supportsInterface(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.interfaceId],
    });
}
exports.supportsInterface = supportsInterface;
