// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721Enumerable/read/nextTokenIdToMint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nextTokenIdToMint = exports.decodeNextTokenIdToMintResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x3b1475a7";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the nextTokenIdToMint function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeNextTokenIdToMintResult } from "thirdweb/extensions/erc721";
 * const result = decodeNextTokenIdToMintResult("...");
 * ```
 */
function decodeNextTokenIdToMintResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeNextTokenIdToMintResult = decodeNextTokenIdToMintResult;
/**
 * Calls the "nextTokenIdToMint" function on the contract.
 * @param options - The options for the nextTokenIdToMint function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { nextTokenIdToMint } from "thirdweb/extensions/erc721";
 *
 * const result = await nextTokenIdToMint();
 *
 * ```
 */
async function nextTokenIdToMint(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.nextTokenIdToMint = nextTokenIdToMint;
