// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721Enumerable/read/tokenOfOwnerByIndex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tokenOfOwnerByIndex = exports.decodeTokenOfOwnerByIndexResult = exports.encodeTokenOfOwnerByIndexParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x2f745c59";
const FN_INPUTS = [
    {
        type: "address",
        name: "_owner",
    },
    {
        type: "uint256",
        name: "_index",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Encodes the parameters for the "tokenOfOwnerByIndex" function.
 * @param options - The options for the tokenOfOwnerByIndex function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeTokenOfOwnerByIndexParams } "thirdweb/extensions/erc721";
 * const result = encodeTokenOfOwnerByIndexParams({
 *  owner: ...,
 *  index: ...,
 * });
 * ```
 */
function encodeTokenOfOwnerByIndexParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.owner, options.index]);
}
exports.encodeTokenOfOwnerByIndexParams = encodeTokenOfOwnerByIndexParams;
/**
 * Decodes the result of the tokenOfOwnerByIndex function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeTokenOfOwnerByIndexResult } from "thirdweb/extensions/erc721";
 * const result = decodeTokenOfOwnerByIndexResult("...");
 * ```
 */
function decodeTokenOfOwnerByIndexResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTokenOfOwnerByIndexResult = decodeTokenOfOwnerByIndexResult;
/**
 * Calls the "tokenOfOwnerByIndex" function on the contract.
 * @param options - The options for the tokenOfOwnerByIndex function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { tokenOfOwnerByIndex } from "thirdweb/extensions/erc721";
 *
 * const result = await tokenOfOwnerByIndex({
 *  owner: ...,
 *  index: ...,
 * });
 *
 * ```
 */
async function tokenOfOwnerByIndex(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.owner, options.index],
    });
}
exports.tokenOfOwnerByIndex = tokenOfOwnerByIndex;
