// thirdweb/dist/esm/extensions/erc721/__generated__/IERC721Enumerable/read/tokenByIndex.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x4f6ccce7";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_index",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Encodes the parameters for the "tokenByIndex" function.
 * @param options - The options for the tokenByIndex function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeTokenByIndexParams } "thirdweb/extensions/erc721";
 * const result = encodeTokenByIndexParams({
 *  index: ...,
 * });
 * ```
 */
export function encodeTokenByIndexParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.index]);
}
/**
 * Decodes the result of the tokenByIndex function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeTokenByIndexResult } from "thirdweb/extensions/erc721";
 * const result = decodeTokenByIndexResult("...");
 * ```
 */
export function decodeTokenByIndexResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "tokenByIndex" function on the contract.
 * @param options - The options for the tokenByIndex function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { tokenByIndex } from "thirdweb/extensions/erc721";
 *
 * const result = await tokenByIndex({
 *  index: ...,
 * });
 *
 * ```
 */
export async function tokenByIndex(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.index],
    });
}
