// thirdweb/dist/cjs/extensions/common/__generated__/IPermissionsEnumerable/read/getRoleMemberCount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRoleMemberCount = exports.decodeGetRoleMemberCountResult = exports.encodeGetRoleMemberCountParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xca15c873";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "role",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Encodes the parameters for the "getRoleMemberCount" function.
 * @param options - The options for the getRoleMemberCount function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeGetRoleMemberCountParams } "thirdweb/extensions/common";
 * const result = encodeGetRoleMemberCountParams({
 *  role: ...,
 * });
 * ```
 */
function encodeGetRoleMemberCountParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.role]);
}
exports.encodeGetRoleMemberCountParams = encodeGetRoleMemberCountParams;
/**
 * Decodes the result of the getRoleMemberCount function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```
 * import { decodeGetRoleMemberCountResult } from "thirdweb/extensions/common";
 * const result = decodeGetRoleMemberCountResult("...");
 * ```
 */
function decodeGetRoleMemberCountResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetRoleMemberCountResult = decodeGetRoleMemberCountResult;
/**
 * Calls the "getRoleMemberCount" function on the contract.
 * @param options - The options for the getRoleMemberCount function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```
 * import { getRoleMemberCount } from "thirdweb/extensions/common";
 *
 * const result = await getRoleMemberCount({
 *  role: ...,
 * });
 *
 * ```
 */
async function getRoleMemberCount(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.role],
    });
}
exports.getRoleMemberCount = getRoleMemberCount;
