// thirdweb/dist/cjs/extensions/common/__generated__/IPermissionsEnumerable/read/getRoleAdmin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRoleAdmin = exports.decodeGetRoleAdminResult = exports.encodeGetRoleAdminParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x248a9ca3";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "role",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes32",
    },
];
/**
 * Encodes the parameters for the "getRoleAdmin" function.
 * @param options - The options for the getRoleAdmin function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```ts
 * import { encodeGetRoleAdminParams } "thirdweb/extensions/common";
 * const result = encodeGetRoleAdminParams({
 *  role: ...,
 * });
 * ```
 */
function encodeGetRoleAdminParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.role]);
}
exports.encodeGetRoleAdminParams = encodeGetRoleAdminParams;
/**
 * Decodes the result of the getRoleAdmin function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```ts
 * import { decodeGetRoleAdminResult } from "thirdweb/extensions/common";
 * const result = decodeGetRoleAdminResult("...");
 * ```
 */
function decodeGetRoleAdminResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetRoleAdminResult = decodeGetRoleAdminResult;
/**
 * Calls the "getRoleAdmin" function on the contract.
 * @param options - The options for the getRoleAdmin function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```ts
 * import { getRoleAdmin } from "thirdweb/extensions/common";
 *
 * const result = await getRoleAdmin({
 *  role: ...,
 * });
 *
 * ```
 */
async function getRoleAdmin(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.role],
    });
}
exports.getRoleAdmin = getRoleAdmin;
