// thirdweb/dist/cjs/exports/wallets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getWalletInfo = exports.getAllWalletsList = exports.injectedProvider = exports.privateKeyAccount = exports.getWalletBalance = exports.embeddedWallet = exports.smartWallet = exports.createWallet = void 0;
var create_wallet_js_1 = require("../wallets/create-wallet.js");
Object.defineProperty(exports, "createWallet", { enumerable: true, get: function () { return create_wallet_js_1.createWallet; } });
Object.defineProperty(exports, "smartWallet", { enumerable: true, get: function () { return create_wallet_js_1.smartWallet; } });
Object.defineProperty(exports, "embeddedWallet", { enumerable: true, get: function () { return create_wallet_js_1.embeddedWallet; } });
var getWalletBalance_js_1 = require("../wallets/utils/getWalletBalance.js");
Object.defineProperty(exports, "getWalletBalance", { enumerable: true, get: function () { return getWalletBalance_js_1.getWalletBalance; } });
// private-key
var private_key_js_1 = require("../wallets/private-key.js");
Object.defineProperty(exports, "privateKeyAccount", { enumerable: true, get: function () { return private_key_js_1.privateKeyAccount; } });
// injected
var mipdStore_js_1 = require("../wallets/injected/mipdStore.js");
Object.defineProperty(exports, "injectedProvider", { enumerable: true, get: function () { return mipdStore_js_1.injectedProvider; } });
var getAllWalletsList_js_1 = require("../wallets/getAllWalletsList.js");
Object.defineProperty(exports, "getAllWalletsList", { enumerable: true, get: function () { return getAllWalletsList_js_1.getAllWalletsList; } });
var getWalletInfo_js_1 = require("../wallets/__generated__/getWalletInfo.js");
Object.defineProperty(exports, "getWalletInfo", { enumerable: true, get: function () { return getWalletInfo_js_1.getWalletInfo; } });
