// thirdweb/dist/esm/extensions/farcaster/__generated__/IKeyGateway/write/add.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x22b1a414";
const FN_INPUTS = [
    {
        type: "uint32",
        name: "keyType",
    },
    {
        type: "bytes",
        name: "key",
    },
    {
        type: "uint8",
        name: "metadataType",
    },
    {
        type: "bytes",
        name: "metadata",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "add" function.
 * @param options - The options for the add function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeAddParams } "thirdweb/extensions/farcaster";
 * const result = encodeAddParams({
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 * });
 * ```
 */
export function encodeAddParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.keyType,
        options.key,
        options.metadataType,
        options.metadata,
    ]);
}
/**
 * Calls the "add" function on the contract.
 * @param options - The options for the "add" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```
 * import { add } from "thirdweb/extensions/farcaster";
 *
 * const transaction = add({
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function add(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.keyType,
                    resolvedParams.key,
                    resolvedParams.metadataType,
                    resolvedParams.metadata,
                ];
            }
            : [
                options.keyType,
                options.key,
                options.metadataType,
                options.metadata,
            ],
    });
}
