// thirdweb/dist/esm/extensions/farcaster/__generated__/IKeyGateway/write/add.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { once } from "../../../../../utils/promise/once.js";
import { detectMethod } from "../../../../../utils/bytecode/detectExtension.js";
export const FN_SELECTOR = "0x22b1a414";
const FN_INPUTS = [
    {
        type: "uint32",
        name: "keyType",
    },
    {
        type: "bytes",
        name: "key",
    },
    {
        type: "uint8",
        name: "metadataType",
    },
    {
        type: "bytes",
        name: "metadata",
    },
];
const FN_OUTPUTS = [];
/**
 * Checks if the `add` method is supported by the given contract.
 * @param contract The ThirdwebContract.
 * @returns A promise that resolves to a boolean indicating if the `add` method is supported.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { isAddSupported } from "thirdweb/extensions/farcaster";
 *
 * const supported = await isAddSupported(contract);
 * ```
 */
export async function isAddSupported(contract) {
    return detectMethod({
        contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
    });
}
/**
 * Encodes the parameters for the "add" function.
 * @param options - The options for the add function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeAddParams } "thirdweb/extensions/farcaster";
 * const result = encodeAddParams({
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 * });
 * ```
 */
export function encodeAddParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.keyType,
        options.key,
        options.metadataType,
        options.metadata,
    ]);
}
/**
 * Encodes the "add" function into a Hex string with its parameters.
 * @param options - The options for the add function.
 * @returns The encoded hexadecimal string.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeAdd } "thirdweb/extensions/farcaster";
 * const result = encodeAdd({
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 * });
 * ```
 */
export function encodeAdd(options) {
    // we do a "manual" concat here to avoid the overhead of the "concatHex" function
    // we can do this because we know the specific formats of the values
    return (FN_SELECTOR +
        encodeAddParams(options).slice(2));
}
/**
 * Calls the "add" function on the contract.
 * @param options - The options for the "add" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { add } from "thirdweb/extensions/farcaster";
 *
 * const transaction = add({
 *  contract,
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function add(options) {
    const asyncOptions = once(async () => {
        return "asyncParams" in options ? await options.asyncParams() : options;
    });
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: async () => {
            const resolvedOptions = await asyncOptions();
            return [
                resolvedOptions.keyType,
                resolvedOptions.key,
                resolvedOptions.metadataType,
                resolvedOptions.metadata,
            ];
        },
        value: async () => (await asyncOptions()).overrides?.value,
    });
}
