// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721A/write/approve.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.approve = exports.encodeApproveParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x095ea7b3";
const FN_INPUTS = [
    {
        type: "address",
        name: "to",
    },
    {
        type: "uint256",
        name: "tokenId",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "approve" function.
 * @param options - The options for the approve function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeApproveParams } "thirdweb/extensions/erc721";
 * const result = encodeApproveParams({
 *  to: ...,
 *  tokenId: ...,
 * });
 * ```
 */
function encodeApproveParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.to, options.tokenId]);
}
exports.encodeApproveParams = encodeApproveParams;
/**
 * Calls the "approve" function on the contract.
 * @param options - The options for the "approve" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```
 * import { approve } from "thirdweb/extensions/erc721";
 *
 * const transaction = approve({
 *  to: ...,
 *  tokenId: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function approve(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.to, resolvedParams.tokenId];
            }
            : [options.to, options.tokenId],
    });
}
exports.approve = approve;
