// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721A/write/setApprovalForAll.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setApprovalForAll = exports.encodeSetApprovalForAllParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa22cb465";
const FN_INPUTS = [
    {
        type: "address",
        name: "operator",
    },
    {
        type: "bool",
        name: "_approved",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setApprovalForAll" function.
 * @param options - The options for the setApprovalForAll function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeSetApprovalForAllParams } "thirdweb/extensions/erc721";
 * const result = encodeSetApprovalForAllParams({
 *  operator: ...,
 *  approved: ...,
 * });
 * ```
 */
function encodeSetApprovalForAllParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.operator, options.approved]);
}
exports.encodeSetApprovalForAllParams = encodeSetApprovalForAllParams;
/**
 * Calls the "setApprovalForAll" function on the contract.
 * @param options - The options for the "setApprovalForAll" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```
 * import { setApprovalForAll } from "thirdweb/extensions/erc721";
 *
 * const transaction = setApprovalForAll({
 *  operator: ...,
 *  approved: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setApprovalForAll(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.operator, resolvedParams.approved];
            }
            : [options.operator, options.approved],
    });
}
exports.setApprovalForAll = setApprovalForAll;
