// thirdweb/dist/cjs/extensions/erc1155/__generated__/ISignatureMintERC1155/write/mintWithSignature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mintWithSignature = exports.encodeMintWithSignatureParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x98a6e993";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "req",
        components: [
            {
                type: "address",
                name: "to",
            },
            {
                type: "address",
                name: "royaltyRecipient",
            },
            {
                type: "uint256",
                name: "royaltyBps",
            },
            {
                type: "address",
                name: "primarySaleRecipient",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "string",
                name: "uri",
            },
            {
                type: "uint256",
                name: "quantity",
            },
            {
                type: "uint256",
                name: "pricePerToken",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "uint128",
                name: "validityStartTimestamp",
            },
            {
                type: "uint128",
                name: "validityEndTimestamp",
            },
            {
                type: "bytes32",
                name: "uid",
            },
        ],
    },
    {
        type: "bytes",
        name: "signature",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "signer",
    },
];
/**
 * Encodes the parameters for the "mintWithSignature" function.
 * @param options - The options for the mintWithSignature function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeMintWithSignatureParams } "thirdweb/extensions/erc1155";
 * const result = encodeMintWithSignatureParams({
 *  req: ...,
 *  signature: ...,
 * });
 * ```
 */
function encodeMintWithSignatureParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.req, options.signature]);
}
exports.encodeMintWithSignatureParams = encodeMintWithSignatureParams;
/**
 * Calls the "mintWithSignature" function on the contract.
 * @param options - The options for the "mintWithSignature" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```ts
 * import { mintWithSignature } from "thirdweb/extensions/erc1155";
 *
 * const transaction = mintWithSignature({
 *  contract,
 *  req: ...,
 *  signature: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function mintWithSignature(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.req, resolvedParams.signature];
            }
            : [options.req, options.signature],
    });
}
exports.mintWithSignature = mintWithSignature;
