// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountFactory/write/createAccount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAccount = exports.encodeCreateAccountParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xd8fd8f44";
const FN_INPUTS = [
    {
        type: "address",
        name: "admin",
    },
    {
        type: "bytes",
        name: "_data",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "account",
    },
];
/**
 * Encodes the parameters for the "createAccount" function.
 * @param options - The options for the createAccount function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```
 * import { encodeCreateAccountParams } "thirdweb/extensions/erc4337";
 * const result = encodeCreateAccountParams({
 *  admin: ...,
 *  data: ...,
 * });
 * ```
 */
function encodeCreateAccountParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.admin, options.data]);
}
exports.encodeCreateAccountParams = encodeCreateAccountParams;
/**
 * Calls the "createAccount" function on the contract.
 * @param options - The options for the "createAccount" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```
 * import { createAccount } from "thirdweb/extensions/erc4337";
 *
 * const transaction = createAccount({
 *  admin: ...,
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function createAccount(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.admin, resolvedParams.data];
            }
            : [options.admin, options.data],
    });
}
exports.createAccount = createAccount;
