// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountFactory/write/onSignerRemoved.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.onSignerRemoved = exports.encodeOnSignerRemovedParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x0db33003";
const FN_INPUTS = [
    {
        type: "address",
        name: "signer",
    },
    {
        type: "address",
        name: "creatorAdmin",
    },
    {
        type: "bytes",
        name: "data",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "onSignerRemoved" function.
 * @param options - The options for the onSignerRemoved function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```ts
 * import { encodeOnSignerRemovedParams } "thirdweb/extensions/erc4337";
 * const result = encodeOnSignerRemovedParams({
 *  signer: ...,
 *  creatorAdmin: ...,
 *  data: ...,
 * });
 * ```
 */
function encodeOnSignerRemovedParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.signer,
        options.creatorAdmin,
        options.data,
    ]);
}
exports.encodeOnSignerRemovedParams = encodeOnSignerRemovedParams;
/**
 * Calls the "onSignerRemoved" function on the contract.
 * @param options - The options for the "onSignerRemoved" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```ts
 * import { onSignerRemoved } from "thirdweb/extensions/erc4337";
 *
 * const transaction = onSignerRemoved({
 *  contract,
 *  signer: ...,
 *  creatorAdmin: ...,
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function onSignerRemoved(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.signer,
                    resolvedParams.creatorAdmin,
                    resolvedParams.data,
                ];
            }
            : [options.signer, options.creatorAdmin, options.data],
    });
}
exports.onSignerRemoved = onSignerRemoved;
