// thirdweb/dist/cjs/wallets/manager/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStoredActiveWalletId = exports.getStoredConnectedWalletIds = exports.createConnectionManager = void 0;
const computedStore_js_1 = require("../../reactive/computedStore.js");
const effect_js_1 = require("../../reactive/effect.js");
const store_js_1 = require("../../reactive/store.js");
const walletStorage_js_1 = require("../storage/walletStorage.js");
const CONNECTED_WALLET_IDS = "thirdweb:connected-wallet-ids";
const ACTIVE_WALLET_ID = "thirdweb:active-wallet-id";
/**
 * Create a connection manager for Wallet connections
 * @param storage - An instance of type [`AsyncStorage`](https://portal.thirdweb.com/references/typescript/v5/AsyncStorage)
 * @example
 * ```ts
 * const manager = createConnectionManager();
 * ```
 * @returns A connection manager object
 * @walletUtils
 */
function createConnectionManager(storage) {
    // stores
    // active wallet/account
    const activeWalletStore = (0, store_js_1.createStore)(undefined);
    const activeAccountStore = (0, store_js_1.createStore)(undefined);
    const activeWalletChainStore = (0, store_js_1.createStore)(undefined);
    const activeWalletConnectionStatusStore = (0, store_js_1.createStore)("disconnected");
    // other connected accounts
    const walletIdToConnectedWalletMap = (0, store_js_1.createStore)(new Map());
    const isAutoConnecting = (0, store_js_1.createStore)(false);
    const connectedWallets = (0, computedStore_js_1.computedStore)(() => {
        return Array.from(walletIdToConnectedWalletMap.getValue().values());
    }, [walletIdToConnectedWalletMap]);
    // actions
    const addConnectedWallet = (wallet) => {
        const oldValue = walletIdToConnectedWalletMap.getValue();
        const newValue = new Map(oldValue);
        newValue.set(wallet.id, wallet);
        walletIdToConnectedWalletMap.setValue(newValue);
    };
    const removeConnectedWallet = (wallet) => {
        const oldValue = walletIdToConnectedWalletMap.getValue();
        const newValue = new Map(oldValue);
        newValue.delete(wallet.id);
        walletIdToConnectedWalletMap.setValue(newValue);
    };
    const onWalletDisconnect = (wallet) => {
        const currentMap = walletIdToConnectedWalletMap.getValue();
        (0, walletStorage_js_1.deleteConnectParamsFromStorage)(storage, wallet.id);
        const newMap = new Map(currentMap);
        newMap.delete(wallet.id);
        walletIdToConnectedWalletMap.setValue(newMap);
        // if disconnecting the active wallet
        if (activeWalletStore.getValue() === wallet) {
            activeWalletStore.setValue(undefined);
            activeAccountStore.setValue(undefined);
            activeWalletChainStore.setValue(undefined);
            activeWalletConnectionStatusStore.setValue("disconnected");
        }
    };
    const disconnectWallet = (wallet) => {
        onWalletDisconnect(wallet);
        wallet.disconnect();
    };
    const setActiveWallet = (wallet) => {
        const account = wallet.getAccount();
        if (!account) {
            throw new Error("Can not a wallet without an account as active");
        }
        // also add it to connected wallets if it's not already there
        const _connectedWalletsMap = walletIdToConnectedWalletMap.getValue();
        if (!_connectedWalletsMap.has(wallet.id)) {
            addConnectedWallet(wallet);
        }
        // update active states
        activeWalletStore.setValue(wallet);
        activeAccountStore.setValue(account);
        activeWalletChainStore.setValue(wallet.getChain());
        activeWalletConnectionStatusStore.setValue("connected");
        // setup listeners
        const onAccountsChanged = (newAccount) => {
            activeAccountStore.setValue(newAccount);
        };
        const unsubAccounts = wallet.subscribe("accountChanged", onAccountsChanged);
        const unsubChainChanged = wallet.subscribe("chainChanged", (chain) => activeWalletChainStore.setValue(chain));
        const unsubDisconnect = wallet.subscribe("disconnect", () => {
            handleDisconnect();
        });
        const handleDisconnect = () => {
            onWalletDisconnect(wallet);
            unsubAccounts();
            unsubChainChanged();
            unsubDisconnect();
        };
    };
    // side effects
    // save last connected wallet ids to storage
    (0, effect_js_1.effect)(() => {
        const accounts = connectedWallets.getValue();
        const ids = accounts.map((acc) => acc?.id).filter((c) => !!c);
        storage.setItem(CONNECTED_WALLET_IDS, JSON.stringify(ids));
    }, [connectedWallets], false);
    // save active wallet id to storage
    (0, effect_js_1.effect)(() => {
        const value = activeWalletStore.getValue()?.id;
        if (value) {
            storage.setItem(ACTIVE_WALLET_ID, value);
        }
        else {
            storage.removeItem(ACTIVE_WALLET_ID);
        }
    }, [activeWalletStore], false);
    const switchActiveWalletChain = async (chain) => {
        const wallet = activeWalletStore.getValue();
        if (!wallet) {
            throw new Error("no wallet found");
        }
        if (!wallet.switchChain) {
            throw new Error("wallet does not support switching chains");
        }
        await wallet.switchChain(chain);
        // for wallets that dont implement events, just set it manually
        activeWalletChainStore.setValue(wallet.getChain());
    };
    return {
        // account
        activeWalletStore: activeWalletStore,
        activeAccountStore: activeAccountStore,
        connectedWallets,
        addConnectedWallet,
        disconnectWallet,
        setActiveWallet,
        activeWalletChainStore: activeWalletChainStore,
        switchActiveWalletChain,
        activeWalletConnectionStatusStore: activeWalletConnectionStatusStore,
        isAutoConnecting,
        removeConnectedWallet,
    };
}
exports.createConnectionManager = createConnectionManager;
/**
 *
 * @internal
 */
async function getStoredConnectedWalletIds(storage) {
    try {
        const value = await storage.getItem(CONNECTED_WALLET_IDS);
        if (value) {
            return JSON.parse(value);
        }
        return [];
    }
    catch {
        return [];
    }
}
exports.getStoredConnectedWalletIds = getStoredConnectedWalletIds;
/**
 * @internal
 */
async function getStoredActiveWalletId(storage) {
    try {
        const value = await storage.getItem(ACTIVE_WALLET_ID);
        if (value) {
            return value;
        }
        return null;
    }
    catch {
        return null;
    }
}
exports.getStoredActiveWalletId = getStoredActiveWalletId;
