// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IThirdwebContract/read/contractType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.contractType = exports.decodeContractTypeResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xcb2ef6f7";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "bytes32",
    },
];
/**
 * Decodes the result of the contractType function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeContractTypeResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeContractTypeResult("...");
 * ```
 */
function decodeContractTypeResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeContractTypeResult = decodeContractTypeResult;
/**
 * Calls the "contractType" function on the contract.
 * @param options - The options for the contractType function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { contractType } from "thirdweb/extensions/thirdweb";
 *
 * const result = await contractType();
 *
 * ```
 */
async function contractType(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.contractType = contractType;
