// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IThirdwebContract/read/contractVersion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.contractVersion = exports.decodeContractVersionResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xa0a8e460";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint8",
    },
];
/**
 * Decodes the result of the contractVersion function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeContractVersionResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeContractVersionResult("...");
 * ```
 */
function decodeContractVersionResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeContractVersionResult = decodeContractVersionResult;
/**
 * Calls the "contractVersion" function on the contract.
 * @param options - The options for the contractVersion function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { contractVersion } from "thirdweb/extensions/thirdweb";
 *
 * const result = await contractVersion();
 *
 * ```
 */
async function contractVersion(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.contractVersion = contractVersion;
