// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IThirdwebContract/read/contractURI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.contractURI = exports.decodeContractURIResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xe8a3d485";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "string",
    },
];
/**
 * Decodes the result of the contractURI function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeContractURIResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeContractURIResult("...");
 * ```
 */
function decodeContractURIResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeContractURIResult = decodeContractURIResult;
/**
 * Calls the "contractURI" function on the contract.
 * @param options - The options for the contractURI function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { contractURI } from "thirdweb/extensions/thirdweb";
 *
 * const result = await contractURI();
 *
 * ```
 */
async function contractURI(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.contractURI = contractURI;
