// thirdweb/dist/esm/extensions/erc20/__generated__/IDropERC20/write/claim.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x5ab31c1a";
const FN_INPUTS = [
    {
        type: "address",
        name: "receiver",
    },
    {
        type: "uint256",
        name: "quantity",
    },
    {
        type: "address",
        name: "currency",
    },
    {
        type: "uint256",
        name: "pricePerToken",
    },
    {
        type: "tuple",
        name: "allowlistProof",
        components: [
            {
                type: "bytes32[]",
                name: "proof",
            },
            {
                type: "uint256",
                name: "maxQuantityInAllowlist",
            },
        ],
    },
    {
        type: "bytes",
        name: "data",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "claim" function.
 * @param options - The options for the claim function.
 * @returns The encoded ABI parameters.
 * @extension ERC20
 * @example
 * ```
 * import { encodeClaimParams } "thirdweb/extensions/erc20";
 * const result = encodeClaimParams({
 *  receiver: ...,
 *  quantity: ...,
 *  currency: ...,
 *  pricePerToken: ...,
 *  allowlistProof: ...,
 *  data: ...,
 * });
 * ```
 */
export function encodeClaimParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.receiver,
        options.quantity,
        options.currency,
        options.pricePerToken,
        options.allowlistProof,
        options.data,
    ]);
}
/**
 * Calls the "claim" function on the contract.
 * @param options - The options for the "claim" function.
 * @returns A prepared transaction object.
 * @extension ERC20
 * @example
 * ```
 * import { claim } from "thirdweb/extensions/erc20";
 *
 * const transaction = claim({
 *  receiver: ...,
 *  quantity: ...,
 *  currency: ...,
 *  pricePerToken: ...,
 *  allowlistProof: ...,
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function claim(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.receiver,
                    resolvedParams.quantity,
                    resolvedParams.currency,
                    resolvedParams.pricePerToken,
                    resolvedParams.allowlistProof,
                    resolvedParams.data,
                ];
            }
            : [
                options.receiver,
                options.quantity,
                options.currency,
                options.pricePerToken,
                options.allowlistProof,
                options.data,
            ],
    });
}
