// thirdweb/dist/esm/extensions/erc721/__generated__/IDelayedReveal/write/reveal.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0xce805642";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "identifier",
    },
    {
        type: "bytes",
        name: "key",
    },
];
const FN_OUTPUTS = [
    {
        type: "string",
        name: "revealedURI",
    },
];
/**
 * Encodes the parameters for the "reveal" function.
 * @param options - The options for the reveal function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeRevealParams } "thirdweb/extensions/erc721";
 * const result = encodeRevealParams({
 *  identifier: ...,
 *  key: ...,
 * });
 * ```
 */
export function encodeRevealParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.identifier, options.key]);
}
/**
 * Calls the "reveal" function on the contract.
 * @param options - The options for the "reveal" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```ts
 * import { reveal } from "thirdweb/extensions/erc721";
 *
 * const transaction = reveal({
 *  contract,
 *  identifier: ...,
 *  key: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function reveal(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.identifier, resolvedParams.key];
            }
            : [options.identifier, options.key],
    });
}
