// thirdweb/dist/esm/extensions/prebuilts/__generated__/TokenERC20/write/initialize.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0xdfad80a6";
const FN_INPUTS = [
    {
        type: "address",
        name: "_defaultAdmin",
    },
    {
        type: "string",
        name: "_name",
    },
    {
        type: "string",
        name: "_symbol",
    },
    {
        type: "string",
        name: "_contractURI",
    },
    {
        type: "address[]",
        name: "_trustedForwarders",
    },
    {
        type: "address",
        name: "_primarySaleRecipient",
    },
    {
        type: "address",
        name: "_platformFeeRecipient",
    },
    {
        type: "uint256",
        name: "_platformFeeBps",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "initialize" function.
 * @param options - The options for the initialize function.
 * @returns The encoded ABI parameters.
 * @extension PREBUILTS
 * @example
 * ```
 * import { encodeInitializeParams } "thirdweb/extensions/prebuilts";
 * const result = encodeInitializeParams({
 *  defaultAdmin: ...,
 *  name: ...,
 *  symbol: ...,
 *  contractURI: ...,
 *  trustedForwarders: ...,
 *  primarySaleRecipient: ...,
 *  platformFeeRecipient: ...,
 *  platformFeeBps: ...,
 * });
 * ```
 */
export function encodeInitializeParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.defaultAdmin,
        options.name,
        options.symbol,
        options.contractURI,
        options.trustedForwarders,
        options.primarySaleRecipient,
        options.platformFeeRecipient,
        options.platformFeeBps,
    ]);
}
/**
 * Calls the "initialize" function on the contract.
 * @param options - The options for the "initialize" function.
 * @returns A prepared transaction object.
 * @extension PREBUILTS
 * @example
 * ```
 * import { initialize } from "thirdweb/extensions/prebuilts";
 *
 * const transaction = initialize({
 *  defaultAdmin: ...,
 *  name: ...,
 *  symbol: ...,
 *  contractURI: ...,
 *  trustedForwarders: ...,
 *  primarySaleRecipient: ...,
 *  platformFeeRecipient: ...,
 *  platformFeeBps: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function initialize(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.defaultAdmin,
                    resolvedParams.name,
                    resolvedParams.symbol,
                    resolvedParams.contractURI,
                    resolvedParams.trustedForwarders,
                    resolvedParams.primarySaleRecipient,
                    resolvedParams.platformFeeRecipient,
                    resolvedParams.platformFeeBps,
                ];
            }
            : [
                options.defaultAdmin,
                options.name,
                options.symbol,
                options.contractURI,
                options.trustedForwarders,
                options.primarySaleRecipient,
                options.platformFeeRecipient,
                options.platformFeeBps,
            ],
    });
}
