// thirdweb/dist/esm/extensions/erc721/__generated__/ILazyMint/write/lazyMint.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0xd37c353b";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "amount",
    },
    {
        type: "string",
        name: "baseURIForTokens",
    },
    {
        type: "bytes",
        name: "extraData",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "batchId",
    },
];
/**
 * Encodes the parameters for the "lazyMint" function.
 * @param options - The options for the lazyMint function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeLazyMintParams } "thirdweb/extensions/erc721";
 * const result = encodeLazyMintParams({
 *  amount: ...,
 *  baseURIForTokens: ...,
 *  extraData: ...,
 * });
 * ```
 */
export function encodeLazyMintParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.amount,
        options.baseURIForTokens,
        options.extraData,
    ]);
}
/**
 * Calls the "lazyMint" function on the contract.
 * @param options - The options for the "lazyMint" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```
 * import { lazyMint } from "thirdweb/extensions/erc721";
 *
 * const transaction = lazyMint({
 *  amount: ...,
 *  baseURIForTokens: ...,
 *  extraData: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function lazyMint(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.amount,
                    resolvedParams.baseURIForTokens,
                    resolvedParams.extraData,
                ];
            }
            : [options.amount, options.baseURIForTokens, options.extraData],
    });
}
