// thirdweb/dist/esm/extensions/thirdweb/__generated__/IThirdwebContract/read/contractVersion.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xa0a8e460";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint8",
    },
];
/**
 * Decodes the result of the contractVersion function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeContractVersionResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeContractVersionResult("...");
 * ```
 */
export function decodeContractVersionResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "contractVersion" function on the contract.
 * @param options - The options for the contractVersion function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { contractVersion } from "thirdweb/extensions/thirdweb";
 *
 * const result = await contractVersion();
 *
 * ```
 */
export async function contractVersion(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
