// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountFactory/read/getAllAccounts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllAccounts = exports.decodeGetAllAccountsResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x08e93d0a";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address[]",
    },
];
/**
 * Decodes the result of the getAllAccounts function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4337
 * @example
 * ```ts
 * import { decodeGetAllAccountsResult } from "thirdweb/extensions/erc4337";
 * const result = decodeGetAllAccountsResult("...");
 * ```
 */
function decodeGetAllAccountsResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetAllAccountsResult = decodeGetAllAccountsResult;
/**
 * Calls the "getAllAccounts" function on the contract.
 * @param options - The options for the getAllAccounts function.
 * @returns The parsed result of the function call.
 * @extension ERC4337
 * @example
 * ```ts
 * import { getAllAccounts } from "thirdweb/extensions/erc4337";
 *
 * const result = await getAllAccounts();
 *
 * ```
 */
async function getAllAccounts(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.getAllAccounts = getAllAccounts;
