// thirdweb/dist/cjs/extensions/common/__generated__/IRoyaltyPayments/write/setRoyaltyEngine.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setRoyaltyEngine = exports.encodeSetRoyaltyEngineParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x21ede032";
const FN_INPUTS = [
    {
        type: "address",
        name: "_royaltyEngineAddress",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setRoyaltyEngine" function.
 * @param options - The options for the setRoyaltyEngine function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeSetRoyaltyEngineParams } "thirdweb/extensions/common";
 * const result = encodeSetRoyaltyEngineParams({
 *  royaltyEngineAddress: ...,
 * });
 * ```
 */
function encodeSetRoyaltyEngineParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.royaltyEngineAddress]);
}
exports.encodeSetRoyaltyEngineParams = encodeSetRoyaltyEngineParams;
/**
 * Calls the "setRoyaltyEngine" function on the contract.
 * @param options - The options for the "setRoyaltyEngine" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```
 * import { setRoyaltyEngine } from "thirdweb/extensions/common";
 *
 * const transaction = setRoyaltyEngine({
 *  royaltyEngineAddress: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setRoyaltyEngine(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.royaltyEngineAddress];
            }
            : [options.royaltyEngineAddress],
    });
}
exports.setRoyaltyEngine = setRoyaltyEngine;
