// thirdweb/dist/cjs/extensions/common/__generated__/IMulticall/write/multicall.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.multicall = exports.encodeMulticallParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xac9650d8";
const FN_INPUTS = [
    {
        type: "bytes[]",
        name: "data",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes[]",
        name: "results",
    },
];
/**
 * Encodes the parameters for the "multicall" function.
 * @param options - The options for the multicall function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeMulticallParams } "thirdweb/extensions/common";
 * const result = encodeMulticallParams({
 *  data: ...,
 * });
 * ```
 */
function encodeMulticallParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.data]);
}
exports.encodeMulticallParams = encodeMulticallParams;
/**
 * Calls the "multicall" function on the contract.
 * @param options - The options for the "multicall" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```
 * import { multicall } from "thirdweb/extensions/common";
 *
 * const transaction = multicall({
 *  data: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function multicall(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.data];
            }
            : [options.data],
    });
}
exports.multicall = multicall;
