// thirdweb/dist/cjs/extensions/erc20/__generated__/IVotes/read/getVotes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getVotes = exports.decodeGetVotesResult = exports.encodeGetVotesParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x9ab24eb0";
const FN_INPUTS = [
    {
        type: "address",
        name: "account",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Encodes the parameters for the "getVotes" function.
 * @param options - The options for the getVotes function.
 * @returns The encoded ABI parameters.
 * @extension ERC20
 * @example
 * ```
 * import { encodeGetVotesParams } "thirdweb/extensions/erc20";
 * const result = encodeGetVotesParams({
 *  account: ...,
 * });
 * ```
 */
function encodeGetVotesParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.account]);
}
exports.encodeGetVotesParams = encodeGetVotesParams;
/**
 * Decodes the result of the getVotes function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC20
 * @example
 * ```
 * import { decodeGetVotesResult } from "thirdweb/extensions/erc20";
 * const result = decodeGetVotesResult("...");
 * ```
 */
function decodeGetVotesResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetVotesResult = decodeGetVotesResult;
/**
 * Calls the "getVotes" function on the contract.
 * @param options - The options for the getVotes function.
 * @returns The parsed result of the function call.
 * @extension ERC20
 * @example
 * ```
 * import { getVotes } from "thirdweb/extensions/erc20";
 *
 * const result = await getVotes({
 *  account: ...,
 * });
 *
 * ```
 */
async function getVotes(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.account],
    });
}
exports.getVotes = getVotes;
