// thirdweb/dist/cjs/extensions/erc1155/__generated__/IAirdropERC1155/write/airdropERC1155.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.airdropERC1155 = exports.encodeAirdropERC1155Params = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x41444690";
const FN_INPUTS = [
    {
        type: "address",
        name: "tokenAddress",
    },
    {
        type: "address",
        name: "tokenOwner",
    },
    {
        type: "tuple[]",
        name: "contents",
        components: [
            {
                type: "address",
                name: "recipient",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "amount",
            },
        ],
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "airdropERC1155" function.
 * @param options - The options for the airdropERC1155 function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeAirdropERC1155Params } "thirdweb/extensions/erc1155";
 * const result = encodeAirdropERC1155Params({
 *  tokenAddress: ...,
 *  tokenOwner: ...,
 *  contents: ...,
 * });
 * ```
 */
function encodeAirdropERC1155Params(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.tokenAddress,
        options.tokenOwner,
        options.contents,
    ]);
}
exports.encodeAirdropERC1155Params = encodeAirdropERC1155Params;
/**
 * Calls the "airdropERC1155" function on the contract.
 * @param options - The options for the "airdropERC1155" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```ts
 * import { airdropERC1155 } from "thirdweb/extensions/erc1155";
 *
 * const transaction = airdropERC1155({
 *  contract,
 *  tokenAddress: ...,
 *  tokenOwner: ...,
 *  contents: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function airdropERC1155(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.tokenAddress,
                    resolvedParams.tokenOwner,
                    resolvedParams.contents,
                ];
            }
            : [options.tokenAddress, options.tokenOwner, options.contents],
    });
}
exports.airdropERC1155 = airdropERC1155;
