// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/write/remove.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.remove = exports.encodeRemoveParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x58edef4c";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "key",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "remove" function.
 * @param options - The options for the remove function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeRemoveParams } "thirdweb/extensions/farcaster";
 * const result = encodeRemoveParams({
 *  key: ...,
 * });
 * ```
 */
function encodeRemoveParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.key]);
}
exports.encodeRemoveParams = encodeRemoveParams;
/**
 * Calls the "remove" function on the contract.
 * @param options - The options for the "remove" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```
 * import { remove } from "thirdweb/extensions/farcaster";
 *
 * const transaction = remove({
 *  key: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function remove(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.key];
            }
            : [options.key],
    });
}
exports.remove = remove;
