// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/write/removeFor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeFor = exports.encodeRemoveForParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x787bd966";
const FN_INPUTS = [
    {
        type: "address",
        name: "fidOwner",
    },
    {
        type: "bytes",
        name: "key",
    },
    {
        type: "uint256",
        name: "deadline",
    },
    {
        type: "bytes",
        name: "sig",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "removeFor" function.
 * @param options - The options for the removeFor function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeRemoveForParams } "thirdweb/extensions/farcaster";
 * const result = encodeRemoveForParams({
 *  fidOwner: ...,
 *  key: ...,
 *  deadline: ...,
 *  sig: ...,
 * });
 * ```
 */
function encodeRemoveForParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.fidOwner,
        options.key,
        options.deadline,
        options.sig,
    ]);
}
exports.encodeRemoveForParams = encodeRemoveForParams;
/**
 * Calls the "removeFor" function on the contract.
 * @param options - The options for the "removeFor" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```
 * import { removeFor } from "thirdweb/extensions/farcaster";
 *
 * const transaction = removeFor({
 *  fidOwner: ...,
 *  key: ...,
 *  deadline: ...,
 *  sig: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function removeFor(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.fidOwner,
                    resolvedParams.key,
                    resolvedParams.deadline,
                    resolvedParams.sig,
                ];
            }
            : [options.fidOwner, options.key, options.deadline, options.sig],
    });
}
exports.removeFor = removeFor;
