// thirdweb/dist/esm/react/web/wallets/embedded/EmbeddedWalletFormUI.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import styled from "@emotion/styled";
import { useContext } from "react";
import { ModalConfigCtx, SetModalConfigCtx, } from "../../providers/wallet-ui-states-provider.js";
import { TOS } from "../../ui/ConnectWallet/Modal/TOS.js";
import { useScreenContext } from "../../ui/ConnectWallet/Modal/screen.js";
import { PoweredByThirdweb } from "../../ui/ConnectWallet/PoweredByTW.js";
import { Img } from "../../ui/components/Img.js";
import { Spacer } from "../../ui/components/Spacer.js";
import { TextDivider } from "../../ui/components/TextDivider.js";
import { Container, ModalHeader } from "../../ui/components/basic.js";
import { Button } from "../../ui/components/buttons.js";
import { useCustomTheme } from "../../ui/design-system/CustomThemeProvider.js";
import { iconSize, spacing, fontSize } from "../../ui/design-system/index.js";
import { socialIcons } from "./socialIcons.js";
import { openOauthSignInWindow } from "./openOauthSignInWindow.js";
import { useWalletConnectionCtx } from "../../../core/hooks/others/useWalletConnectionCtx.js";
import { InputSelectionUI } from "./InputSelectionUI.js";
const defaultAuthOptions = [
    "email",
    "google",
    "apple",
    "facebook",
];
/**
 * @internal
 */
export const EmbeddedWalletFormUI = (props) => {
    const locale = props.locale;
    const { chain, client } = useWalletConnectionCtx();
    const { done, wallet } = props;
    const { modalSize } = useContext(ModalConfigCtx);
    const setModalConfig = useContext(SetModalConfigCtx);
    function saveState(data) {
        setModalConfig((p) => ({
            ...p,
            data,
        }));
    }
    const themeObj = useCustomTheme();
    const loginMethodsLabel = {
        google: locale.signInWithGoogle,
        facebook: locale.signInWithFacebook,
        apple: locale.signInWithApple,
    };
    const config = props.wallet.getConfig();
    const authOptions = config?.auth?.options || defaultAuthOptions;
    const enableEmailLogin = authOptions.includes("email");
    const socialLogins = authOptions.filter((x) => x !== "email");
    const hasSocialLogins = socialLogins.length > 0;
    // Need to trigger login on button click to avoid popup from being blocked
    const handleSocialLogin = async (strategy) => {
        try {
            const socialLoginWindow = openOauthSignInWindow(strategy, themeObj);
            if (!socialLoginWindow) {
                throw new Error("Failed to open login window");
            }
            const connectPromise = wallet.connect({
                chain,
                client,
                strategy: strategy,
                openedWindow: socialLoginWindow,
                closeOpenedWindow: (openedWindow) => {
                    openedWindow.close();
                },
            });
            saveState({
                socialLogin: {
                    type: strategy,
                    connectionPromise: connectPromise,
                },
            });
            props.select();
            await connectPromise;
            done();
        }
        catch (e) {
            console.error(`Error sign in with ${strategy}`, e);
        }
    };
    const showOnlyIcons = socialLogins.length > 1;
    return (_jsxs(Container, { flex: "column", gap: "lg", children: [hasSocialLogins && (_jsx(Container, { flex: showOnlyIcons ? "row" : "column", center: "x", gap: "sm", style: {
                    justifyContent: "space-between",
                }, children: socialLogins.map((loginMethod) => {
                    const imgIconSize = showOnlyIcons ? iconSize.lg : iconSize.md;
                    return (_jsxs(SocialButton, { "aria-label": `Login with ${loginMethod}`, "data-variant": showOnlyIcons ? "icon" : "full", variant: showOnlyIcons ? "outline" : "secondary", fullWidth: !showOnlyIcons, onClick: () => {
                            handleSocialLogin(loginMethod);
                        }, children: [_jsx(Img, { src: socialIcons[loginMethod], width: imgIconSize, height: imgIconSize }), !showOnlyIcons && loginMethodsLabel[loginMethod]] }, loginMethod));
                }) })), modalSize === "wide" && hasSocialLogins && enableEmailLogin && (_jsx(TextDivider, { text: locale.or })), enableEmailLogin && (_jsx(InputSelectionUI, { onSelect: (email) => {
                    saveState({
                        emailLogin: email,
                    });
                    props.select();
                }, placeholder: locale.emailPlaceholder, name: "email", type: "email", errorMessage: (_input) => {
                    const input = _input.replace(/\+/g, "").toLowerCase();
                    const emailRegex = /^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,})$/g;
                    const isValidEmail = emailRegex.test(input);
                    if (!isValidEmail) {
                        return locale.invalidEmail;
                    }
                    return undefined;
                }, emptyErrorMessage: locale.emailRequired, submitButtonText: locale.submitEmail }))] }));
};
/**
 * @internal
 */
export function EmbeddedWalletFormUIScreen(props) {
    const locale = props.locale.emailLoginScreen;
    const modalConfig = useContext(ModalConfigCtx);
    const isCompact = modalConfig.modalSize === "compact";
    const { initialScreen, screen } = useScreenContext();
    return (_jsxs(Container, { fullHeight: true, flex: "column", p: "lg", animate: "fadein", style: {
            minHeight: "250px",
        }, children: [_jsx(ModalHeader, { onBack: screen === props.wallet && initialScreen === props.wallet
                    ? undefined
                    : props.goBack, title: locale.title }), isCompact ? _jsx(Spacer, { y: "xl" }) : null, _jsx(Container, { expand: true, flex: "column", center: "y", p: isCompact ? undefined : "lg", children: _jsx(EmbeddedWalletFormUI, { ...props }) }), isCompact &&
                (modalConfig.showThirdwebBranding !== false ||
                    modalConfig.termsOfServiceUrl ||
                    modalConfig.privacyPolicyUrl) && _jsx(Spacer, { y: "xl" }), _jsxs(Container, { flex: "column", gap: "lg", children: [_jsx(TOS, { termsOfServiceUrl: modalConfig.termsOfServiceUrl, privacyPolicyUrl: modalConfig.privacyPolicyUrl }), modalConfig.showThirdwebBranding !== false && _jsx(PoweredByThirdweb, {})] })] }));
}
const SocialButton = /* @__PURE__ */ styled(Button)({
    "&[data-variant='full']": {
        display: "flex",
        justifyContent: "center",
        gap: spacing.md,
        fontSize: fontSize.md,
        transition: "background-color 0.2s ease",
        "&:active": {
            boxShadow: "none",
        },
    },
    "&[data-variant='icon']": {
        padding: spacing.sm,
        flexGrow: 1,
    },
});
