// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IRulesEngine/write/setRulesEngineOverride.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setRulesEngineOverride = exports.encodeSetRulesEngineOverrideParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x0eb0adb6";
const FN_INPUTS = [
    {
        type: "address",
        name: "_rulesEngineAddress",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setRulesEngineOverride" function.
 * @param options - The options for the setRulesEngineOverride function.
 * @returns The encoded ABI parameters.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { encodeSetRulesEngineOverrideParams } "thirdweb/extensions/thirdweb";
 * const result = encodeSetRulesEngineOverrideParams({
 *  rulesEngineAddress: ...,
 * });
 * ```
 */
function encodeSetRulesEngineOverrideParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.rulesEngineAddress]);
}
exports.encodeSetRulesEngineOverrideParams = encodeSetRulesEngineOverrideParams;
/**
 * Calls the "setRulesEngineOverride" function on the contract.
 * @param options - The options for the "setRulesEngineOverride" function.
 * @returns A prepared transaction object.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { setRulesEngineOverride } from "thirdweb/extensions/thirdweb";
 *
 * const transaction = setRulesEngineOverride({
 *  contract,
 *  rulesEngineAddress: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setRulesEngineOverride(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.rulesEngineAddress];
            }
            : [options.rulesEngineAddress],
    });
}
exports.setRulesEngineOverride = setRulesEngineOverride;
