// thirdweb/dist/esm/extensions/uniswap/__generated__/ISwapRouter/write/exactOutput.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0xf28c0498";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "params",
        components: [
            {
                type: "bytes",
                name: "path",
            },
            {
                type: "address",
                name: "recipient",
            },
            {
                type: "uint256",
                name: "deadline",
            },
            {
                type: "uint256",
                name: "amountOut",
            },
            {
                type: "uint256",
                name: "amountInMaximum",
            },
        ],
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "amountIn",
    },
];
/**
 * Encodes the parameters for the "exactOutput" function.
 * @param options - The options for the exactOutput function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeExactOutputParams } "thirdweb/extensions/uniswap";
 * const result = encodeExactOutputParams({
 *  params: ...,
 * });
 * ```
 */
export function encodeExactOutputParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.params]);
}
/**
 * Calls the "exactOutput" function on the contract.
 * @param options - The options for the "exactOutput" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```
 * import { exactOutput } from "thirdweb/extensions/uniswap";
 *
 * const transaction = exactOutput({
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function exactOutput(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
