// thirdweb/dist/esm/extensions/common/__generated__/IOwnable/read/owner.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x8da5cb5b";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Decodes the result of the owner function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```
 * import { decodeOwnerResult } from "thirdweb/extensions/common";
 * const result = decodeOwnerResult("...");
 * ```
 */
export function decodeOwnerResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "owner" function on the contract.
 * @param options - The options for the owner function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```
 * import { owner } from "thirdweb/extensions/common";
 *
 * const result = await owner();
 *
 * ```
 */
export async function owner(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
