// thirdweb/dist/cjs/react/core/providers/thirdweb-provider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdwebProvider = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_query_1 = require("@tanstack/react-query");
const react_1 = require("react");
const type_guards_js_1 = require("../../../utils/type-guards.js");
const wait_for_tx_receipt_js_1 = require("../../../transaction/actions/wait-for-tx-receipt.js");
const types_js_1 = require("../../../transaction/types.js");
/**
 * The ThirdwebProvider is component is a provider component that sets up the React Query client.
 * @param props - The props for the ThirdwebProvider
 * @example
 * ```jsx
 * import { ThirdwebProvider } from "thirdweb/react";
 *
 * function Example() {
 *  return (
 *    <ThirdwebProvider>
 *      <App />
 *    </ThirdwebProvider>
 *   )
 * }
 * ```
 * @component
 */
function ThirdwebProvider(props) {
    const [queryClient] = (0, react_1.useState)(() => new react_query_1.QueryClient({
        defaultOptions: {
            mutations: {
                onSettled: (data, error, variables) => {
                    if (error) {
                        // TODO: remove - but useful for debug now
                        console.error("[Mutation Error]", error);
                    }
                    if ((0, types_js_1.isBaseTransactionOptions)(variables)) {
                        if ((0, type_guards_js_1.isObjectWithKeys)(data, ["transactionHash", "client", "chain"])) {
                            (0, wait_for_tx_receipt_js_1.waitForReceipt)(data)
                                .catch((e) => {
                                // swallow errors for receipts, but log
                                console.error("[Transaction Error]", e);
                            })
                                .then(() => {
                                return queryClient.invalidateQueries({
                                    queryKey: [
                                        // invalidate any readContract queries for this chainId:contractAddress
                                        [
                                            "readContract",
                                            variables.contract.chain.id,
                                            variables.contract.address,
                                        ],
                                        // invalidate any walletBalance queries for this chainId
                                        // TODO: add wallet address in here if we can get it somehow
                                        [
                                            "walletBalance",
                                            variables.contract.chain.id,
                                        ],
                                    ],
                                });
                            });
                        }
                    }
                },
            },
            queries: {
                // With SSR, we usually want to set some default staleTime
                // above 0 to avoid refetching immediately on the client
                staleTime: 60 * 1000,
            },
        },
    }));
    return ((0, jsx_runtime_1.jsx)(react_query_1.QueryClientProvider, { client: queryClient, children: props.children }));
}
exports.ThirdwebProvider = ThirdwebProvider;
