// thirdweb/dist/cjs/extensions/common/__generated__/IPlatformFee/read/getPlatformFeeInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPlatformFeeInfo = exports.decodeGetPlatformFeeInfoResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xd45573f6";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
    },
    {
        type: "uint16",
    },
];
/**
 * Decodes the result of the getPlatformFeeInfo function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```ts
 * import { decodeGetPlatformFeeInfoResult } from "thirdweb/extensions/common";
 * const result = decodeGetPlatformFeeInfoResult("...");
 * ```
 */
function decodeGetPlatformFeeInfoResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result);
}
exports.decodeGetPlatformFeeInfoResult = decodeGetPlatformFeeInfoResult;
/**
 * Calls the "getPlatformFeeInfo" function on the contract.
 * @param options - The options for the getPlatformFeeInfo function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```ts
 * import { getPlatformFeeInfo } from "thirdweb/extensions/common";
 *
 * const result = await getPlatformFeeInfo();
 *
 * ```
 */
async function getPlatformFeeInfo(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.getPlatformFeeInfo = getPlatformFeeInfo;
