// thirdweb/dist/cjs/extensions/erc721/__generated__/ISignatureDropERC721/read/totalMinted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.totalMinted = exports.decodeTotalMintedResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xa2309ff8";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the totalMinted function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeTotalMintedResult } from "thirdweb/extensions/erc721";
 * const result = decodeTotalMintedResult("...");
 * ```
 */
function decodeTotalMintedResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTotalMintedResult = decodeTotalMintedResult;
/**
 * Calls the "totalMinted" function on the contract.
 * @param options - The options for the totalMinted function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { totalMinted } from "thirdweb/extensions/erc721";
 *
 * const result = await totalMinted();
 *
 * ```
 */
async function totalMinted(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.totalMinted = totalMinted;
