// thirdweb/dist/cjs/react/web/ui/ConnectWallet/screens/Buy/swap/SwapFees.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SwapFees = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const formatNumber_js_1 = require("../../../../../../../utils/formatNumber.js");
const Spacer_js_1 = require("../../../../components/Spacer.js");
const basic_js_1 = require("../../../../components/basic.js");
const text_js_1 = require("../../../../components/text.js");
const CustomThemeProvider_js_1 = require("../../../../design-system/CustomThemeProvider.js");
const elements_js_1 = require("../../../../design-system/elements.js");
const index_js_1 = require("../../../../design-system/index.js");
/**
 * @internal
 */
function SwapFees(props) {
    return ((0, jsx_runtime_1.jsx)(basic_js_1.Container, { animate: "fadein", children: (0, jsx_runtime_1.jsxs)(BorderContainer, { children: [(0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", style: {
                        justifyContent: "space-between",
                    }, children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { color: "primaryText", size: "xs", children: "Processing Fees" }), (0, jsx_runtime_1.jsx)(basic_js_1.Container, { flex: "column", gap: "xxs", style: {
                                alignItems: "flex-end",
                            }, children: props.quote.processingFees.map((fee) => {
                                const feeAmount = (0, formatNumber_js_1.formatNumber)(Number(fee.amount), 4);
                                return ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", gap: "xxs", children: [(0, jsx_runtime_1.jsxs)(text_js_1.Text, { color: "primaryText", size: "xs", children: [feeAmount === 0 ? "~" : "", feeAmount, " ", fee.token.symbol] }), (0, jsx_runtime_1.jsxs)(text_js_1.Text, { color: "secondaryText", size: "xs", children: ["($", fee.amountUSDCents / 100, ")"] })] }, fee.token.symbol));
                            }) })] }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "sm" }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", center: "y", style: {
                        justifyContent: "space-between",
                    }, children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "xs", color: "primaryText", children: "Estimated Duration" }), (0, jsx_runtime_1.jsxs)(text_js_1.Text, { size: "xs", color: "primaryText", children: [props.quote.swapDetails.estimated.durationSeconds, "s"] })] })] }) }));
}
exports.SwapFees = SwapFees;
const BorderContainer = /* @__PURE__ */ (0, elements_js_1.StyledDiv)(() => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        padding: index_js_1.spacing.sm,
        border: `1px solid ${theme.colors.borderColor}`,
        borderRadius: index_js_1.radius.md,
    };
});
