// thirdweb/dist/cjs/react/web/ui/ConnectWallet/screens/Buy/swap/useSwapSupportedChains.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSwapSupportedChains = exports.fetchSwapSupportedChains = void 0;
const react_query_1 = require("@tanstack/react-query");
const utils_js_1 = require("../../../../../../../chains/utils.js");
const definitions_js_1 = require("../../../../../../../pay/buyWithCrypto/utils/definitions.js");
const fetch_js_1 = require("../../../../../../../utils/fetch.js");
const withCache_js_1 = require("../../../../../../../utils/promise/withCache.js");
async function fetchSwapSupportedChains(client) {
    return (0, withCache_js_1.withCache)(async () => {
        const fetchWithHeaders = (0, fetch_js_1.getClientFetch)(client);
        const res = await fetchWithHeaders((0, definitions_js_1.getPayChainsEndpoint)());
        const data = await res.json();
        const chainIds = data.result.chainIds;
        return chainIds.map(utils_js_1.defineChain);
    }, {
        cacheKey: "swapSupportedChains",
        cacheTime: 5 * 60 * 1000,
    });
}
exports.fetchSwapSupportedChains = fetchSwapSupportedChains;
/**
 * @internal
 */
function useSwapSupportedChains(client) {
    return (0, react_query_1.useQuery)({
        queryKey: ["swapSupportedChains", client],
        queryFn: async () => {
            return fetchSwapSupportedChains(client);
        },
    });
}
exports.useSwapSupportedChains = useSwapSupportedChains;
