// thirdweb/dist/esm/extensions/marketplace/__generated__/IOffers/read/getAllOffers.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xc1edcfbe";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_startId",
    },
    {
        type: "uint256",
        name: "_endId",
    },
];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "offers",
        components: [
            {
                type: "uint256",
                name: "offerId",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "quantity",
            },
            {
                type: "uint256",
                name: "totalPrice",
            },
            {
                type: "uint256",
                name: "expirationTimestamp",
            },
            {
                type: "address",
                name: "offeror",
            },
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "uint8",
                name: "tokenType",
            },
            {
                type: "uint8",
                name: "status",
            },
        ],
    },
];
/**
 * Encodes the parameters for the "getAllOffers" function.
 * @param options - The options for the getAllOffers function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { encodeGetAllOffersParams } "thirdweb/extensions/marketplace";
 * const result = encodeGetAllOffersParams({
 *  startId: ...,
 *  endId: ...,
 * });
 * ```
 */
export function encodeGetAllOffersParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.startId, options.endId]);
}
/**
 * Decodes the result of the getAllOffers function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { decodeGetAllOffersResult } from "thirdweb/extensions/marketplace";
 * const result = decodeGetAllOffersResult("...");
 * ```
 */
export function decodeGetAllOffersResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "getAllOffers" function on the contract.
 * @param options - The options for the getAllOffers function.
 * @returns The parsed result of the function call.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { getAllOffers } from "thirdweb/extensions/marketplace";
 *
 * const result = await getAllOffers({
 *  startId: ...,
 *  endId: ...,
 * });
 *
 * ```
 */
export async function getAllOffers(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.startId, options.endId],
    });
}
