// thirdweb/dist/esm/react/web/ui/ConnectWallet/screens/Buy/swap/SwapFees.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { formatNumber } from "../../../../../../../utils/formatNumber.js";
import { Spacer } from "../../../../components/Spacer.js";
import { Container } from "../../../../components/basic.js";
import { Text } from "../../../../components/text.js";
import { useCustomTheme } from "../../../../design-system/CustomThemeProvider.js";
import { StyledDiv } from "../../../../design-system/elements.js";
import { radius, spacing } from "../../../../design-system/index.js";
/**
 * @internal
 */
export function SwapFees(props) {
    return (_jsx(Container, { animate: "fadein", children: _jsxs(BorderContainer, { children: [_jsxs(Container, { flex: "row", style: {
                        justifyContent: "space-between",
                    }, children: [_jsx(Text, { color: "primaryText", size: "xs", children: "Processing Fees" }), _jsx(Container, { flex: "column", gap: "xxs", style: {
                                alignItems: "flex-end",
                            }, children: props.quote.processingFees.map((fee) => {
                                const feeAmount = formatNumber(Number(fee.amount), 4);
                                return (_jsxs(Container, { flex: "row", gap: "xxs", children: [_jsxs(Text, { color: "primaryText", size: "xs", children: [feeAmount === 0 ? "~" : "", feeAmount, " ", fee.token.symbol] }), _jsxs(Text, { color: "secondaryText", size: "xs", children: ["($", fee.amountUSDCents / 100, ")"] })] }, fee.token.symbol));
                            }) })] }), _jsx(Spacer, { y: "sm" }), _jsxs(Container, { flex: "row", center: "y", style: {
                        justifyContent: "space-between",
                    }, children: [_jsx(Text, { size: "xs", color: "primaryText", children: "Estimated Duration" }), _jsxs(Text, { size: "xs", color: "primaryText", children: [props.quote.swapDetails.estimated.durationSeconds, "s"] })] })] }) }));
}
const BorderContainer = /* @__PURE__ */ StyledDiv(() => {
    const theme = useCustomTheme();
    return {
        padding: spacing.sm,
        border: `1px solid ${theme.colors.borderColor}`,
        borderRadius: radius.md,
    };
});
