// thirdweb/dist/cjs/extensions/erc20/__generated__/IERC20Permit/write/permit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.permit = exports.encodePermitParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xd505accf";
const FN_INPUTS = [
    {
        type: "address",
        name: "owner",
    },
    {
        type: "address",
        name: "spender",
    },
    {
        type: "uint256",
        name: "value",
    },
    {
        type: "uint256",
        name: "deadline",
    },
    {
        type: "uint8",
        name: "v",
    },
    {
        type: "bytes32",
        name: "r",
    },
    {
        type: "bytes32",
        name: "s",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "permit" function.
 * @param options - The options for the permit function.
 * @returns The encoded ABI parameters.
 * @extension ERC20
 * @example
 * ```
 * import { encodePermitParams } "thirdweb/extensions/erc20";
 * const result = encodePermitParams({
 *  owner: ...,
 *  spender: ...,
 *  value: ...,
 *  deadline: ...,
 *  v: ...,
 *  r: ...,
 *  s: ...,
 * });
 * ```
 */
function encodePermitParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.owner,
        options.spender,
        options.value,
        options.deadline,
        options.v,
        options.r,
        options.s,
    ]);
}
exports.encodePermitParams = encodePermitParams;
/**
 * Calls the "permit" function on the contract.
 * @param options - The options for the "permit" function.
 * @returns A prepared transaction object.
 * @extension ERC20
 * @example
 * ```
 * import { permit } from "thirdweb/extensions/erc20";
 *
 * const transaction = permit({
 *  owner: ...,
 *  spender: ...,
 *  value: ...,
 *  deadline: ...,
 *  v: ...,
 *  r: ...,
 *  s: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function permit(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.owner,
                    resolvedParams.spender,
                    resolvedParams.value,
                    resolvedParams.deadline,
                    resolvedParams.v,
                    resolvedParams.r,
                    resolvedParams.s,
                ];
            }
            : [
                options.owner,
                options.spender,
                options.value,
                options.deadline,
                options.v,
                options.r,
                options.s,
            ],
    });
}
exports.permit = permit;
