// thirdweb/dist/cjs/extensions/marketplace/__generated__/IEnglishAuctions/read/getAllAuctions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllAuctions = exports.decodeGetAllAuctionsResult = exports.encodeGetAllAuctionsParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xc291537c";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_startId",
    },
    {
        type: "uint256",
        name: "_endId",
    },
];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "auctions",
        components: [
            {
                type: "uint256",
                name: "auctionId",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "quantity",
            },
            {
                type: "uint256",
                name: "minimumBidAmount",
            },
            {
                type: "uint256",
                name: "buyoutBidAmount",
            },
            {
                type: "uint64",
                name: "timeBufferInSeconds",
            },
            {
                type: "uint64",
                name: "bidBufferBps",
            },
            {
                type: "uint64",
                name: "startTimestamp",
            },
            {
                type: "uint64",
                name: "endTimestamp",
            },
            {
                type: "address",
                name: "auctionCreator",
            },
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "uint8",
                name: "tokenType",
            },
            {
                type: "uint8",
                name: "status",
            },
        ],
    },
];
/**
 * Encodes the parameters for the "getAllAuctions" function.
 * @param options - The options for the getAllAuctions function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { encodeGetAllAuctionsParams } "thirdweb/extensions/marketplace";
 * const result = encodeGetAllAuctionsParams({
 *  startId: ...,
 *  endId: ...,
 * });
 * ```
 */
function encodeGetAllAuctionsParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.startId, options.endId]);
}
exports.encodeGetAllAuctionsParams = encodeGetAllAuctionsParams;
/**
 * Decodes the result of the getAllAuctions function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { decodeGetAllAuctionsResult } from "thirdweb/extensions/marketplace";
 * const result = decodeGetAllAuctionsResult("...");
 * ```
 */
function decodeGetAllAuctionsResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetAllAuctionsResult = decodeGetAllAuctionsResult;
/**
 * Calls the "getAllAuctions" function on the contract.
 * @param options - The options for the getAllAuctions function.
 * @returns The parsed result of the function call.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { getAllAuctions } from "thirdweb/extensions/marketplace";
 *
 * const result = await getAllAuctions({
 *  startId: ...,
 *  endId: ...,
 * });
 *
 * ```
 */
async function getAllAuctions(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.startId, options.endId],
    });
}
exports.getAllAuctions = getAllAuctions;
