// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountPermissions/read/getAllSigners.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllSigners = exports.decodeGetAllSignersResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xd42f2f35";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "signers",
        components: [
            {
                type: "address",
                name: "signer",
            },
            {
                type: "address[]",
                name: "approvedTargets",
            },
            {
                type: "uint256",
                name: "nativeTokenLimitPerTransaction",
            },
            {
                type: "uint128",
                name: "startTimestamp",
            },
            {
                type: "uint128",
                name: "endTimestamp",
            },
        ],
    },
];
/**
 * Decodes the result of the getAllSigners function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4337
 * @example
 * ```
 * import { decodeGetAllSignersResult } from "thirdweb/extensions/erc4337";
 * const result = decodeGetAllSignersResult("...");
 * ```
 */
function decodeGetAllSignersResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetAllSignersResult = decodeGetAllSignersResult;
/**
 * Calls the "getAllSigners" function on the contract.
 * @param options - The options for the getAllSigners function.
 * @returns The parsed result of the function call.
 * @extension ERC4337
 * @example
 * ```
 * import { getAllSigners } from "thirdweb/extensions/erc4337";
 *
 * const result = await getAllSigners();
 *
 * ```
 */
async function getAllSigners(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.getAllSigners = getAllSigners;
