// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountPermissions/read/getPermissionsForSigner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPermissionsForSigner = exports.decodeGetPermissionsForSignerResult = exports.encodeGetPermissionsForSignerParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xf15d424e";
const FN_INPUTS = [
    {
        type: "address",
        name: "signer",
    },
];
const FN_OUTPUTS = [
    {
        type: "tuple",
        name: "permissions",
        components: [
            {
                type: "address",
                name: "signer",
            },
            {
                type: "address[]",
                name: "approvedTargets",
            },
            {
                type: "uint256",
                name: "nativeTokenLimitPerTransaction",
            },
            {
                type: "uint128",
                name: "startTimestamp",
            },
            {
                type: "uint128",
                name: "endTimestamp",
            },
        ],
    },
];
/**
 * Encodes the parameters for the "getPermissionsForSigner" function.
 * @param options - The options for the getPermissionsForSigner function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```ts
 * import { encodeGetPermissionsForSignerParams } "thirdweb/extensions/erc4337";
 * const result = encodeGetPermissionsForSignerParams({
 *  signer: ...,
 * });
 * ```
 */
function encodeGetPermissionsForSignerParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.signer]);
}
exports.encodeGetPermissionsForSignerParams = encodeGetPermissionsForSignerParams;
/**
 * Decodes the result of the getPermissionsForSigner function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4337
 * @example
 * ```ts
 * import { decodeGetPermissionsForSignerResult } from "thirdweb/extensions/erc4337";
 * const result = decodeGetPermissionsForSignerResult("...");
 * ```
 */
function decodeGetPermissionsForSignerResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetPermissionsForSignerResult = decodeGetPermissionsForSignerResult;
/**
 * Calls the "getPermissionsForSigner" function on the contract.
 * @param options - The options for the getPermissionsForSigner function.
 * @returns The parsed result of the function call.
 * @extension ERC4337
 * @example
 * ```ts
 * import { getPermissionsForSigner } from "thirdweb/extensions/erc4337";
 *
 * const result = await getPermissionsForSigner({
 *  signer: ...,
 * });
 *
 * ```
 */
async function getPermissionsForSigner(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.signer],
    });
}
exports.getPermissionsForSigner = getPermissionsForSigner;
