// thirdweb/dist/cjs/extensions/uniswap/__generated__/IUniswapV3Factory/write/createPool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPool = exports.encodeCreatePoolParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa1671295";
const FN_INPUTS = [
    {
        type: "address",
        name: "tokenA",
    },
    {
        type: "address",
        name: "tokenB",
    },
    {
        type: "uint24",
        name: "fee",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "pool",
    },
];
/**
 * Encodes the parameters for the "createPool" function.
 * @param options - The options for the createPool function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { encodeCreatePoolParams } "thirdweb/extensions/uniswap";
 * const result = encodeCreatePoolParams({
 *  tokenA: ...,
 *  tokenB: ...,
 *  fee: ...,
 * });
 * ```
 */
function encodeCreatePoolParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.tokenA,
        options.tokenB,
        options.fee,
    ]);
}
exports.encodeCreatePoolParams = encodeCreatePoolParams;
/**
 * Calls the "createPool" function on the contract.
 * @param options - The options for the "createPool" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { createPool } from "thirdweb/extensions/uniswap";
 *
 * const transaction = createPool({
 *  contract,
 *  tokenA: ...,
 *  tokenB: ...,
 *  fee: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function createPool(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.tokenA,
                    resolvedParams.tokenB,
                    resolvedParams.fee,
                ];
            }
            : [options.tokenA, options.tokenB, options.fee],
    });
}
exports.createPool = createPool;
