// thirdweb/dist/cjs/extensions/uniswap/__generated__/IUniswapV3Factory/write/enableFeeAmount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.enableFeeAmount = exports.encodeEnableFeeAmountParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x8a7c195f";
const FN_INPUTS = [
    {
        type: "uint24",
        name: "fee",
    },
    {
        type: "int24",
        name: "tickSpacing",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "enableFeeAmount" function.
 * @param options - The options for the enableFeeAmount function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeEnableFeeAmountParams } "thirdweb/extensions/uniswap";
 * const result = encodeEnableFeeAmountParams({
 *  fee: ...,
 *  tickSpacing: ...,
 * });
 * ```
 */
function encodeEnableFeeAmountParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.fee, options.tickSpacing]);
}
exports.encodeEnableFeeAmountParams = encodeEnableFeeAmountParams;
/**
 * Calls the "enableFeeAmount" function on the contract.
 * @param options - The options for the "enableFeeAmount" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```
 * import { enableFeeAmount } from "thirdweb/extensions/uniswap";
 *
 * const transaction = enableFeeAmount({
 *  fee: ...,
 *  tickSpacing: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function enableFeeAmount(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.fee, resolvedParams.tickSpacing];
            }
            : [options.fee, options.tickSpacing],
    });
}
exports.enableFeeAmount = enableFeeAmount;
