// thirdweb/dist/cjs/extensions/uniswap/__generated__/IUniswapV3Factory/write/setOwner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setOwner = exports.encodeSetOwnerParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x13af4035";
const FN_INPUTS = [
    {
        type: "address",
        name: "newOwner",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setOwner" function.
 * @param options - The options for the setOwner function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { encodeSetOwnerParams } "thirdweb/extensions/uniswap";
 * const result = encodeSetOwnerParams({
 *  newOwner: ...,
 * });
 * ```
 */
function encodeSetOwnerParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.newOwner]);
}
exports.encodeSetOwnerParams = encodeSetOwnerParams;
/**
 * Calls the "setOwner" function on the contract.
 * @param options - The options for the "setOwner" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { setOwner } from "thirdweb/extensions/uniswap";
 *
 * const transaction = setOwner({
 *  contract,
 *  newOwner: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setOwner(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.newOwner];
            }
            : [options.newOwner],
    });
}
exports.setOwner = setOwner;
