// thirdweb/dist/cjs/extensions/erc20/__generated__/IERC20Permit/read/DOMAIN_SEPARATOR.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DOMAIN_SEPARATOR = exports.decodeDOMAIN_SEPARATORResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x3644e515";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "bytes32",
    },
];
/**
 * Decodes the result of the DOMAIN_SEPARATOR function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC20
 * @example
 * ```ts
 * import { decodeDOMAIN_SEPARATORResult } from "thirdweb/extensions/erc20";
 * const result = decodeDOMAIN_SEPARATORResult("...");
 * ```
 */
function decodeDOMAIN_SEPARATORResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeDOMAIN_SEPARATORResult = decodeDOMAIN_SEPARATORResult;
/**
 * Calls the "DOMAIN_SEPARATOR" function on the contract.
 * @param options - The options for the DOMAIN_SEPARATOR function.
 * @returns The parsed result of the function call.
 * @extension ERC20
 * @example
 * ```ts
 * import { DOMAIN_SEPARATOR } from "thirdweb/extensions/erc20";
 *
 * const result = await DOMAIN_SEPARATOR();
 *
 * ```
 */
async function DOMAIN_SEPARATOR(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.DOMAIN_SEPARATOR = DOMAIN_SEPARATOR;
