// thirdweb/dist/cjs/extensions/erc721/__generated__/IDelayedReveal/read/encryptDecrypt.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.encryptDecrypt = exports.decodeEncryptDecryptResult = exports.encodeEncryptDecryptParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xe7150322";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "data",
    },
    {
        type: "bytes",
        name: "key",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes",
        name: "result",
    },
];
/**
 * Encodes the parameters for the "encryptDecrypt" function.
 * @param options - The options for the encryptDecrypt function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeEncryptDecryptParams } "thirdweb/extensions/erc721";
 * const result = encodeEncryptDecryptParams({
 *  data: ...,
 *  key: ...,
 * });
 * ```
 */
function encodeEncryptDecryptParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.data, options.key]);
}
exports.encodeEncryptDecryptParams = encodeEncryptDecryptParams;
/**
 * Decodes the result of the encryptDecrypt function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeEncryptDecryptResult } from "thirdweb/extensions/erc721";
 * const result = decodeEncryptDecryptResult("...");
 * ```
 */
function decodeEncryptDecryptResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeEncryptDecryptResult = decodeEncryptDecryptResult;
/**
 * Calls the "encryptDecrypt" function on the contract.
 * @param options - The options for the encryptDecrypt function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { encryptDecrypt } from "thirdweb/extensions/erc721";
 *
 * const result = await encryptDecrypt({
 *  data: ...,
 *  key: ...,
 * });
 *
 * ```
 */
async function encryptDecrypt(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.data, options.key],
    });
}
exports.encryptDecrypt = encryptDecrypt;
