// thirdweb/dist/cjs/extensions/erc1155/__generated__/IERC1155/read/isApprovedForAll.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isApprovedForAll = exports.decodeIsApprovedForAllResult = exports.encodeIsApprovedForAllParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xe985e9c5";
const FN_INPUTS = [
    {
        type: "address",
        name: "_owner",
    },
    {
        type: "address",
        name: "_operator",
    },
];
const FN_OUTPUTS = [
    {
        type: "bool",
    },
];
/**
 * Encodes the parameters for the "isApprovedForAll" function.
 * @param options - The options for the isApprovedForAll function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeIsApprovedForAllParams } "thirdweb/extensions/erc1155";
 * const result = encodeIsApprovedForAllParams({
 *  owner: ...,
 *  operator: ...,
 * });
 * ```
 */
function encodeIsApprovedForAllParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.owner, options.operator]);
}
exports.encodeIsApprovedForAllParams = encodeIsApprovedForAllParams;
/**
 * Decodes the result of the isApprovedForAll function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC1155
 * @example
 * ```ts
 * import { decodeIsApprovedForAllResult } from "thirdweb/extensions/erc1155";
 * const result = decodeIsApprovedForAllResult("...");
 * ```
 */
function decodeIsApprovedForAllResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeIsApprovedForAllResult = decodeIsApprovedForAllResult;
/**
 * Calls the "isApprovedForAll" function on the contract.
 * @param options - The options for the isApprovedForAll function.
 * @returns The parsed result of the function call.
 * @extension ERC1155
 * @example
 * ```ts
 * import { isApprovedForAll } from "thirdweb/extensions/erc1155";
 *
 * const result = await isApprovedForAll({
 *  owner: ...,
 *  operator: ...,
 * });
 *
 * ```
 */
async function isApprovedForAll(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.owner, options.operator],
    });
}
exports.isApprovedForAll = isApprovedForAll;
