// thirdweb/dist/cjs/extensions/erc4337/__generated__/IAccountPermissions/write/setPermissionsForSigner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setPermissionsForSigner = exports.encodeSetPermissionsForSignerParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x5892e236";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "req",
        components: [
            {
                type: "address",
                name: "signer",
            },
            {
                type: "uint8",
                name: "isAdmin",
            },
            {
                type: "address[]",
                name: "approvedTargets",
            },
            {
                type: "uint256",
                name: "nativeTokenLimitPerTransaction",
            },
            {
                type: "uint128",
                name: "permissionStartTimestamp",
            },
            {
                type: "uint128",
                name: "permissionEndTimestamp",
            },
            {
                type: "uint128",
                name: "reqValidityStartTimestamp",
            },
            {
                type: "uint128",
                name: "reqValidityEndTimestamp",
            },
            {
                type: "bytes32",
                name: "uid",
            },
        ],
    },
    {
        type: "bytes",
        name: "signature",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setPermissionsForSigner" function.
 * @param options - The options for the setPermissionsForSigner function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```ts
 * import { encodeSetPermissionsForSignerParams } "thirdweb/extensions/erc4337";
 * const result = encodeSetPermissionsForSignerParams({
 *  req: ...,
 *  signature: ...,
 * });
 * ```
 */
function encodeSetPermissionsForSignerParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.req, options.signature]);
}
exports.encodeSetPermissionsForSignerParams = encodeSetPermissionsForSignerParams;
/**
 * Calls the "setPermissionsForSigner" function on the contract.
 * @param options - The options for the "setPermissionsForSigner" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```ts
 * import { setPermissionsForSigner } from "thirdweb/extensions/erc4337";
 *
 * const transaction = setPermissionsForSigner({
 *  contract,
 *  req: ...,
 *  signature: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setPermissionsForSigner(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.req, resolvedParams.signature];
            }
            : [options.req, options.signature],
    });
}
exports.setPermissionsForSigner = setPermissionsForSigner;
