// thirdweb/dist/cjs/extensions/ens/__generated__/AddressResolver/read/addr.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addr = exports.decodeAddrResult = exports.encodeAddrParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x3b3b57de";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "name",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Encodes the parameters for the "addr" function.
 * @param options - The options for the addr function.
 * @returns The encoded ABI parameters.
 * @extension ENS
 * @example
 * ```
 * import { encodeAddrParams } "thirdweb/extensions/ens";
 * const result = encodeAddrParams({
 *  name: ...,
 * });
 * ```
 */
function encodeAddrParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.name]);
}
exports.encodeAddrParams = encodeAddrParams;
/**
 * Decodes the result of the addr function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ENS
 * @example
 * ```
 * import { decodeAddrResult } from "thirdweb/extensions/ens";
 * const result = decodeAddrResult("...");
 * ```
 */
function decodeAddrResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeAddrResult = decodeAddrResult;
/**
 * Calls the "addr" function on the contract.
 * @param options - The options for the addr function.
 * @returns The parsed result of the function call.
 * @extension ENS
 * @example
 * ```
 * import { addr } from "thirdweb/extensions/ens";
 *
 * const result = await addr({
 *  name: ...,
 * });
 *
 * ```
 */
async function addr(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.name],
    });
}
exports.addr = addr;
