// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/read/keyDataOf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.keyDataOf = exports.decodeKeyDataOfResult = exports.encodeKeyDataOfParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xac34cc5a";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "fid",
    },
    {
        type: "bytes",
        name: "key",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Encodes the parameters for the "keyDataOf" function.
 * @param options - The options for the keyDataOf function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeKeyDataOfParams } "thirdweb/extensions/farcaster";
 * const result = encodeKeyDataOfParams({
 *  fid: ...,
 *  key: ...,
 * });
 * ```
 */
function encodeKeyDataOfParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.fid, options.key]);
}
exports.encodeKeyDataOfParams = encodeKeyDataOfParams;
/**
 * Decodes the result of the keyDataOf function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeKeyDataOfResult } from "thirdweb/extensions/farcaster";
 * const result = decodeKeyDataOfResult("...");
 * ```
 */
function decodeKeyDataOfResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeKeyDataOfResult = decodeKeyDataOfResult;
/**
 * Calls the "keyDataOf" function on the contract.
 * @param options - The options for the keyDataOf function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { keyDataOf } from "thirdweb/extensions/farcaster";
 *
 * const result = await keyDataOf({
 *  fid: ...,
 *  key: ...,
 * });
 *
 * ```
 */
async function keyDataOf(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.fid, options.key],
    });
}
exports.keyDataOf = keyDataOf;
