// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/read/keyGateway.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.keyGateway = exports.decodeKeyGatewayResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x80334737";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Decodes the result of the keyGateway function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeKeyGatewayResult } from "thirdweb/extensions/farcaster";
 * const result = decodeKeyGatewayResult("...");
 * ```
 */
function decodeKeyGatewayResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeKeyGatewayResult = decodeKeyGatewayResult;
/**
 * Calls the "keyGateway" function on the contract.
 * @param options - The options for the keyGateway function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { keyGateway } from "thirdweb/extensions/farcaster";
 *
 * const result = await keyGateway();
 *
 * ```
 */
async function keyGateway(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.keyGateway = keyGateway;
