// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/read/keysOf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.keysOf = exports.decodeKeysOfResult = exports.encodeKeysOfParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x1f64222f";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "fid",
    },
    {
        type: "uint8",
        name: "state",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes[]",
    },
];
/**
 * Encodes the parameters for the "keysOf" function.
 * @param options - The options for the keysOf function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeKeysOfParams } "thirdweb/extensions/farcaster";
 * const result = encodeKeysOfParams({
 *  fid: ...,
 *  state: ...,
 * });
 * ```
 */
function encodeKeysOfParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.fid, options.state]);
}
exports.encodeKeysOfParams = encodeKeysOfParams;
/**
 * Decodes the result of the keysOf function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeKeysOfResult } from "thirdweb/extensions/farcaster";
 * const result = decodeKeysOfResult("...");
 * ```
 */
function decodeKeysOfResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeKeysOfResult = decodeKeysOfResult;
/**
 * Calls the "keysOf" function on the contract.
 * @param options - The options for the keysOf function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { keysOf } from "thirdweb/extensions/farcaster";
 *
 * const result = await keysOf({
 *  fid: ...,
 *  state: ...,
 * });
 *
 * ```
 */
async function keysOf(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.fid, options.state],
    });
}
exports.keysOf = keysOf;
