// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyRegistry/read/REMOVE_TYPEHASH.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.REMOVE_TYPEHASH = exports.decodeREMOVE_TYPEHASHResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xb5775561";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "bytes32",
    },
];
/**
 * Decodes the result of the REMOVE_TYPEHASH function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeREMOVE_TYPEHASHResult } from "thirdweb/extensions/farcaster";
 * const result = decodeREMOVE_TYPEHASHResult("...");
 * ```
 */
function decodeREMOVE_TYPEHASHResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeREMOVE_TYPEHASHResult = decodeREMOVE_TYPEHASHResult;
/**
 * Calls the "REMOVE_TYPEHASH" function on the contract.
 * @param options - The options for the REMOVE_TYPEHASH function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { REMOVE_TYPEHASH } from "thirdweb/extensions/farcaster";
 *
 * const result = await REMOVE_TYPEHASH();
 *
 * ```
 */
async function REMOVE_TYPEHASH(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.REMOVE_TYPEHASH = REMOVE_TYPEHASH;
