// thirdweb/dist/cjs/contract/deployment/utils/create-2-factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deployCreate2Factory = exports.getDeployedCreate2Factory = exports.computeCreate2FactoryAddress = void 0;
const viem_1 = require("viem");
const contract_js_1 = require("../../contract.js");
const is_contract_deployed_js_1 = require("../../../utils/bytecode/is-contract-deployed.js");
const is_eip155_enforced_js_1 = require("../../../utils/any-evm/is-eip155-enforced.js");
const keyless_transaction_js_1 = require("../../../utils/any-evm/keyless-transaction.js");
const eth_sendRawTransaction_js_1 = require("../../../rpc/actions/eth_sendRawTransaction.js");
const rpc_js_1 = require("../../../rpc/rpc.js");
const eth_getBalance_js_1 = require("../../../rpc/actions/eth_getBalance.js");
const prepare_transaction_js_1 = require("../../../transaction/prepare-transaction.js");
const send_transaction_js_1 = require("../../../transaction/actions/send-transaction.js");
const wait_for_tx_receipt_js_1 = require("../../../transaction/actions/wait-for-tx-receipt.js");
const COMMON_FACTORY_ADDRESS = "0x4e59b44847b379578588920cA78FbF26c0B4956C"; // for pre-eip-155 supporting chains
/**
 * @internal
 */
const CREATE2_FACTORY_BYTECODE = "0x604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3";
/**
 * @internal
 */
const SIGNATURE = {
    v: 27n,
    r: "0x2222222222222222222222222222222222222222222222222222222222222222",
    s: "0x2222222222222222222222222222222222222222222222222222222222222222",
};
/**
 * Computes the address of the Create2 factory contract and checks if it is deployed.
 * @param options - The options for retrieving the Create2 factory address.
 * @returns whether the Create2 factory is deployed.
 * @internal
 */
async function computeCreate2FactoryAddress(options) {
    // TODO add LRU cache
    const commonFactory = (0, contract_js_1.getContract)({
        ...options,
        address: COMMON_FACTORY_ADDRESS,
    });
    // if the common factory is deployed, return the address
    if (await (0, is_contract_deployed_js_1.isContractDeployed)(commonFactory)) {
        return COMMON_FACTORY_ADDRESS;
    }
    const enforceEip155 = await (0, is_eip155_enforced_js_1.isEIP155Enforced)(options);
    const chainId = options.chain.id;
    const custom = CUSTOM_GAS_FOR_CHAIN[chainId.toString()];
    const eipChain = enforceEip155 ? chainId : 0;
    const deploymentInfo = custom
        ? await _getCreate2FactoryDeploymentInfo(eipChain, {
            gasPrice: custom.gasPrice,
            gasLimit: custom.gasLimit,
        })
        : await _getCreate2FactoryDeploymentInfo(eipChain, {});
    return deploymentInfo.predictedAddress;
}
exports.computeCreate2FactoryAddress = computeCreate2FactoryAddress;
/**
 * @internal
 */
async function getDeployedCreate2Factory(options) {
    const address = await computeCreate2FactoryAddress(options);
    const factory = (0, contract_js_1.getContract)({
        ...options,
        address,
    });
    const isDeployed = await (0, is_contract_deployed_js_1.isContractDeployed)(factory);
    if (!isDeployed) {
        return null;
    }
    return factory;
}
exports.getDeployedCreate2Factory = getDeployedCreate2Factory;
/**
 * Deploys the Create2 factory contract using a keyless transaction.
 * @internal
 */
async function deployCreate2Factory(options) {
    const { client, chain, account } = options;
    const enforceEip155 = await (0, is_eip155_enforced_js_1.isEIP155Enforced)(options);
    const chainId = options.chain.id;
    const custom = CUSTOM_GAS_FOR_CHAIN[chainId.toString()];
    const eipChain = enforceEip155 ? chainId : 0;
    const deploymentInfo = custom
        ? await _getCreate2FactoryDeploymentInfo(eipChain, {
            gasPrice: custom.gasPrice,
            gasLimit: custom.gasLimit,
        })
        : await _getCreate2FactoryDeploymentInfo(eipChain, {});
    const rpcRequest = (0, rpc_js_1.getRpcClient)({
        client: client,
        chain,
    });
    const balance = await (0, eth_getBalance_js_1.eth_getBalance)(rpcRequest, {
        address: deploymentInfo.signerAddress,
    });
    if (balance < deploymentInfo.valueToSend) {
        const transaction = (0, prepare_transaction_js_1.prepareTransaction)({
            chain,
            client,
            to: deploymentInfo.signerAddress,
            value: deploymentInfo.valueToSend,
        });
        const res = await (0, send_transaction_js_1.sendTransaction)({ transaction, account });
        await (0, wait_for_tx_receipt_js_1.waitForReceipt)(res);
    }
    const transactionHash = await (0, eth_sendRawTransaction_js_1.eth_sendRawTransaction)(rpcRequest, deploymentInfo.transaction);
    return {
        transactionHash,
    };
}
exports.deployCreate2Factory = deployCreate2Factory;
/**
 * Retrieves the deployment information for the Create2 factory contract.
 * @param chainId - The chain ID.
 * @param gasOptions - The gas options for the deployment transaction.
 * @returns The deployment information, including the deployment transaction and the create2 factory address.
 * @internal
 */
async function _getCreate2FactoryDeploymentInfo(chainId, gasOptions) {
    const gasPrice = gasOptions.gasPrice ? gasOptions.gasPrice : 100n * 10n ** 9n;
    const gas = gasOptions.gasLimit ? gasOptions.gasLimit : 100000n;
    const deploymentTransaction = await (0, keyless_transaction_js_1.getKeylessTransaction)({
        transaction: {
            gasPrice,
            gas,
            nonce: 0,
            data: CREATE2_FACTORY_BYTECODE,
            chainId: Number(chainId),
        },
        signature: SIGNATURE,
    });
    const create2FactoryAddress = (0, viem_1.getContractAddress)({
        from: deploymentTransaction.signerAddress,
        nonce: 0n,
    });
    return {
        ...deploymentTransaction,
        valueToSend: gasPrice * gas,
        predictedAddress: create2FactoryAddress,
    };
}
const CUSTOM_GAS_FOR_CHAIN = {
    "5001": {
        name: "Mantle Testnet",
        gasPrice: 1n,
    },
    "71402": {
        name: "Godwoken Mainnet",
        gasPrice: 40000n * 10n ** 9n,
    },
    "1351057110": {
        name: "Chaos (SKALE Testnet)",
        gasPrice: 100000n,
    },
    "361": {
        name: "Theta Mainnet",
        gasPrice: 4000n * 10n ** 9n,
    },
    "365": {
        name: "Theta Testnet",
        gasPrice: 4000n * 10n ** 9n,
    },
    "7700": {
        name: "Canto",
        gasPrice: 1000n * 10n ** 9n,
    },
    "7701": {
        name: "Canto Testnet",
        gasPrice: 1000n * 10n ** 9n,
    },
    "338": {
        name: "Cronos Testnet",
        gasPrice: 2000n * 10n ** 9n,
    },
    "199": {
        name: "BitTorrent Chain",
        gasPrice: 300000n * 10n ** 9n,
    },
    "88882": {
        name: "Spicy Chain",
        gasPrice: 2500n * 10n ** 9n,
        gasLimit: 200000n,
    },
    "88888": {
        name: "Chiliz Chain",
        gasPrice: 2500n * 10n ** 9n,
        gasLimit: 200000n,
    },
};
