// thirdweb/dist/cjs/react/web/wallets/embedded/EmbeddedWalletOTPLoginUI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbeddedWalletOTPLoginUI = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const FadeIn_js_1 = require("../../ui/components/FadeIn.js");
const OTPInput_js_1 = require("../../ui/components/OTPInput.js");
const Spacer_js_1 = require("../../ui/components/Spacer.js");
const Spinner_js_1 = require("../../ui/components/Spinner.js");
const basic_js_1 = require("../../ui/components/basic.js");
const buttons_js_1 = require("../../ui/components/buttons.js");
const CustomThemeProvider_js_1 = require("../../ui/design-system/CustomThemeProvider.js");
const elements_js_1 = require("../../ui/design-system/elements.js");
const index_js_1 = require("../../ui/design-system/index.js");
const text_js_1 = require("../../ui/components/text.js");
const wallet_ui_states_provider_js_1 = require("../../providers/wallet-ui-states-provider.js");
const useWalletConnectionCtx_js_1 = require("../../../core/hooks/others/useWalletConnectionCtx.js");
const index_js_2 = require("../../../../wallets/embedded/core/authentication/index.js");
/**
 * @internal
 */
function EmbeddedWalletOTPLoginUI(props) {
    const { wallet, done, goBack } = props;
    const email = props.email;
    const { modalSize } = (0, react_1.useContext)(wallet_ui_states_provider_js_1.ModalConfigCtx);
    const isWideModal = modalSize === "wide";
    const locale = props.locale;
    const [otpInput, setOtpInput] = (0, react_1.useState)("");
    const { client, chain } = (0, useWalletConnectionCtx_js_1.useWalletConnectionCtx)();
    // const { createInstance, done, chain } = props.connectUIProps.connection;
    // const { goBack } = props.connectUIProps.screenConfig;
    // const [wallet, setWallet] = useState<EmbeddedWallet | null>(null);
    const [verifyStatus, setVerifyStatus] = (0, react_1.useState)("idle");
    const [emailStatus, setEmailStatus] = (0, react_1.useState)("sending");
    const [screen] = (0, react_1.useState)("base");
    const sendEmail = (0, react_1.useCallback)(async () => {
        setOtpInput("");
        setVerifyStatus("idle");
        setEmailStatus("sending");
        try {
            const status = await (0, index_js_2.preAuthenticate)({
                email,
                strategy: "email",
                client,
            });
            setEmailStatus(status);
        }
        catch (e) {
            console.error(e);
            setVerifyStatus("idle");
            setEmailStatus("error");
        }
    }, [client, email]);
    const verify = async (otp) => {
        if (typeof emailStatus !== "object" || otp.length !== 6) {
            return;
        }
        setVerifyStatus("idle");
        if (typeof emailStatus !== "object") {
            return;
        }
        if (!wallet) {
            return;
        }
        try {
            setVerifyStatus("verifying");
            const needsRecoveryCode = emailStatus.recoveryShareManagement === "USER_MANAGED" &&
                (emailStatus.isNewUser || emailStatus.isNewDevice);
            // USER_MANAGED
            if (needsRecoveryCode) {
                if (emailStatus.isNewUser) {
                    try {
                        await wallet.connect({
                            chain,
                            strategy: "email",
                            email,
                            verificationCode: otp,
                            client,
                        });
                    }
                    catch (e) {
                        if (e instanceof Error && e.message.includes("encryption key")) {
                            // TODO: do we need this?
                            // setScreen("create-password");
                        }
                        else {
                            throw e;
                        }
                    }
                }
                else {
                    try {
                        // verifies otp for UI feedback
                        await wallet.connect({
                            chain,
                            strategy: "email",
                            email,
                            verificationCode: otp,
                            client,
                        });
                    }
                    catch (e) {
                        if (e instanceof Error && e.message.includes("encryption key")) {
                            // TODO: do we need this?
                            // setScreen("enter-password-or-recovery-code");
                        }
                        else {
                            throw e;
                        }
                    }
                }
            }
            // AWS_MANAGED
            else {
                const authResult = await wallet.connect({
                    chain,
                    strategy: "email",
                    email,
                    verificationCode: otp,
                    client,
                });
                if (!authResult) {
                    throw new Error("Failed to verify OTP");
                }
                done();
            }
            setVerifyStatus("valid");
        }
        catch (e) {
            if (e?.message?.includes("PAYMENT_METHOD_REQUIRED")) {
                setVerifyStatus("payment_required");
            }
            else {
                setVerifyStatus("invalid");
            }
            console.error("Authentication Error", e);
        }
    };
    // send email on mount
    const emailSentOnMount = (0, react_1.useRef)(false);
    (0, react_1.useEffect)(() => {
        if (!emailSentOnMount.current) {
            emailSentOnMount.current = true;
            sendEmail();
        }
    }, [sendEmail]);
    // if (screen === "create-password") {
    //   return (
    //     <CreatePassword
    //       modalSize={size}
    //       email={email}
    //       goBack={goBack}
    //       onPassword={async (password) => {
    //         if (!wallet || typeof emailStatus !== "object") {
    //           return;
    //         }
    //         const authResult = await wallet.connect({
    //          chain,
    //           strategy: "email",
    //           email,
    //           verificationCode: otpInput,
    //           // recoveryCode: password,
    //         });
    //         if (!authResult) {
    //           throw new Error("Failed to verify recovery code");
    //         }
    //         await wallet.connect({
    //           authResult,
    //         });
    //         setConnectedWallet(wallet);
    //         props.connected();
    //       }}
    //     />
    //   );
    // }
    // if (screen === "enter-password-or-recovery-code") {
    //   return (
    //     <EnterPasswordOrRecovery
    //       modalSize={size}
    //       goBack={goBack}
    //       email={email}
    //       onVerify={async (passwordOrRecoveryCode) => {
    //         if (!wallet || typeof emailStatus !== "object") {
    //           return;
    //         }
    //         const authResult = await wallet.authenticate({
    //           strategy: "email_verification",
    //           email,
    //           verificationCode: otpInput,
    //           recoveryCode: passwordOrRecoveryCode,
    //         });
    //         if (!authResult) {
    //           throw new Error("Failed to verify recovery code");
    //         }
    //         await wallet.connect({
    //           authResult,
    //         });
    //         setConnectedWallet(wallet);
    //         props.connected();
    //       }}
    //     />
    //   );
    // }
    if (screen === "base") {
        return ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { fullHeight: true, flex: "column", animate: "fadein", children: [(0, jsx_runtime_1.jsx)(basic_js_1.Container, { p: "lg", children: (0, jsx_runtime_1.jsx)(basic_js_1.ModalHeader, { title: locale.signIn, onBack: goBack }) }), (0, jsx_runtime_1.jsx)(basic_js_1.Container, { expand: true, flex: "column", center: "y", children: (0, jsx_runtime_1.jsxs)("form", { onSubmit: (e) => {
                            e.preventDefault();
                        }, children: [(0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", center: "x", px: "lg", children: [!isWideModal && (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xl" }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { children: locale.emailLoginScreen.enterCodeSendTo }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "sm" }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { color: "primaryText", children: email }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xl" })] }), (0, jsx_runtime_1.jsx)(OTPInput_js_1.OTPInput, { isInvalid: verifyStatus === "invalid", digits: 6, value: otpInput, setValue: (value) => {
                                    setOtpInput(value);
                                    setVerifyStatus("idle"); // reset error
                                    verify(value);
                                }, onEnter: () => {
                                    verify(otpInput);
                                } }), verifyStatus === "invalid" && ((0, jsx_runtime_1.jsxs)(FadeIn_js_1.FadeIn, { children: [(0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "md" }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", color: "danger", center: true, children: locale.emailLoginScreen.invalidCode })] })), verifyStatus === "payment_required" && ((0, jsx_runtime_1.jsxs)(FadeIn_js_1.FadeIn, { children: [(0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "md" }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", color: "danger", center: true, children: locale.maxAccountsExceeded })] })), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xl" }), (0, jsx_runtime_1.jsx)(basic_js_1.Container, { px: isWideModal ? "xxl" : "lg", children: verifyStatus === "verifying" ? ((0, jsx_runtime_1.jsx)(jsx_runtime_1.Fragment, { children: (0, jsx_runtime_1.jsx)(basic_js_1.Container, { flex: "row", center: "x", animate: "fadein", children: (0, jsx_runtime_1.jsx)(Spinner_js_1.Spinner, { size: "lg", color: "accentText" }) }) })) : ((0, jsx_runtime_1.jsx)(basic_js_1.Container, { animate: "fadein", children: (0, jsx_runtime_1.jsx)(buttons_js_1.Button, { onClick: () => verify(otpInput), variant: "accent", type: "submit", style: {
                                            width: "100%",
                                        }, children: locale.emailLoginScreen.verify }) }, "btn-container")) }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xl" }), !isWideModal && (0, jsx_runtime_1.jsx)(basic_js_1.Line, {}), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { p: isWideModal ? undefined : "lg", children: [emailStatus === "error" && ((0, jsx_runtime_1.jsx)(jsx_runtime_1.Fragment, { children: (0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", center: true, color: "danger", children: locale.emailLoginScreen.failedToSendCode }) })), emailStatus === "sending" && ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", center: "both", gap: "xs", style: {
                                            textAlign: "center",
                                        }, children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", children: locale.emailLoginScreen.sendingCode }), (0, jsx_runtime_1.jsx)(Spinner_js_1.Spinner, { size: "xs", color: "secondaryText" })] })), typeof emailStatus === "object" && ((0, jsx_runtime_1.jsx)(LinkButton, { onClick: sendEmail, type: "button", children: locale.emailLoginScreen.resendCode }))] })] }) })] }));
    }
    return null;
}
exports.EmbeddedWalletOTPLoginUI = EmbeddedWalletOTPLoginUI;
const LinkButton = /* @__PURE__ */ (0, elements_js_1.StyledButton)(() => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        all: "unset",
        color: theme.colors.accentText,
        fontSize: index_js_1.fontSize.sm,
        cursor: "pointer",
        textAlign: "center",
        fontWeight: 500,
        width: "100%",
        "&:hover": {
            color: theme.colors.primaryText,
        },
    };
});
