// thirdweb/dist/cjs/extensions/erc721/__generated__/ISharedMetadataBatch/write/deleteSharedMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteSharedMetadata = exports.encodeDeleteSharedMetadata = exports.encodeDeleteSharedMetadataParams = exports.isDeleteSharedMetadataSupported = exports.FN_SELECTOR = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const once_js_1 = require("../../../../../utils/promise/once.js");
const detectExtension_js_1 = require("../../../../../utils/bytecode/detectExtension.js");
exports.FN_SELECTOR = "0x1ebb2422";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "id",
    },
];
const FN_OUTPUTS = [];
/**
 * Checks if the `deleteSharedMetadata` method is supported by the given contract.
 * @param contract The ThirdwebContract.
 * @returns A promise that resolves to a boolean indicating if the `deleteSharedMetadata` method is supported.
 * @extension ERC721
 * @example
 * ```ts
 * import { isDeleteSharedMetadataSupported } from "thirdweb/extensions/erc721";
 *
 * const supported = await isDeleteSharedMetadataSupported(contract);
 * ```
 */
async function isDeleteSharedMetadataSupported(contract) {
    return (0, detectExtension_js_1.detectMethod)({
        contract,
        method: [exports.FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
    });
}
exports.isDeleteSharedMetadataSupported = isDeleteSharedMetadataSupported;
/**
 * Encodes the parameters for the "deleteSharedMetadata" function.
 * @param options - The options for the deleteSharedMetadata function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeDeleteSharedMetadataParams } "thirdweb/extensions/erc721";
 * const result = encodeDeleteSharedMetadataParams({
 *  id: ...,
 * });
 * ```
 */
function encodeDeleteSharedMetadataParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.id]);
}
exports.encodeDeleteSharedMetadataParams = encodeDeleteSharedMetadataParams;
/**
 * Encodes the "deleteSharedMetadata" function into a Hex string with its parameters.
 * @param options - The options for the deleteSharedMetadata function.
 * @returns The encoded hexadecimal string.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeDeleteSharedMetadata } "thirdweb/extensions/erc721";
 * const result = encodeDeleteSharedMetadata({
 *  id: ...,
 * });
 * ```
 */
function encodeDeleteSharedMetadata(options) {
    // we do a "manual" concat here to avoid the overhead of the "concatHex" function
    // we can do this because we know the specific formats of the values
    return (exports.FN_SELECTOR +
        encodeDeleteSharedMetadataParams(options).slice(2));
}
exports.encodeDeleteSharedMetadata = encodeDeleteSharedMetadata;
/**
 * Calls the "deleteSharedMetadata" function on the contract.
 * @param options - The options for the "deleteSharedMetadata" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```ts
 * import { deleteSharedMetadata } from "thirdweb/extensions/erc721";
 *
 * const transaction = deleteSharedMetadata({
 *  contract,
 *  id: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function deleteSharedMetadata(options) {
    const asyncOptions = (0, once_js_1.once)(async () => {
        return "asyncParams" in options ? await options.asyncParams() : options;
    });
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [exports.FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: async () => {
            const resolvedOptions = await asyncOptions();
            return [resolvedOptions.id];
        },
        value: async () => (await asyncOptions()).overrides?.value,
    });
}
exports.deleteSharedMetadata = deleteSharedMetadata;
