// thirdweb/dist/cjs/extensions/erc721/__generated__/ISharedMetadataBatch/write/deleteSharedMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteSharedMetadata = exports.encodeDeleteSharedMetadataParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x1ebb2422";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "id",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "deleteSharedMetadata" function.
 * @param options - The options for the deleteSharedMetadata function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeDeleteSharedMetadataParams } "thirdweb/extensions/erc721";
 * const result = encodeDeleteSharedMetadataParams({
 *  id: ...,
 * });
 * ```
 */
function encodeDeleteSharedMetadataParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.id]);
}
exports.encodeDeleteSharedMetadataParams = encodeDeleteSharedMetadataParams;
/**
 * Calls the "deleteSharedMetadata" function on the contract.
 * @param options - The options for the "deleteSharedMetadata" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```ts
 * import { deleteSharedMetadata } from "thirdweb/extensions/erc721";
 *
 * const transaction = deleteSharedMetadata({
 *  contract,
 *  id: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function deleteSharedMetadata(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.id];
            }
            : [options.id],
    });
}
exports.deleteSharedMetadata = deleteSharedMetadata;
