// thirdweb/dist/cjs/extensions/thirdweb/__generated__/IAppURI/read/appURI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.appURI = exports.decodeAppURIResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x094ec830";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "string",
    },
];
/**
 * Decodes the result of the appURI function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeAppURIResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeAppURIResult("...");
 * ```
 */
function decodeAppURIResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeAppURIResult = decodeAppURIResult;
/**
 * Calls the "appURI" function on the contract.
 * @param options - The options for the appURI function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { appURI } from "thirdweb/extensions/thirdweb";
 *
 * const result = await appURI();
 *
 * ```
 */
async function appURI(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.appURI = appURI;
