// thirdweb/dist/cjs/extensions/farcaster/__generated__/IIdRegistry/read/custodyOf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.custodyOf = exports.decodeCustodyOfResult = exports.encodeCustodyOfParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x65269e47";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "fid",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "owner",
    },
];
/**
 * Encodes the parameters for the "custodyOf" function.
 * @param options - The options for the custodyOf function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeCustodyOfParams } "thirdweb/extensions/farcaster";
 * const result = encodeCustodyOfParams({
 *  fid: ...,
 * });
 * ```
 */
function encodeCustodyOfParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.fid]);
}
exports.encodeCustodyOfParams = encodeCustodyOfParams;
/**
 * Decodes the result of the custodyOf function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeCustodyOfResult } from "thirdweb/extensions/farcaster";
 * const result = decodeCustodyOfResult("...");
 * ```
 */
function decodeCustodyOfResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeCustodyOfResult = decodeCustodyOfResult;
/**
 * Calls the "custodyOf" function on the contract.
 * @param options - The options for the custodyOf function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { custodyOf } from "thirdweb/extensions/farcaster";
 *
 * const result = await custodyOf({
 *  fid: ...,
 * });
 *
 * ```
 */
async function custodyOf(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.fid],
    });
}
exports.custodyOf = custodyOf;
