// thirdweb/dist/cjs/extensions/farcaster/__generated__/IIdRegistry/read/verifyFidSignature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifyFidSignature = exports.decodeVerifyFidSignatureResult = exports.encodeVerifyFidSignatureParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x32faac70";
const FN_INPUTS = [
    {
        type: "address",
        name: "custodyAddress",
    },
    {
        type: "uint256",
        name: "fid",
    },
    {
        type: "bytes32",
        name: "digest",
    },
    {
        type: "bytes",
        name: "sig",
    },
];
const FN_OUTPUTS = [
    {
        type: "bool",
        name: "isValid",
    },
];
/**
 * Encodes the parameters for the "verifyFidSignature" function.
 * @param options - The options for the verifyFidSignature function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeVerifyFidSignatureParams } "thirdweb/extensions/farcaster";
 * const result = encodeVerifyFidSignatureParams({
 *  custodyAddress: ...,
 *  fid: ...,
 *  digest: ...,
 *  sig: ...,
 * });
 * ```
 */
function encodeVerifyFidSignatureParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.custodyAddress,
        options.fid,
        options.digest,
        options.sig,
    ]);
}
exports.encodeVerifyFidSignatureParams = encodeVerifyFidSignatureParams;
/**
 * Decodes the result of the verifyFidSignature function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeVerifyFidSignatureResult } from "thirdweb/extensions/farcaster";
 * const result = decodeVerifyFidSignatureResult("...");
 * ```
 */
function decodeVerifyFidSignatureResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeVerifyFidSignatureResult = decodeVerifyFidSignatureResult;
/**
 * Calls the "verifyFidSignature" function on the contract.
 * @param options - The options for the verifyFidSignature function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { verifyFidSignature } from "thirdweb/extensions/farcaster";
 *
 * const result = await verifyFidSignature({
 *  custodyAddress: ...,
 *  fid: ...,
 *  digest: ...,
 *  sig: ...,
 * });
 *
 * ```
 */
async function verifyFidSignature(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.custodyAddress, options.fid, options.digest, options.sig],
    });
}
exports.verifyFidSignature = verifyFidSignature;
