// thirdweb/dist/esm/extensions/thirdweb/__generated__/ITWMultichainRegistry/read/count.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x05d85eda";
const FN_INPUTS = [
    {
        type: "address",
        name: "_deployer",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "deploymentCount",
    },
];
/**
 * Encodes the parameters for the "count" function.
 * @param options - The options for the count function.
 * @returns The encoded ABI parameters.
 * @extension THIRDWEB
 * @example
 * ```
 * import { encodeCountParams } "thirdweb/extensions/thirdweb";
 * const result = encodeCountParams({
 *  deployer: ...,
 * });
 * ```
 */
export function encodeCountParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.deployer]);
}
/**
 * Decodes the result of the count function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeCountResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeCountResult("...");
 * ```
 */
export function decodeCountResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "count" function on the contract.
 * @param options - The options for the count function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { count } from "thirdweb/extensions/thirdweb";
 *
 * const result = await count({
 *  deployer: ...,
 * });
 *
 * ```
 */
export async function count(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.deployer],
    });
}
