// thirdweb/dist/esm/extensions/thirdweb/__generated__/ITWMultichainRegistry/read/getAll.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xeb077342";
const FN_INPUTS = [
    {
        type: "address",
        name: "_deployer",
    },
];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "allDeployments",
        components: [
            {
                type: "address",
                name: "deploymentAddress",
            },
            {
                type: "uint256",
                name: "chainId",
            },
            {
                type: "string",
                name: "metadataURI",
            },
        ],
    },
];
/**
 * Encodes the parameters for the "getAll" function.
 * @param options - The options for the getAll function.
 * @returns The encoded ABI parameters.
 * @extension THIRDWEB
 * @example
 * ```
 * import { encodeGetAllParams } "thirdweb/extensions/thirdweb";
 * const result = encodeGetAllParams({
 *  deployer: ...,
 * });
 * ```
 */
export function encodeGetAllParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.deployer]);
}
/**
 * Decodes the result of the getAll function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeGetAllResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeGetAllResult("...");
 * ```
 */
export function decodeGetAllResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "getAll" function on the contract.
 * @param options - The options for the getAll function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { getAll } from "thirdweb/extensions/thirdweb";
 *
 * const result = await getAll({
 *  deployer: ...,
 * });
 *
 * ```
 */
export async function getAll(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.deployer],
    });
}
