// thirdweb/dist/esm/extensions/thirdweb/__generated__/IRulesEngine/read/getRulesEngineOverride.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0xa7145eb4";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "rulesEngineAddress",
    },
];
/**
 * Decodes the result of the getRulesEngineOverride function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeGetRulesEngineOverrideResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeGetRulesEngineOverrideResult("...");
 * ```
 */
export function decodeGetRulesEngineOverrideResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "getRulesEngineOverride" function on the contract.
 * @param options - The options for the getRulesEngineOverride function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { getRulesEngineOverride } from "thirdweb/extensions/thirdweb";
 *
 * const result = await getRulesEngineOverride();
 *
 * ```
 */
export async function getRulesEngineOverride(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
