// thirdweb/dist/esm/exports/extensions/erc1155.js
// READ
export { balanceOfBatch, } from "../../extensions/erc1155/__generated__/IERC1155/read/balanceOfBatch.js";
export { getNFT, } from "../../extensions/erc1155/read/getNFT.js";
export { getNFTs, } from "../../extensions/erc1155/read/getNFTs.js";
export { getOwnedNFTs, } from "../../extensions/erc1155/read/getOwnedNFTs.js";
export { nextTokenIdToMint } from "../../extensions/erc1155/__generated__/IERC1155Enumerable/read/nextTokenIdToMint.js";
export { uri, 
/**
 * @alias for `uri`
 */
uri as tokenUri, } from "../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
export { totalSupply, } from "../../extensions/erc1155/__generated__/IERC1155/read/totalSupply.js";
export { balanceOf, } from "../../extensions/erc1155/__generated__/IERC1155/read/balanceOf.js";
//WRITE
export { burn, } from "../../extensions/erc1155/__generated__/IBurnableERC1155/write/burn.js";
export { burnBatch, } from "../../extensions/erc1155/__generated__/IBurnableERC1155/write/burnBatch.js";
/**
 * DROPS extension for ERC1155
 */
export { getActiveClaimCondition } from "../../extensions/erc1155/drops/read/getActiveClaimCondition.js";
export { claimTo, } from "../../extensions/erc1155/drops/write/claimTo.js";
