// thirdweb/dist/esm/exports/extensions/erc1155.js
// READ
export { isERC1155 } from "../../extensions/erc1155/read/isERC1155.js";
export { balanceOfBatch, } from "../../extensions/erc1155/__generated__/IERC1155/read/balanceOfBatch.js";
export { getNFT, } from "../../extensions/erc1155/read/getNFT.js";
export { getNFTs, } from "../../extensions/erc1155/read/getNFTs.js";
export { getOwnedNFTs, } from "../../extensions/erc1155/read/getOwnedNFTs.js";
export { nextTokenIdToMint } from "../../extensions/erc1155/__generated__/IERC1155Enumerable/read/nextTokenIdToMint.js";
export { uri, 
/**
 * @alias for `uri`
 */
uri as tokenUri, } from "../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
export { totalSupply, } from "../../extensions/erc1155/__generated__/IERC1155/read/totalSupply.js";
export { balanceOf, } from "../../extensions/erc1155/__generated__/IERC1155/read/balanceOf.js";
export { isApprovedForAll, } from "../../extensions/erc1155/__generated__/IERC1155/read/isApprovedForAll.js";
//WRITE
export { burn, } from "../../extensions/erc1155/__generated__/IBurnableERC1155/write/burn.js";
export { burnBatch, } from "../../extensions/erc1155/__generated__/IBurnableERC1155/write/burnBatch.js";
export { setApprovalForAll, } from "../../extensions/erc1155/__generated__/IERC1155/write/setApprovalForAll.js";
export { safeTransferFrom, } from "../../extensions/erc1155/__generated__/IERC1155/write/safeTransferFrom.js";
export { safeBatchTransferFrom, } from "../../extensions/erc1155/__generated__/IERC1155/write/safeBatchTransferFrom.js";
export { mintTo, } from "../../extensions/erc1155/write/mintTo.js";
export { mintAdditionalSupplyTo, } from "../../extensions/erc1155/write/mintAdditionalSupplyTo.js";
// EVENTS
export { transferSingleEvent, } from "../../extensions/erc1155/__generated__/IERC1155/events/TransferSingle.js";
export { transferBatchEvent, } from "../../extensions/erc1155/__generated__/IERC1155/events/TransferBatch.js";
export { approvalForAllEvent, } from "../../extensions/erc1155/__generated__/IERC1155/events/ApprovalForAll.js";
/**
 * DROPS extension for ERC1155
 */
export { getActiveClaimCondition } from "../../extensions/erc1155/drops/read/getActiveClaimCondition.js";
export { claimTo, } from "../../extensions/erc1155/drops/write/claimTo.js";
export { lazyMint, } from "../../extensions/erc1155/write/lazyMint.js";
export { setClaimConditions, } from "../../extensions/erc1155/drops/write/setClaimConditions.js";
// EVENTS
export { tokensLazyMintedEvent } from "../../extensions/erc1155/__generated__/ILazyMint/events/TokensLazyMinted.js";
export { tokensClaimedEvent } from "../../extensions/erc1155/__generated__/IDrop1155/events/TokensClaimed.js";
