// thirdweb/dist/esm/exports/extensions/erc20.js
// read
export { getBalance, } from "../../extensions/erc20/read/getBalance.js";
export { balanceOf, } from "../../extensions/erc20/__generated__/IERC20/read/balanceOf.js";
export { decimals } from "../../extensions/erc20/read/decimals.js";
export { totalSupply } from "../../extensions/erc20/__generated__/IERC20/read/totalSupply.js";
export { allowance, } from "../../extensions/erc20/__generated__/IERC20/read/allowance.js";
export { getCurrencyMetadata, } from "../../extensions/erc20/read/getCurrencyMetadata.js";
// write
export { mintTo, } from "../../extensions/erc20/write/mintTo.js";
export { transfer, } from "../../extensions/erc20/write/transfer.js";
export { transferFrom, } from "../../extensions/erc20/write/transferFrom.js";
export { approve, } from "../../extensions/erc20/write/approve.js";
export { burn, } from "../../extensions/erc20/__generated__/IBurnableERC20/write/burn.js";
export { burnFrom, } from "../../extensions/erc20/__generated__/IBurnableERC20/write/burnFrom.js";
