// thirdweb/dist/esm/extensions/farcaster/__generated__/IIdRegistry/read/custodyOf.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x65269e47";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "fid",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "owner",
    },
];
/**
 * Encodes the parameters for the "custodyOf" function.
 * @param options - The options for the custodyOf function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeCustodyOfParams } "thirdweb/extensions/farcaster";
 * const result = encodeCustodyOfParams({
 *  fid: ...,
 * });
 * ```
 */
export function encodeCustodyOfParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.fid]);
}
/**
 * Decodes the result of the custodyOf function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```
 * import { decodeCustodyOfResult } from "thirdweb/extensions/farcaster";
 * const result = decodeCustodyOfResult("...");
 * ```
 */
export function decodeCustodyOfResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "custodyOf" function on the contract.
 * @param options - The options for the custodyOf function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```
 * import { custodyOf } from "thirdweb/extensions/farcaster";
 *
 * const result = await custodyOf({
 *  fid: ...,
 * });
 *
 * ```
 */
export async function custodyOf(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.fid],
    });
}
