// thirdweb/dist/cjs/extensions/erc4626/__generated__/IERC4626/read/asset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.asset = exports.decodeAssetResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x38d52e0f";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        name: "assetTokenAddress",
        type: "address",
        internalType: "contract ERC20",
    },
];
/**
 * Decodes the result of the asset function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4626
 * @example
 * ```
 * import { decodeAssetResult } from "thirdweb/extensions/erc4626";
 * const result = decodeAssetResult("...");
 * ```
 */
function decodeAssetResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeAssetResult = decodeAssetResult;
/**
 * Calls the "asset" function on the contract.
 * @param options - The options for the asset function.
 * @returns The parsed result of the function call.
 * @extension ERC4626
 * @example
 * ```
 * import { asset } from "thirdweb/extensions/erc4626";
 *
 * const result = await asset();
 *
 * ```
 */
async function asset(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.asset = asset;
