// thirdweb/dist/cjs/extensions/erc4626/__generated__/IERC4626/read/maxMint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.maxMint = exports.decodeMaxMintResult = exports.encodeMaxMintParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xc63d75b6";
const FN_INPUTS = [
    {
        name: "receiver",
        type: "address",
        internalType: "address",
    },
];
const FN_OUTPUTS = [
    {
        name: "maxShares",
        type: "uint256",
        internalType: "uint256",
    },
];
/**
 * Encodes the parameters for the "maxMint" function.
 * @param options - The options for the maxMint function.
 * @returns The encoded ABI parameters.
 * @extension ERC4626
 * @example
 * ```ts
 * import { encodeMaxMintParams } "thirdweb/extensions/erc4626";
 * const result = encodeMaxMintParams({
 *  receiver: ...,
 * });
 * ```
 */
function encodeMaxMintParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.receiver]);
}
exports.encodeMaxMintParams = encodeMaxMintParams;
/**
 * Decodes the result of the maxMint function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4626
 * @example
 * ```ts
 * import { decodeMaxMintResult } from "thirdweb/extensions/erc4626";
 * const result = decodeMaxMintResult("...");
 * ```
 */
function decodeMaxMintResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeMaxMintResult = decodeMaxMintResult;
/**
 * Calls the "maxMint" function on the contract.
 * @param options - The options for the maxMint function.
 * @returns The parsed result of the function call.
 * @extension ERC4626
 * @example
 * ```ts
 * import { maxMint } from "thirdweb/extensions/erc4626";
 *
 * const result = await maxMint({
 *  receiver: ...,
 * });
 *
 * ```
 */
async function maxMint(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.receiver],
    });
}
exports.maxMint = maxMint;
