// thirdweb/dist/cjs/extensions/erc4626/__generated__/IERC4626/read/totalAssets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.totalAssets = exports.decodeTotalAssetsResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x01e1d114";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        name: "totalManagedAssets",
        type: "uint256",
        internalType: "uint256",
    },
];
/**
 * Decodes the result of the totalAssets function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC4626
 * @example
 * ```ts
 * import { decodeTotalAssetsResult } from "thirdweb/extensions/erc4626";
 * const result = decodeTotalAssetsResult("...");
 * ```
 */
function decodeTotalAssetsResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTotalAssetsResult = decodeTotalAssetsResult;
/**
 * Calls the "totalAssets" function on the contract.
 * @param options - The options for the totalAssets function.
 * @returns The parsed result of the function call.
 * @extension ERC4626
 * @example
 * ```ts
 * import { totalAssets } from "thirdweb/extensions/erc4626";
 *
 * const result = await totalAssets();
 *
 * ```
 */
async function totalAssets(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.totalAssets = totalAssets;
