// thirdweb/dist/cjs/extensions/erc1155/__generated__/IStaking1155/write/withdraw.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.withdraw = exports.encodeWithdrawParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xc434dcfe";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "tokenId",
    },
    {
        type: "uint64",
        name: "amount",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "withdraw" function.
 * @param options - The options for the withdraw function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```
 * import { encodeWithdrawParams } "thirdweb/extensions/erc1155";
 * const result = encodeWithdrawParams({
 *  tokenId: ...,
 *  amount: ...,
 * });
 * ```
 */
function encodeWithdrawParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.tokenId, options.amount]);
}
exports.encodeWithdrawParams = encodeWithdrawParams;
/**
 * Calls the "withdraw" function on the contract.
 * @param options - The options for the "withdraw" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```
 * import { withdraw } from "thirdweb/extensions/erc1155";
 *
 * const transaction = withdraw({
 *  tokenId: ...,
 *  amount: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function withdraw(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.tokenId, resolvedParams.amount];
            }
            : [options.tokenId, options.amount],
    });
}
exports.withdraw = withdraw;
