// thirdweb/dist/esm/extensions/common/__generated__/IClaimConditionsSinglePhase/write/setClaimConditions.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x426cfaf3";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "phase",
        components: [
            {
                type: "uint256",
                name: "startTimestamp",
            },
            {
                type: "uint256",
                name: "maxClaimableSupply",
            },
            {
                type: "uint256",
                name: "supplyClaimed",
            },
            {
                type: "uint256",
                name: "quantityLimitPerWallet",
            },
            {
                type: "bytes32",
                name: "merkleRoot",
            },
            {
                type: "uint256",
                name: "pricePerToken",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "string",
                name: "metadata",
            },
        ],
    },
    {
        type: "bool",
        name: "resetClaimEligibility",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setClaimConditions" function.
 * @param options - The options for the setClaimConditions function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```ts
 * import { encodeSetClaimConditionsParams } "thirdweb/extensions/common";
 * const result = encodeSetClaimConditionsParams({
 *  phase: ...,
 *  resetClaimEligibility: ...,
 * });
 * ```
 */
export function encodeSetClaimConditionsParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.phase,
        options.resetClaimEligibility,
    ]);
}
/**
 * Calls the "setClaimConditions" function on the contract.
 * @param options - The options for the "setClaimConditions" function.
 * @returns A prepared transaction object.
 * @extension COMMON
 * @example
 * ```ts
 * import { setClaimConditions } from "thirdweb/extensions/common";
 *
 * const transaction = setClaimConditions({
 *  contract,
 *  phase: ...,
 *  resetClaimEligibility: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function setClaimConditions(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.phase,
                    resolvedParams.resetClaimEligibility,
                ];
            }
            : [options.phase, options.resetClaimEligibility],
    });
}
