// thirdweb/dist/cjs/extensions/erc721/__generated__/IDelayedReveal/write/reveal.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reveal = exports.encodeRevealParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xce805642";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "identifier",
    },
    {
        type: "bytes",
        name: "key",
    },
];
const FN_OUTPUTS = [
    {
        type: "string",
        name: "revealedURI",
    },
];
/**
 * Encodes the parameters for the "reveal" function.
 * @param options - The options for the reveal function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeRevealParams } "thirdweb/extensions/erc721";
 * const result = encodeRevealParams({
 *  identifier: ...,
 *  key: ...,
 * });
 * ```
 */
function encodeRevealParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.identifier, options.key]);
}
exports.encodeRevealParams = encodeRevealParams;
/**
 * Calls the "reveal" function on the contract.
 * @param options - The options for the "reveal" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```
 * import { reveal } from "thirdweb/extensions/erc721";
 *
 * const transaction = reveal({
 *  identifier: ...,
 *  key: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function reveal(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.identifier, resolvedParams.key];
            }
            : [options.identifier, options.key],
    });
}
exports.reveal = reveal;
