// thirdweb/dist/cjs/extensions/thirdweb/__generated__/ITWMultichainRegistry/read/count.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.count = exports.decodeCountResult = exports.encodeCountParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x05d85eda";
const FN_INPUTS = [
    {
        type: "address",
        name: "_deployer",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "deploymentCount",
    },
];
/**
 * Encodes the parameters for the "count" function.
 * @param options - The options for the count function.
 * @returns The encoded ABI parameters.
 * @extension THIRDWEB
 * @example
 * ```
 * import { encodeCountParams } "thirdweb/extensions/thirdweb";
 * const result = encodeCountParams({
 *  deployer: ...,
 * });
 * ```
 */
function encodeCountParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.deployer]);
}
exports.encodeCountParams = encodeCountParams;
/**
 * Decodes the result of the count function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension THIRDWEB
 * @example
 * ```
 * import { decodeCountResult } from "thirdweb/extensions/thirdweb";
 * const result = decodeCountResult("...");
 * ```
 */
function decodeCountResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeCountResult = decodeCountResult;
/**
 * Calls the "count" function on the contract.
 * @param options - The options for the count function.
 * @returns The parsed result of the function call.
 * @extension THIRDWEB
 * @example
 * ```
 * import { count } from "thirdweb/extensions/thirdweb";
 *
 * const result = await count({
 *  deployer: ...,
 * });
 *
 * ```
 */
async function count(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.deployer],
    });
}
exports.count = count;
