// thirdweb/dist/cjs/extensions/erc6551/__generated__/IERC6551Account/read/isValidSigner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isValidSigner = exports.decodeIsValidSignerResult = exports.encodeIsValidSignerParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x523e3260";
const FN_INPUTS = [
    {
        type: "address",
        name: "signer",
    },
    {
        type: "bytes",
        name: "context",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes4",
        name: "magicValue",
    },
];
/**
 * Encodes the parameters for the "isValidSigner" function.
 * @param options - The options for the isValidSigner function.
 * @returns The encoded ABI parameters.
 * @extension ERC6551
 * @example
 * ```
 * import { encodeIsValidSignerParams } "thirdweb/extensions/erc6551";
 * const result = encodeIsValidSignerParams({
 *  signer: ...,
 *  context: ...,
 * });
 * ```
 */
function encodeIsValidSignerParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.signer, options.context]);
}
exports.encodeIsValidSignerParams = encodeIsValidSignerParams;
/**
 * Decodes the result of the isValidSigner function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC6551
 * @example
 * ```
 * import { decodeIsValidSignerResult } from "thirdweb/extensions/erc6551";
 * const result = decodeIsValidSignerResult("...");
 * ```
 */
function decodeIsValidSignerResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeIsValidSignerResult = decodeIsValidSignerResult;
/**
 * Calls the "isValidSigner" function on the contract.
 * @param options - The options for the isValidSigner function.
 * @returns The parsed result of the function call.
 * @extension ERC6551
 * @example
 * ```
 * import { isValidSigner } from "thirdweb/extensions/erc6551";
 *
 * const result = await isValidSigner({
 *  signer: ...,
 *  context: ...,
 * });
 *
 * ```
 */
async function isValidSigner(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.signer, options.context],
    });
}
exports.isValidSigner = isValidSigner;
