// thirdweb/dist/cjs/extensions/erc6551/__generated__/IERC6551Account/read/token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.token = exports.decodeTokenResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xfc0c546a";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "chainId",
    },
    {
        type: "address",
        name: "tokenContract",
    },
    {
        type: "uint256",
        name: "tokenId",
    },
];
/**
 * Decodes the result of the token function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC6551
 * @example
 * ```
 * import { decodeTokenResult } from "thirdweb/extensions/erc6551";
 * const result = decodeTokenResult("...");
 * ```
 */
function decodeTokenResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result);
}
exports.decodeTokenResult = decodeTokenResult;
/**
 * Calls the "token" function on the contract.
 * @param options - The options for the token function.
 * @returns The parsed result of the function call.
 * @extension ERC6551
 * @example
 * ```
 * import { token } from "thirdweb/extensions/erc6551";
 *
 * const result = await token();
 *
 * ```
 */
async function token(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.token = token;
