// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721AQueryable/read/tokensOfOwnerIn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tokensOfOwnerIn = exports.decodeTokensOfOwnerInResult = exports.encodeTokensOfOwnerInParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x99a2557a";
const FN_INPUTS = [
    {
        type: "address",
        name: "owner",
    },
    {
        type: "uint256",
        name: "start",
    },
    {
        type: "uint256",
        name: "stop",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256[]",
    },
];
/**
 * Encodes the parameters for the "tokensOfOwnerIn" function.
 * @param options - The options for the tokensOfOwnerIn function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeTokensOfOwnerInParams } "thirdweb/extensions/erc721";
 * const result = encodeTokensOfOwnerInParams({
 *  owner: ...,
 *  start: ...,
 *  stop: ...,
 * });
 * ```
 */
function encodeTokensOfOwnerInParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.owner,
        options.start,
        options.stop,
    ]);
}
exports.encodeTokensOfOwnerInParams = encodeTokensOfOwnerInParams;
/**
 * Decodes the result of the tokensOfOwnerIn function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeTokensOfOwnerInResult } from "thirdweb/extensions/erc721";
 * const result = decodeTokensOfOwnerInResult("...");
 * ```
 */
function decodeTokensOfOwnerInResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTokensOfOwnerInResult = decodeTokensOfOwnerInResult;
/**
 * Calls the "tokensOfOwnerIn" function on the contract.
 * @param options - The options for the tokensOfOwnerIn function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { tokensOfOwnerIn } from "thirdweb/extensions/erc721";
 *
 * const result = await tokensOfOwnerIn({
 *  owner: ...,
 *  start: ...,
 *  stop: ...,
 * });
 *
 * ```
 */
async function tokensOfOwnerIn(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.owner, options.start, options.stop],
    });
}
exports.tokensOfOwnerIn = tokensOfOwnerIn;
