// thirdweb/dist/cjs/extensions/erc721/__generated__/ISharedMetadataBatch/read/getAllSharedMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllSharedMetadata = exports.decodeGetAllSharedMetadataResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xfc3c2a73";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "metadata",
        components: [
            {
                type: "bytes32",
                name: "id",
            },
            {
                type: "tuple",
                name: "metadata",
                components: [
                    {
                        type: "string",
                        name: "name",
                    },
                    {
                        type: "string",
                        name: "description",
                    },
                    {
                        type: "string",
                        name: "imageURI",
                    },
                    {
                        type: "string",
                        name: "animationURI",
                    },
                ],
            },
        ],
    },
];
/**
 * Decodes the result of the getAllSharedMetadata function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeGetAllSharedMetadataResult } from "thirdweb/extensions/erc721";
 * const result = decodeGetAllSharedMetadataResult("...");
 * ```
 */
function decodeGetAllSharedMetadataResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeGetAllSharedMetadataResult = decodeGetAllSharedMetadataResult;
/**
 * Calls the "getAllSharedMetadata" function on the contract.
 * @param options - The options for the getAllSharedMetadata function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { getAllSharedMetadata } from "thirdweb/extensions/erc721";
 *
 * const result = await getAllSharedMetadata();
 *
 * ```
 */
async function getAllSharedMetadata(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.getAllSharedMetadata = getAllSharedMetadata;
