// thirdweb/dist/cjs/extensions/common/__generated__/IRoyalty/read/royaltyInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.royaltyInfo = exports.decodeRoyaltyInfoResult = exports.encodeRoyaltyInfoParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x2a55205a";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "tokenId",
    },
    {
        type: "uint256",
        name: "salePrice",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "receiver",
    },
    {
        type: "uint256",
        name: "royaltyAmount",
    },
];
/**
 * Encodes the parameters for the "royaltyInfo" function.
 * @param options - The options for the royaltyInfo function.
 * @returns The encoded ABI parameters.
 * @extension COMMON
 * @example
 * ```
 * import { encodeRoyaltyInfoParams } "thirdweb/extensions/common";
 * const result = encodeRoyaltyInfoParams({
 *  tokenId: ...,
 *  salePrice: ...,
 * });
 * ```
 */
function encodeRoyaltyInfoParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.tokenId, options.salePrice]);
}
exports.encodeRoyaltyInfoParams = encodeRoyaltyInfoParams;
/**
 * Decodes the result of the royaltyInfo function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```
 * import { decodeRoyaltyInfoResult } from "thirdweb/extensions/common";
 * const result = decodeRoyaltyInfoResult("...");
 * ```
 */
function decodeRoyaltyInfoResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result);
}
exports.decodeRoyaltyInfoResult = decodeRoyaltyInfoResult;
/**
 * Calls the "royaltyInfo" function on the contract.
 * @param options - The options for the royaltyInfo function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```
 * import { royaltyInfo } from "thirdweb/extensions/common";
 *
 * const result = await royaltyInfo({
 *  tokenId: ...,
 *  salePrice: ...,
 * });
 *
 * ```
 */
async function royaltyInfo(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.tokenId, options.salePrice],
    });
}
exports.royaltyInfo = royaltyInfo;
