// thirdweb/dist/cjs/extensions/uniswap/__generated__/IQuoter/write/quoteExactInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.quoteExactInput = exports.encodeQuoteExactInputParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xcdca1753";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "path",
    },
    {
        type: "uint256",
        name: "amountIn",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "amountOut",
    },
];
/**
 * Encodes the parameters for the "quoteExactInput" function.
 * @param options - The options for the quoteExactInput function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { encodeQuoteExactInputParams } "thirdweb/extensions/uniswap";
 * const result = encodeQuoteExactInputParams({
 *  path: ...,
 *  amountIn: ...,
 * });
 * ```
 */
function encodeQuoteExactInputParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.path, options.amountIn]);
}
exports.encodeQuoteExactInputParams = encodeQuoteExactInputParams;
/**
 * Calls the "quoteExactInput" function on the contract.
 * @param options - The options for the "quoteExactInput" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```ts
 * import { quoteExactInput } from "thirdweb/extensions/uniswap";
 *
 * const transaction = quoteExactInput({
 *  contract,
 *  path: ...,
 *  amountIn: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function quoteExactInput(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.path, resolvedParams.amountIn];
            }
            : [options.path, options.amountIn],
    });
}
exports.quoteExactInput = quoteExactInput;
