// thirdweb/dist/cjs/extensions/uniswap/__generated__/IQuoter/write/quoteExactOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.quoteExactOutput = exports.encodeQuoteExactOutputParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x2f80bb1d";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "path",
    },
    {
        type: "uint256",
        name: "amountOut",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "amountIn",
    },
];
/**
 * Encodes the parameters for the "quoteExactOutput" function.
 * @param options - The options for the quoteExactOutput function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeQuoteExactOutputParams } "thirdweb/extensions/uniswap";
 * const result = encodeQuoteExactOutputParams({
 *  path: ...,
 *  amountOut: ...,
 * });
 * ```
 */
function encodeQuoteExactOutputParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.path, options.amountOut]);
}
exports.encodeQuoteExactOutputParams = encodeQuoteExactOutputParams;
/**
 * Calls the "quoteExactOutput" function on the contract.
 * @param options - The options for the "quoteExactOutput" function.
 * @returns A prepared transaction object.
 * @extension UNISWAP
 * @example
 * ```
 * import { quoteExactOutput } from "thirdweb/extensions/uniswap";
 *
 * const transaction = quoteExactOutput({
 *  path: ...,
 *  amountOut: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function quoteExactOutput(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.path, resolvedParams.amountOut];
            }
            : [options.path, options.amountOut],
    });
}
exports.quoteExactOutput = quoteExactOutput;
