// thirdweb/dist/esm/extensions/thirdweb/__generated__/ITWMultichainRegistry/write/add.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x26c5b516";
const FN_INPUTS = [
    {
        type: "address",
        name: "_deployer",
    },
    {
        type: "address",
        name: "_deployment",
    },
    {
        type: "uint256",
        name: "_chainId",
    },
    {
        type: "string",
        name: "metadataUri",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "add" function.
 * @param options - The options for the add function.
 * @returns The encoded ABI parameters.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { encodeAddParams } "thirdweb/extensions/thirdweb";
 * const result = encodeAddParams({
 *  deployer: ...,
 *  deployment: ...,
 *  chainId: ...,
 *  metadataUri: ...,
 * });
 * ```
 */
export function encodeAddParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.deployer,
        options.deployment,
        options.chainId,
        options.metadataUri,
    ]);
}
/**
 * Calls the "add" function on the contract.
 * @param options - The options for the "add" function.
 * @returns A prepared transaction object.
 * @extension THIRDWEB
 * @example
 * ```ts
 * import { add } from "thirdweb/extensions/thirdweb";
 *
 * const transaction = add({
 *  contract,
 *  deployer: ...,
 *  deployment: ...,
 *  chainId: ...,
 *  metadataUri: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function add(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.deployer,
                    resolvedParams.deployment,
                    resolvedParams.chainId,
                    resolvedParams.metadataUri,
                ];
            }
            : [
                options.deployer,
                options.deployment,
                options.chainId,
                options.metadataUri,
            ],
    });
}
