// thirdweb/dist/cjs/extensions/common/__generated__/IOwnable/read/owner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.owner = exports.decodeOwnerResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x8da5cb5b";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Decodes the result of the owner function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension COMMON
 * @example
 * ```ts
 * import { decodeOwnerResult } from "thirdweb/extensions/common";
 * const result = decodeOwnerResult("...");
 * ```
 */
function decodeOwnerResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeOwnerResult = decodeOwnerResult;
/**
 * Calls the "owner" function on the contract.
 * @param options - The options for the owner function.
 * @returns The parsed result of the function call.
 * @extension COMMON
 * @example
 * ```ts
 * import { owner } from "thirdweb/extensions/common";
 *
 * const result = await owner();
 *
 * ```
 */
async function owner(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.owner = owner;
