// thirdweb/dist/cjs/extensions/erc721/__generated__/ISharedMetadata/write/setSharedMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setSharedMetadata = exports.encodeSetSharedMetadataParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa7d27d9d";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "_metadata",
        components: [
            {
                type: "string",
                name: "name",
            },
            {
                type: "string",
                name: "description",
            },
            {
                type: "string",
                name: "imageURI",
            },
            {
                type: "string",
                name: "animationURI",
            },
        ],
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setSharedMetadata" function.
 * @param options - The options for the setSharedMetadata function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeSetSharedMetadataParams } "thirdweb/extensions/erc721";
 * const result = encodeSetSharedMetadataParams({
 *  metadata: ...,
 * });
 * ```
 */
function encodeSetSharedMetadataParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.metadata]);
}
exports.encodeSetSharedMetadataParams = encodeSetSharedMetadataParams;
/**
 * Calls the "setSharedMetadata" function on the contract.
 * @param options - The options for the "setSharedMetadata" function.
 * @returns A prepared transaction object.
 * @extension ERC721
 * @example
 * ```
 * import { setSharedMetadata } from "thirdweb/extensions/erc721";
 *
 * const transaction = setSharedMetadata({
 *  metadata: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setSharedMetadata(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.metadata];
            }
            : [options.metadata],
    });
}
exports.setSharedMetadata = setSharedMetadata;
