// thirdweb/dist/cjs/extensions/erc20/__generated__/IStaking20/write/stake.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stake = exports.encodeStakeParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa694fc3a";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "amount",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "stake" function.
 * @param options - The options for the stake function.
 * @returns The encoded ABI parameters.
 * @extension ERC20
 * @example
 * ```ts
 * import { encodeStakeParams } "thirdweb/extensions/erc20";
 * const result = encodeStakeParams({
 *  amount: ...,
 * });
 * ```
 */
function encodeStakeParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.amount]);
}
exports.encodeStakeParams = encodeStakeParams;
/**
 * Calls the "stake" function on the contract.
 * @param options - The options for the "stake" function.
 * @returns A prepared transaction object.
 * @extension ERC20
 * @example
 * ```ts
 * import { stake } from "thirdweb/extensions/erc20";
 *
 * const transaction = stake({
 *  contract,
 *  amount: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function stake(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.amount];
            }
            : [options.amount],
    });
}
exports.stake = stake;
