// thirdweb/dist/cjs/extensions/farcaster/__generated__/IKeyGateway/write/addFor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addFor = exports.encodeAddForParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa005d3d2";
const FN_INPUTS = [
    {
        type: "address",
        name: "fidOwner",
    },
    {
        type: "uint32",
        name: "keyType",
    },
    {
        type: "bytes",
        name: "key",
    },
    {
        type: "uint8",
        name: "metadataType",
    },
    {
        type: "bytes",
        name: "metadata",
    },
    {
        type: "uint256",
        name: "deadline",
    },
    {
        type: "bytes",
        name: "sig",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "addFor" function.
 * @param options - The options for the addFor function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { encodeAddForParams } "thirdweb/extensions/farcaster";
 * const result = encodeAddForParams({
 *  fidOwner: ...,
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 *  deadline: ...,
 *  sig: ...,
 * });
 * ```
 */
function encodeAddForParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.fidOwner,
        options.keyType,
        options.key,
        options.metadataType,
        options.metadata,
        options.deadline,
        options.sig,
    ]);
}
exports.encodeAddForParams = encodeAddForParams;
/**
 * Calls the "addFor" function on the contract.
 * @param options - The options for the "addFor" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { addFor } from "thirdweb/extensions/farcaster";
 *
 * const transaction = addFor({
 *  contract,
 *  fidOwner: ...,
 *  keyType: ...,
 *  key: ...,
 *  metadataType: ...,
 *  metadata: ...,
 *  deadline: ...,
 *  sig: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function addFor(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.fidOwner,
                    resolvedParams.keyType,
                    resolvedParams.key,
                    resolvedParams.metadataType,
                    resolvedParams.metadata,
                    resolvedParams.deadline,
                    resolvedParams.sig,
                ];
            }
            : [
                options.fidOwner,
                options.keyType,
                options.key,
                options.metadataType,
                options.metadata,
                options.deadline,
                options.sig,
            ],
    });
}
exports.addFor = addFor;
