// thirdweb/dist/cjs/react/web/ui/ConnectWallet/screens/SwapTransactionsScreen.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSwapTransactions = exports.SwapTransactionsScreen = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const basic_js_1 = require("../../components/basic.js");
const Spacer_js_1 = require("../../components/Spacer.js");
const index_js_1 = require("../../design-system/index.js");
const elements_js_1 = require("../../design-system/elements.js");
const CustomThemeProvider_js_1 = require("../../design-system/CustomThemeProvider.js");
const text_js_1 = require("../../components/text.js");
const react_icons_1 = require("@radix-ui/react-icons");
const animations_js_1 = require("../../design-system/animations.js");
const Spinner_js_1 = require("../../components/Spinner.js");
const buttons_js_1 = require("../../components/buttons.js");
const useChainQuery_js_1 = require("../../../../core/hooks/others/useChainQuery.js");
const wallet_hooks_js_1 = require("../../../../core/hooks/wallets/wallet-hooks.js");
const useBuyWithCryptoHistory_js_1 = require("../../../../core/hooks/pay/useBuyWithCryptoHistory.js");
const BuyIcon_js_1 = require("../icons/BuyIcon.js");
const CryptoIcon_js_1 = require("../icons/CryptoIcon.js");
const Skeleton_js_1 = require("../../components/Skeleton.js");
const utils_js_1 = require("../../../../../chains/utils.js");
const pendingSwapTx_js_1 = require("./Buy/swap/pendingSwapTx.js");
const formatNumber_js_1 = require("../../../../../utils/formatNumber.js");
const PAGE_SIZE = 10;
/**
 * @internal
 */
function SwapTransactionsScreen(props) {
    const [pageIndex, setPageIndex] = (0, react_1.useState)(0);
    const _historyQuery = useSwapTransactions(pageIndex, props.client);
    const inMemoryPendingTxs = (0, react_1.useSyncExternalStore)(pendingSwapTx_js_1.swapTransactionsStore.subscribe, pendingSwapTx_js_1.swapTransactionsStore.getValue);
    const txInfosToShow = [];
    const txHashSet = new Set();
    _historyQuery.data?.page.forEach((tx) => {
        txHashSet.add(tx.source.transactionHash);
    });
    // add in-memory pending transactions
    inMemoryPendingTxs.forEach((tx) => {
        if (pageIndex > 0) {
            return;
        }
        // if tx is already in history endpoint, don't add it
        if (txHashSet.has(tx.transactionHash)) {
            return;
        }
        txInfosToShow.push({
            fromChainId: tx.from.chainId,
            transactionHash: tx.transactionHash,
            boughtTokenAmount: tx.to.value,
            boughtTokenSymbol: tx.to.symbol,
            status: "PENDING",
        });
    });
    // Add data from endpoint
    _historyQuery.data?.page.forEach((tx) => {
        txInfosToShow.push({
            fromChainId: tx.source.token.chainId,
            transactionHash: tx.source.transactionHash,
            boughtTokenAmount: tx.destination?.amount || tx.quote.toAmount,
            boughtTokenSymbol: tx.destination?.token.symbol || tx.quote.toToken.symbol || "",
            status: tx.status,
            subStatus: tx.subStatus,
        });
    });
    const activeChain = (0, wallet_hooks_js_1.useActiveWalletChain)();
    const chainQuery = (0, useChainQuery_js_1.useChainQuery)(activeChain);
    const activeAccount = (0, wallet_hooks_js_1.useActiveAccount)();
    const noTransactions = txInfosToShow.length === 0;
    const hidePagination = !_historyQuery.data ||
        (_historyQuery.data && !_historyQuery.data.hasNextPage && pageIndex === 0);
    return ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { animate: "fadein", children: [(0, jsx_runtime_1.jsx)(basic_js_1.Container, { p: "lg", children: (0, jsx_runtime_1.jsx)(basic_js_1.ModalHeader, { title: "Transactions", onBack: props.onBack }) }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { scrollY: true, flex: "column", fullHeight: true, style: {
                    minHeight: "250px",
                    maxHeight: "370px",
                }, children: [(0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", gap: "sm", px: "lg", expand: true, children: [noTransactions && !_historyQuery.isLoading && ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", gap: "md", center: "both", color: "secondaryText", style: {
                                    minHeight: "250px",
                                }, children: [(0, jsx_runtime_1.jsx)(react_icons_1.CrossCircledIcon, { width: index_js_1.iconSize.xl, height: index_js_1.iconSize.xl }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { children: " No Transactions " })] })), noTransactions && _historyQuery.isLoading && ((0, jsx_runtime_1.jsx)(basic_js_1.Container, { flex: "row", center: "both", style: {
                                    minHeight: "250px",
                                }, children: (0, jsx_runtime_1.jsx)(Spinner_js_1.Spinner, { size: "xl", color: "accentText" }) })), txInfosToShow.map((txInfo) => {
                                return ((0, jsx_runtime_1.jsx)(TransactionInfo, { txInfo: txInfo }, txInfo.transactionHash));
                            }), _historyQuery.isLoading && txInfosToShow.length > 0 && ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(Skeleton_js_1.Skeleton, { width: "100%", height: "68px" }), (0, jsx_runtime_1.jsx)(Skeleton_js_1.Skeleton, { width: "100%", height: "68px" }), (0, jsx_runtime_1.jsx)(Skeleton_js_1.Skeleton, { width: "100%", height: "68px" })] }))] }), (0, jsx_runtime_1.jsx)(basic_js_1.Container, { p: "lg", children: _historyQuery.data && !hidePagination && ((0, jsx_runtime_1.jsxs)("div", { style: {
                                display: "grid",
                                gridTemplateColumns: "1fr 1fr",
                                gap: index_js_1.spacing.sm,
                            }, children: [(0, jsx_runtime_1.jsxs)(buttons_js_1.Button, { variant: "outline", gap: "xs", disabled: pageIndex === 0, "data-disabled": pageIndex === 0, style: {
                                        fontSize: index_js_1.fontSize.sm,
                                        paddingBlock: index_js_1.spacing.sm,
                                    }, onClick: () => {
                                        setPageIndex((prev) => prev - 1);
                                    }, children: [(0, jsx_runtime_1.jsx)(react_icons_1.ArrowRightIcon, { width: index_js_1.iconSize.sm, height: index_js_1.iconSize.sm, style: {
                                                transform: "rotate(180deg)",
                                            } }), "Prev"] }), (0, jsx_runtime_1.jsxs)(buttons_js_1.Button, { variant: "outline", gap: "xs", disabled: !_historyQuery.data.hasNextPage, "data-disabled": !_historyQuery.data.hasNextPage, style: {
                                        fontSize: index_js_1.fontSize.sm,
                                        paddingBlock: index_js_1.spacing.sm,
                                    }, onClick: () => {
                                        setPageIndex((prev) => prev + 1);
                                    }, children: ["Next", (0, jsx_runtime_1.jsx)(react_icons_1.ArrowRightIcon, { width: index_js_1.iconSize.sm, height: index_js_1.iconSize.sm })] })] })) })] }), (0, jsx_runtime_1.jsx)(basic_js_1.Line, {}), (0, jsx_runtime_1.jsx)(basic_js_1.Container, { p: "lg", children: (0, jsx_runtime_1.jsx)(ButtonLink, { fullWidth: true, variant: "accent", href: chainQuery.data?.explorers?.[0]?.url +
                        "/address/" +
                        activeAccount?.address, target: "_blank", as: "a", style: {
                        textDecoration: "none",
                        color: "inherit",
                    }, children: "View on Explorer" }) })] }));
}
exports.SwapTransactionsScreen = SwapTransactionsScreen;
/**
 * @internal
 */
function useSwapTransactions(pageIndex, client) {
    const account = (0, wallet_hooks_js_1.useActiveAccount)();
    const historyQuery = (0, useBuyWithCryptoHistory_js_1.useBuyWithCryptoHistory)({
        walletAddress: account?.address || "",
        start: pageIndex * PAGE_SIZE,
        count: PAGE_SIZE,
        client,
    }, {
        // 30 seconds
        refetchInterval: 30 * 1000,
    });
    return historyQuery;
}
exports.useSwapTransactions = useSwapTransactions;
function TransactionInfo(props) {
    const { fromChainId, transactionHash, boughtTokenAmount, boughtTokenSymbol, status, } = props.txInfo;
    const fromChain = (0, react_1.useMemo)(() => (0, utils_js_1.defineChain)(fromChainId), [fromChainId]);
    const chainQuery = (0, useChainQuery_js_1.useChainQuery)(fromChain);
    const statusMeta = getStatusMeta(status, props.txInfo.subStatus);
    return ((0, jsx_runtime_1.jsx)(TxHashLink, { href: `${chainQuery.data?.explorers?.[0]?.url || ""}/tx/${transactionHash}`, target: "_blank", children: (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", center: "y", gap: "md", children: [(0, jsx_runtime_1.jsxs)(IconBox, { "data-box": true, children: [(0, jsx_runtime_1.jsx)(BuyIcon_js_1.BuyIcon, { size: index_js_1.iconSize.sm }), (0, jsx_runtime_1.jsx)("div", { style: {
                                position: "absolute",
                                bottom: 0,
                                right: 0,
                                transform: "translate(30%, 30%)",
                            }, children: (0, jsx_runtime_1.jsx)(CryptoIcon_js_1.CryptoIcon, { size: index_js_1.iconSize.sm }) })] }), (0, jsx_runtime_1.jsxs)("div", { style: {
                        flex: 1,
                    }, children: [(0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", gap: "xs", center: "y", style: {
                                justifyContent: "space-between",
                            }, children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { color: "primaryText", children: " Buy" }), (0, jsx_runtime_1.jsxs)(text_js_1.Text, { size: "sm", color: "primaryText", children: ["+ ", (0, formatNumber_js_1.formatNumber)(Number(boughtTokenAmount), 4), " ", boughtTokenSymbol] }), " "] }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xs" }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", center: "y", gap: "xxs", style: {
                                justifyContent: "space-between",
                            }, children: [(0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "row", gap: "xxs", center: "y", children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", color: statusMeta.color, children: statusMeta.status }), statusMeta.loading && (0, jsx_runtime_1.jsx)(Spinner_js_1.Spinner, { size: "xs", color: "accentText" })] }), chainQuery.data?.name ? ((0, jsx_runtime_1.jsxs)(text_js_1.Text, { size: "sm", children: [" ", chainQuery.data.name] })) : ((0, jsx_runtime_1.jsx)(Skeleton_js_1.Skeleton, { width: "120px", height: index_js_1.fontSize.sm }))] })] })] }) }));
}
const ButtonLink = /* @__PURE__ */ (() => buttons_js_1.Button.withComponent("a"))();
const IconBox = /* @__PURE__ */ (0, elements_js_1.StyledDiv)(() => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        color: theme.colors.secondaryText,
        padding: index_js_1.spacing.sm,
        border: `2px solid ${theme.colors.borderColor}`,
        borderRadius: index_js_1.radius.lg,
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        position: "relative",
    };
});
const TxHashLink = /* @__PURE__ */ (0, elements_js_1.StyledAnchor)(() => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        padding: index_js_1.spacing.sm,
        borderRadius: index_js_1.radius.lg,
        cursor: "pointer",
        animation: `${animations_js_1.fadeInAnimation} 300ms ease`,
        background: theme.colors.walletSelectorButtonHoverBg,
        "&:hover": {
            transition: "background 250ms ease",
            background: theme.colors.secondaryButtonBg,
        },
        height: "68px",
    };
});
function getStatusMeta(status, subStatus) {
    if (subStatus === "WAITING_BRIDGE") {
        return {
            status: "Bridging",
            color: "accentText",
            loading: true,
        };
    }
    if (subStatus === "PARTIAL_SUCCESS") {
        return {
            status: "Incomplete",
            color: "secondaryText",
            loading: false,
        };
    }
    if (status === "PENDING") {
        return {
            status: "Pending",
            color: "accentText",
            loading: true,
        };
    }
    if (status === "FAILED") {
        return {
            status: "Failed",
            color: "danger",
            loading: false,
        };
    }
    if (status === "COMPLETED") {
        return {
            status: "Completed",
            color: "success",
            loading: false,
        };
    }
    return {
        status: "Unknown",
        color: "secondaryText",
    };
}
