// thirdweb/dist/esm/extensions/marketplace/__generated__/IMarketplace/write/createListing.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x296f4e16";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "_params",
        components: [
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "startTime",
            },
            {
                type: "uint256",
                name: "secondsUntilEndTime",
            },
            {
                type: "uint256",
                name: "quantityToList",
            },
            {
                type: "address",
                name: "currencyToAccept",
            },
            {
                type: "uint256",
                name: "reservePricePerToken",
            },
            {
                type: "uint256",
                name: "buyoutPricePerToken",
            },
            {
                type: "uint8",
                name: "listingType",
            },
        ],
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "createListing" function.
 * @param options - The options for the createListing function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { encodeCreateListingParams } "thirdweb/extensions/marketplace";
 * const result = encodeCreateListingParams({
 *  params: ...,
 * });
 * ```
 */
export function encodeCreateListingParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.params]);
}
/**
 * Calls the "createListing" function on the contract.
 * @param options - The options for the "createListing" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { createListing } from "thirdweb/extensions/marketplace";
 *
 * const transaction = createListing({
 *  contract,
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function createListing(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
