// thirdweb/dist/esm/react/web/ui/ConnectWallet/screens/Buy/PaymentSelection.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import styled from "@emotion/styled";
import { Spacer } from "../../../components/Spacer.js";
import { Container } from "../../../components/basic.js";
import { Button } from "../../../components/buttons.js";
import { Text } from "../../../components/text.js";
import { useCustomTheme } from "../../../design-system/CustomThemeProvider.js";
import { StyledDiv } from "../../../design-system/elements.js";
import { fontSize, radius, spacing } from "../../../design-system/index.js";
/**
 * @internal
 */
export function PaymentSelection() {
    return (_jsxs("div", { children: [_jsx(Text, { size: "sm", children: "Pay with " }), _jsx(Spacer, { y: "xs" }), _jsxs(Container, { style: {
                    display: "grid",
                    gridTemplateColumns: "1fr 1fr",
                    gridGap: spacing.sm,
                }, children: [_jsx(CheckButton, { isChecked: true, variant: "outline", children: _jsx(Container, { gap: "xxs", flex: "row", center: "y", children: "Crypto" }) }), _jsxs("div", { style: {
                            position: "relative",
                        }, children: [_jsx(CheckButton, { variant: "outline", isChecked: false, style: {
                                    opacity: 0.5,
                                }, disabled: true, children: "Credit Card" }), _jsx(FloatingBadge, { children: " Coming Soon " })] })] })] }));
}
const CheckButton = /* @__PURE__ */ styled(Button)((props) => {
    const theme = useCustomTheme();
    return {
        fontSize: fontSize.sm,
        borderColor: props.isChecked
            ? theme.colors.accentText
            : theme.colors.borderColor,
        gap: spacing.xs,
        paddingInline: spacing.xxs,
        paddingBlock: spacing.sm,
        width: "100%",
    };
});
const FloatingBadge = /* @__PURE__ */ StyledDiv(() => {
    const theme = useCustomTheme();
    return {
        position: "absolute",
        top: 0,
        right: 0,
        transform: "translate(10%, -60%)",
        backgroundColor: theme.colors.secondaryButtonBg,
        paddingBlock: "3px",
        paddingInline: spacing.xs,
        fontSize: fontSize.xs,
        borderRadius: radius.sm,
        color: theme.colors.accentText,
    };
});
