// thirdweb/dist/esm/extensions/farcaster/__generated__/IKeyRegistry/read/keyGateway.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x80334737";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "address",
    },
];
/**
 * Decodes the result of the keyGateway function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { decodeKeyGatewayResult } from "thirdweb/extensions/farcaster";
 * const result = decodeKeyGatewayResult("...");
 * ```
 */
export function decodeKeyGatewayResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "keyGateway" function on the contract.
 * @param options - The options for the keyGateway function.
 * @returns The parsed result of the function call.
 * @extension FARCASTER
 * @example
 * ```ts
 * import { keyGateway } from "thirdweb/extensions/farcaster";
 *
 * const result = await keyGateway();
 *
 * ```
 */
export async function keyGateway(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
