// thirdweb/dist/esm/react/web/ui/ConnectWallet/Modal/ConnectModalInline.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { Cross2Icon } from "@radix-ui/react-icons";
import { DynamicHeight } from "../../components/DynamicHeight.js";
import { CrossContainer } from "../../components/Modal.js";
import { IconButton } from "../../components/buttons.js";
import { useCustomTheme, CustomThemeProvider, } from "../../design-system/CustomThemeProvider.js";
import { StyledDiv } from "../../design-system/elements.js";
import { iconSize, radius, shadow, } from "../../design-system/index.js";
import { wideModalMaxHeight, modalMaxWidthCompact, modalMaxWidthWide, } from "../constants.js";
import { SyncedWalletUIStates } from "./ConnectEmbed.js";
import { useSetupScreen } from "./screen.js";
import { WalletUIStatesProvider } from "../../../providers/wallet-ui-states-provider.js";
import { ConnectModalContent } from "./ConnectModalContent.js";
import { canFitWideModal } from "../../../utils/canFitWideModal.js";
import { useWalletConnectionCtx } from "../../../../core/hooks/others/useWalletConnectionCtx.js";
/**
 * @internal
 */
export const ConnectModalInline = (props) => {
    const walletConfigs = useWalletConnectionCtx().wallets;
    const modalSize = !canFitWideModal() || walletConfigs.length === 1
        ? "compact"
        : props.modalSize;
    const ctxTheme = useCustomTheme();
    const walletUIStatesProps = {
        theme: props.theme || ctxTheme,
        modalSize: modalSize,
        title: props.modalTitle,
        termsOfServiceUrl: props.termsOfServiceUrl,
        privacyPolicyUrl: props.privacyPolicyUrl,
        welcomeScreen: props.welcomeScreen,
        titleIconUrl: props.modalTitleIconUrl,
        showThirdwebBranding: props.showThirdwebBranding,
    };
    return (_jsx(WalletUIStatesProvider, { ...walletUIStatesProps, children: _jsxs(CustomThemeProvider, { theme: walletUIStatesProps.theme, children: [_jsx(ConnectModalInlineContent, { className: props.className, modalSize: modalSize }), _jsx(SyncedWalletUIStates, { ...walletUIStatesProps })] }) }));
};
function ConnectModalInlineContent(props) {
    const screenSetup = useSetupScreen();
    const content = (_jsxs(_Fragment, { children: [_jsx(ConnectModalContent, { screenSetup: screenSetup, onHide: () => {
                    // no op
                }, isOpen: true, onClose: () => {
                    // no op
                }, onShow: () => {
                    // no op
                } }), _jsx(CrossContainer, { children: _jsx(IconButton, { type: "button", "aria-label": "Close", children: _jsx(Cross2Icon, { width: iconSize.md, height: iconSize.md, style: {
                            color: "inherit",
                        } }) }) })] }));
    return (_jsx(ConnectModalInlineContainer, { className: props.className, style: {
            height: props.modalSize === "compact" ? "auto" : wideModalMaxHeight,
            maxWidth: props.modalSize === "compact"
                ? modalMaxWidthCompact
                : modalMaxWidthWide,
            ...props.style,
        }, children: props.modalSize === "compact" ? (_jsxs(DynamicHeight, { children: [" ", content, " "] })) : (content) }));
}
const ConnectModalInlineContainer = /* @__PURE__ */ StyledDiv(() => {
    const theme = useCustomTheme();
    return {
        background: theme.colors.modalBg,
        color: theme.colors.primaryText,
        transition: "background 0.2s ease",
        borderRadius: radius.xl,
        width: "100%",
        boxSizing: "border-box",
        boxShadow: shadow.lg,
        position: "relative",
        border: `1px solid ${theme.colors.borderColor}`,
        lineHeight: "normal",
        overflow: "hidden",
        fontFamily: theme.fontFamily,
        "& *::selection": {
            backgroundColor: theme.colors.primaryText,
            color: theme.colors.modalBg,
        },
    };
});
