// thirdweb/dist/cjs/extensions/marketplace/__generated__/IMarketplace/write/acceptOffer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.acceptOffer = exports.encodeAcceptOfferParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xb13c0e63";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_listingId",
    },
    {
        type: "address",
        name: "_offeror",
    },
    {
        type: "address",
        name: "_currency",
    },
    {
        type: "uint256",
        name: "_totalPrice",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "acceptOffer" function.
 * @param options - The options for the acceptOffer function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { encodeAcceptOfferParams } "thirdweb/extensions/marketplace";
 * const result = encodeAcceptOfferParams({
 *  listingId: ...,
 *  offeror: ...,
 *  currency: ...,
 *  totalPrice: ...,
 * });
 * ```
 */
function encodeAcceptOfferParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.listingId,
        options.offeror,
        options.currency,
        options.totalPrice,
    ]);
}
exports.encodeAcceptOfferParams = encodeAcceptOfferParams;
/**
 * Calls the "acceptOffer" function on the contract.
 * @param options - The options for the "acceptOffer" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { acceptOffer } from "thirdweb/extensions/marketplace";
 *
 * const transaction = acceptOffer({
 *  listingId: ...,
 *  offeror: ...,
 *  currency: ...,
 *  totalPrice: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function acceptOffer(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.listingId,
                    resolvedParams.offeror,
                    resolvedParams.currency,
                    resolvedParams.totalPrice,
                ];
            }
            : [
                options.listingId,
                options.offeror,
                options.currency,
                options.totalPrice,
            ],
    });
}
exports.acceptOffer = acceptOffer;
