// thirdweb/dist/cjs/extensions/marketplace/__generated__/IMarketplace/write/offer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.offer = exports.encodeOfferParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x5fef45e7";
const FN_INPUTS = [
    {
        type: "uint256",
        name: "_listingId",
    },
    {
        type: "uint256",
        name: "_quantityWanted",
    },
    {
        type: "address",
        name: "_currency",
    },
    {
        type: "uint256",
        name: "_pricePerToken",
    },
    {
        type: "uint256",
        name: "_expirationTimestamp",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "offer" function.
 * @param options - The options for the offer function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { encodeOfferParams } "thirdweb/extensions/marketplace";
 * const result = encodeOfferParams({
 *  listingId: ...,
 *  quantityWanted: ...,
 *  currency: ...,
 *  pricePerToken: ...,
 *  expirationTimestamp: ...,
 * });
 * ```
 */
function encodeOfferParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.listingId,
        options.quantityWanted,
        options.currency,
        options.pricePerToken,
        options.expirationTimestamp,
    ]);
}
exports.encodeOfferParams = encodeOfferParams;
/**
 * Calls the "offer" function on the contract.
 * @param options - The options for the "offer" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { offer } from "thirdweb/extensions/marketplace";
 *
 * const transaction = offer({
 *  listingId: ...,
 *  quantityWanted: ...,
 *  currency: ...,
 *  pricePerToken: ...,
 *  expirationTimestamp: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function offer(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.listingId,
                    resolvedParams.quantityWanted,
                    resolvedParams.currency,
                    resolvedParams.pricePerToken,
                    resolvedParams.expirationTimestamp,
                ];
            }
            : [
                options.listingId,
                options.quantityWanted,
                options.currency,
                options.pricePerToken,
                options.expirationTimestamp,
            ],
    });
}
exports.offer = offer;
