// thirdweb/dist/cjs/extensions/marketplace/__generated__/IMarketplace/write/setContractURI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setContractURI = exports.encodeSetContractURIParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x938e3d7b";
const FN_INPUTS = [
    {
        type: "string",
        name: "_uri",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setContractURI" function.
 * @param options - The options for the setContractURI function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { encodeSetContractURIParams } "thirdweb/extensions/marketplace";
 * const result = encodeSetContractURIParams({
 *  uri: ...,
 * });
 * ```
 */
function encodeSetContractURIParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.uri]);
}
exports.encodeSetContractURIParams = encodeSetContractURIParams;
/**
 * Calls the "setContractURI" function on the contract.
 * @param options - The options for the "setContractURI" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { setContractURI } from "thirdweb/extensions/marketplace";
 *
 * const transaction = setContractURI({
 *  contract,
 *  uri: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setContractURI(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.uri];
            }
            : [options.uri],
    });
}
exports.setContractURI = setContractURI;
