// thirdweb/dist/cjs/extensions/marketplace/__generated__/IMarketplace/write/setPlatformFeeInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setPlatformFeeInfo = exports.encodeSetPlatformFeeInfoParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x1e7ac488";
const FN_INPUTS = [
    {
        type: "address",
        name: "_platformFeeRecipient",
    },
    {
        type: "uint256",
        name: "_platformFeeBps",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "setPlatformFeeInfo" function.
 * @param options - The options for the setPlatformFeeInfo function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { encodeSetPlatformFeeInfoParams } "thirdweb/extensions/marketplace";
 * const result = encodeSetPlatformFeeInfoParams({
 *  platformFeeRecipient: ...,
 *  platformFeeBps: ...,
 * });
 * ```
 */
function encodeSetPlatformFeeInfoParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.platformFeeRecipient,
        options.platformFeeBps,
    ]);
}
exports.encodeSetPlatformFeeInfoParams = encodeSetPlatformFeeInfoParams;
/**
 * Calls the "setPlatformFeeInfo" function on the contract.
 * @param options - The options for the "setPlatformFeeInfo" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { setPlatformFeeInfo } from "thirdweb/extensions/marketplace";
 *
 * const transaction = setPlatformFeeInfo({
 *  contract,
 *  platformFeeRecipient: ...,
 *  platformFeeBps: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function setPlatformFeeInfo(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.platformFeeRecipient,
                    resolvedParams.platformFeeBps,
                ];
            }
            : [options.platformFeeRecipient, options.platformFeeBps],
    });
}
exports.setPlatformFeeInfo = setPlatformFeeInfo;
