// thirdweb/dist/cjs/extensions/marketplace/__generated__/IEnglishAuctions/write/createAuction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAuction = exports.encodeCreateAuctionParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x16654d40";
const FN_INPUTS = [
    {
        type: "tuple",
        name: "_params",
        components: [
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "quantity",
            },
            {
                type: "address",
                name: "currency",
            },
            {
                type: "uint256",
                name: "minimumBidAmount",
            },
            {
                type: "uint256",
                name: "buyoutBidAmount",
            },
            {
                type: "uint64",
                name: "timeBufferInSeconds",
            },
            {
                type: "uint64",
                name: "bidBufferBps",
            },
            {
                type: "uint64",
                name: "startTimestamp",
            },
            {
                type: "uint64",
                name: "endTimestamp",
            },
        ],
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256",
        name: "auctionId",
    },
];
/**
 * Encodes the parameters for the "createAuction" function.
 * @param options - The options for the createAuction function.
 * @returns The encoded ABI parameters.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { encodeCreateAuctionParams } "thirdweb/extensions/marketplace";
 * const result = encodeCreateAuctionParams({
 *  params: ...,
 * });
 * ```
 */
function encodeCreateAuctionParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.params]);
}
exports.encodeCreateAuctionParams = encodeCreateAuctionParams;
/**
 * Calls the "createAuction" function on the contract.
 * @param options - The options for the "createAuction" function.
 * @returns A prepared transaction object.
 * @extension MARKETPLACE
 * @example
 * ```
 * import { createAuction } from "thirdweb/extensions/marketplace";
 *
 * const transaction = createAuction({
 *  params: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function createAuction(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.params];
            }
            : [options.params],
    });
}
exports.createAuction = createAuction;
