// thirdweb/dist/cjs/extensions/erc4337/__generated__/IEntryPoint/write/getSenderAddress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSenderAddress = exports.encodeGetSenderAddressParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x9b249f69";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "initCode",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "getSenderAddress" function.
 * @param options - The options for the getSenderAddress function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```
 * import { encodeGetSenderAddressParams } "thirdweb/extensions/erc4337";
 * const result = encodeGetSenderAddressParams({
 *  initCode: ...,
 * });
 * ```
 */
function encodeGetSenderAddressParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.initCode]);
}
exports.encodeGetSenderAddressParams = encodeGetSenderAddressParams;
/**
 * Calls the "getSenderAddress" function on the contract.
 * @param options - The options for the "getSenderAddress" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```
 * import { getSenderAddress } from "thirdweb/extensions/erc4337";
 *
 * const transaction = getSenderAddress({
 *  initCode: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function getSenderAddress(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.initCode];
            }
            : [options.initCode],
    });
}
exports.getSenderAddress = getSenderAddress;
