// thirdweb/dist/cjs/extensions/erc4337/__generated__/IEntryPoint/write/incrementNonce.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.incrementNonce = exports.encodeIncrementNonceParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0x0bd28e3b";
const FN_INPUTS = [
    {
        type: "uint192",
        name: "key",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "incrementNonce" function.
 * @param options - The options for the incrementNonce function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```
 * import { encodeIncrementNonceParams } "thirdweb/extensions/erc4337";
 * const result = encodeIncrementNonceParams({
 *  key: ...,
 * });
 * ```
 */
function encodeIncrementNonceParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.key]);
}
exports.encodeIncrementNonceParams = encodeIncrementNonceParams;
/**
 * Calls the "incrementNonce" function on the contract.
 * @param options - The options for the "incrementNonce" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```
 * import { incrementNonce } from "thirdweb/extensions/erc4337";
 *
 * const transaction = incrementNonce({
 *  key: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function incrementNonce(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.key];
            }
            : [options.key],
    });
}
exports.incrementNonce = incrementNonce;
