// thirdweb/dist/cjs/extensions/erc4337/__generated__/IEntryPoint/write/withdrawStake.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.withdrawStake = exports.encodeWithdrawStakeParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xc23a5cea";
const FN_INPUTS = [
    {
        type: "address",
        name: "withdrawAddress",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "withdrawStake" function.
 * @param options - The options for the withdrawStake function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```ts
 * import { encodeWithdrawStakeParams } "thirdweb/extensions/erc4337";
 * const result = encodeWithdrawStakeParams({
 *  withdrawAddress: ...,
 * });
 * ```
 */
function encodeWithdrawStakeParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.withdrawAddress]);
}
exports.encodeWithdrawStakeParams = encodeWithdrawStakeParams;
/**
 * Calls the "withdrawStake" function on the contract.
 * @param options - The options for the "withdrawStake" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```ts
 * import { withdrawStake } from "thirdweb/extensions/erc4337";
 *
 * const transaction = withdrawStake({
 *  contract,
 *  withdrawAddress: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function withdrawStake(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.withdrawAddress];
            }
            : [options.withdrawAddress],
    });
}
exports.withdrawStake = withdrawStake;
