// thirdweb/dist/esm/extensions/erc1155/__generated__/IBurnableERC1155/write/burnBatch.js
import { prepareContractCall } from "../../../../../transaction/prepare-contract-call.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
const FN_SELECTOR = "0x6b20c454";
const FN_INPUTS = [
    {
        type: "address",
        name: "account",
    },
    {
        type: "uint256[]",
        name: "ids",
    },
    {
        type: "uint256[]",
        name: "values",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "burnBatch" function.
 * @param options - The options for the burnBatch function.
 * @returns The encoded ABI parameters.
 * @extension ERC1155
 * @example
 * ```ts
 * import { encodeBurnBatchParams } "thirdweb/extensions/erc1155";
 * const result = encodeBurnBatchParams({
 *  account: ...,
 *  ids: ...,
 *  values: ...,
 * });
 * ```
 */
export function encodeBurnBatchParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.account,
        options.ids,
        options.values,
    ]);
}
/**
 * Calls the "burnBatch" function on the contract.
 * @param options - The options for the "burnBatch" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```ts
 * import { burnBatch } from "thirdweb/extensions/erc1155";
 *
 * const transaction = burnBatch({
 *  contract,
 *  account: ...,
 *  ids: ...,
 *  values: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
export function burnBatch(options) {
    return prepareContractCall({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.account,
                    resolvedParams.ids,
                    resolvedParams.values,
                ];
            }
            : [options.account, options.ids, options.values],
    });
}
