// thirdweb/dist/esm/extensions/erc721/__generated__/INFTMetadata/read/supportsInterface.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x01ffc9a7";
const FN_INPUTS = [
    {
        type: "bytes4",
        name: "interfaceId",
    },
];
const FN_OUTPUTS = [
    {
        type: "bool",
    },
];
/**
 * Encodes the parameters for the "supportsInterface" function.
 * @param options - The options for the supportsInterface function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```
 * import { encodeSupportsInterfaceParams } "thirdweb/extensions/erc721";
 * const result = encodeSupportsInterfaceParams({
 *  interfaceId: ...,
 * });
 * ```
 */
export function encodeSupportsInterfaceParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.interfaceId]);
}
/**
 * Decodes the result of the supportsInterface function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeSupportsInterfaceResult } from "thirdweb/extensions/erc721";
 * const result = decodeSupportsInterfaceResult("...");
 * ```
 */
export function decodeSupportsInterfaceResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "supportsInterface" function on the contract.
 * @param options - The options for the supportsInterface function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { supportsInterface } from "thirdweb/extensions/erc721";
 *
 * const result = await supportsInterface({
 *  interfaceId: ...,
 * });
 *
 * ```
 */
export async function supportsInterface(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.interfaceId],
    });
}
