// thirdweb/dist/cjs/extensions/erc4337/__generated__/IPaymaster/write/postOp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postOp = exports.encodePostOpParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa9a23409";
const FN_INPUTS = [
    {
        type: "uint8",
        name: "mode",
    },
    {
        type: "bytes",
        name: "context",
    },
    {
        type: "uint256",
        name: "actualGasCost",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "postOp" function.
 * @param options - The options for the postOp function.
 * @returns The encoded ABI parameters.
 * @extension ERC4337
 * @example
 * ```ts
 * import { encodePostOpParams } "thirdweb/extensions/erc4337";
 * const result = encodePostOpParams({
 *  mode: ...,
 *  context: ...,
 *  actualGasCost: ...,
 * });
 * ```
 */
function encodePostOpParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [
        options.mode,
        options.context,
        options.actualGasCost,
    ]);
}
exports.encodePostOpParams = encodePostOpParams;
/**
 * Calls the "postOp" function on the contract.
 * @param options - The options for the "postOp" function.
 * @returns A prepared transaction object.
 * @extension ERC4337
 * @example
 * ```ts
 * import { postOp } from "thirdweb/extensions/erc4337";
 *
 * const transaction = postOp({
 *  contract,
 *  mode: ...,
 *  context: ...,
 *  actualGasCost: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function postOp(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [
                    resolvedParams.mode,
                    resolvedParams.context,
                    resolvedParams.actualGasCost,
                ];
            }
            : [options.mode, options.context, options.actualGasCost],
    });
}
exports.postOp = postOp;
