// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721A/read/name.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.name = exports.decodeNameResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x06fdde03";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "string",
    },
];
/**
 * Decodes the result of the name function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```
 * import { decodeNameResult } from "thirdweb/extensions/erc721";
 * const result = decodeNameResult("...");
 * ```
 */
function decodeNameResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeNameResult = decodeNameResult;
/**
 * Calls the "name" function on the contract.
 * @param options - The options for the name function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```
 * import { name } from "thirdweb/extensions/erc721";
 *
 * const result = await name();
 *
 * ```
 */
async function name(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.name = name;
