// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721A/read/symbol.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.symbol = exports.decodeSymbolResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x95d89b41";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "string",
    },
];
/**
 * Decodes the result of the symbol function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeSymbolResult } from "thirdweb/extensions/erc721";
 * const result = decodeSymbolResult("...");
 * ```
 */
function decodeSymbolResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeSymbolResult = decodeSymbolResult;
/**
 * Calls the "symbol" function on the contract.
 * @param options - The options for the symbol function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { symbol } from "thirdweb/extensions/erc721";
 *
 * const result = await symbol();
 *
 * ```
 */
async function symbol(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.symbol = symbol;
