// thirdweb/dist/cjs/extensions/erc721/__generated__/IERC721A/read/totalSupply.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.totalSupply = exports.decodeTotalSupplyResult = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0x18160ddd";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the totalSupply function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeTotalSupplyResult } from "thirdweb/extensions/erc721";
 * const result = decodeTotalSupplyResult("...");
 * ```
 */
function decodeTotalSupplyResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTotalSupplyResult = decodeTotalSupplyResult;
/**
 * Calls the "totalSupply" function on the contract.
 * @param options - The options for the totalSupply function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { totalSupply } from "thirdweb/extensions/erc721";
 *
 * const result = await totalSupply();
 *
 * ```
 */
async function totalSupply(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
exports.totalSupply = totalSupply;
