// thirdweb/dist/cjs/exports/extensions/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ownerUpdatedEvent = exports.setOwner = exports.setContractURI = exports.symbol = exports.name = exports.owner = exports.getContractMetadata = exports.contractURI = void 0;
// read
var contractURI_js_1 = require("../../extensions/common/__generated__/IContractMetadata/read/contractURI.js");
Object.defineProperty(exports, "contractURI", { enumerable: true, get: function () { return contractURI_js_1.contractURI; } });
var getContractMetadata_js_1 = require("../../extensions/common/read/getContractMetadata.js");
Object.defineProperty(exports, "getContractMetadata", { enumerable: true, get: function () { return getContractMetadata_js_1.getContractMetadata; } });
var owner_js_1 = require("../../extensions/common/__generated__/IOwnable/read/owner.js");
Object.defineProperty(exports, "owner", { enumerable: true, get: function () { return owner_js_1.owner; } });
var name_js_1 = require("../../extensions/common/read/name.js");
Object.defineProperty(exports, "name", { enumerable: true, get: function () { return name_js_1.name; } });
var symbol_js_1 = require("../../extensions/common/read/symbol.js");
Object.defineProperty(exports, "symbol", { enumerable: true, get: function () { return symbol_js_1.symbol; } });
// write
var setContractURI_js_1 = require("../../extensions/common/__generated__/IContractMetadata/write/setContractURI.js");
Object.defineProperty(exports, "setContractURI", { enumerable: true, get: function () { return setContractURI_js_1.setContractURI; } });
var setOwner_js_1 = require("../../extensions/common/__generated__/IOwnable/write/setOwner.js");
Object.defineProperty(exports, "setOwner", { enumerable: true, get: function () { return setOwner_js_1.setOwner; } });
// events
var OwnerUpdated_js_1 = require("../../extensions/common/__generated__/IOwnable/events/OwnerUpdated.js");
Object.defineProperty(exports, "ownerUpdatedEvent", { enumerable: true, get: function () { return OwnerUpdated_js_1.ownerUpdatedEvent; } });
