// thirdweb/dist/cjs/exports/extensions/erc1155.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.claimTo = exports.getActiveClaimCondition = exports.burnBatch = exports.burn = exports.balanceOf = exports.totalSupply = exports.tokenUri = exports.uri = exports.nextTokenIdToMint = exports.getOwnedNFTs = exports.getNFTs = exports.getNFT = exports.balanceOfBatch = void 0;
// READ
var balanceOfBatch_js_1 = require("../../extensions/erc1155/__generated__/IERC1155/read/balanceOfBatch.js");
Object.defineProperty(exports, "balanceOfBatch", { enumerable: true, get: function () { return balanceOfBatch_js_1.balanceOfBatch; } });
var getNFT_js_1 = require("../../extensions/erc1155/read/getNFT.js");
Object.defineProperty(exports, "getNFT", { enumerable: true, get: function () { return getNFT_js_1.getNFT; } });
var getNFTs_js_1 = require("../../extensions/erc1155/read/getNFTs.js");
Object.defineProperty(exports, "getNFTs", { enumerable: true, get: function () { return getNFTs_js_1.getNFTs; } });
var getOwnedNFTs_js_1 = require("../../extensions/erc1155/read/getOwnedNFTs.js");
Object.defineProperty(exports, "getOwnedNFTs", { enumerable: true, get: function () { return getOwnedNFTs_js_1.getOwnedNFTs; } });
var nextTokenIdToMint_js_1 = require("../../extensions/erc1155/__generated__/IERC1155Enumerable/read/nextTokenIdToMint.js");
Object.defineProperty(exports, "nextTokenIdToMint", { enumerable: true, get: function () { return nextTokenIdToMint_js_1.nextTokenIdToMint; } });
var uri_js_1 = require("../../extensions/erc1155/__generated__/IERC1155/read/uri.js");
Object.defineProperty(exports, "uri", { enumerable: true, get: function () { return uri_js_1.uri; } });
/**
 * @alias for `uri`
 */
Object.defineProperty(exports, "tokenUri", { enumerable: true, get: function () { return uri_js_1.uri; } });
var totalSupply_js_1 = require("../../extensions/erc1155/__generated__/IERC1155/read/totalSupply.js");
Object.defineProperty(exports, "totalSupply", { enumerable: true, get: function () { return totalSupply_js_1.totalSupply; } });
var balanceOf_js_1 = require("../../extensions/erc1155/__generated__/IERC1155/read/balanceOf.js");
Object.defineProperty(exports, "balanceOf", { enumerable: true, get: function () { return balanceOf_js_1.balanceOf; } });
//WRITE
var burn_js_1 = require("../../extensions/erc1155/__generated__/IBurnableERC1155/write/burn.js");
Object.defineProperty(exports, "burn", { enumerable: true, get: function () { return burn_js_1.burn; } });
var burnBatch_js_1 = require("../../extensions/erc1155/__generated__/IBurnableERC1155/write/burnBatch.js");
Object.defineProperty(exports, "burnBatch", { enumerable: true, get: function () { return burnBatch_js_1.burnBatch; } });
/**
 * DROPS extension for ERC1155
 */
var getActiveClaimCondition_js_1 = require("../../extensions/erc1155/drops/read/getActiveClaimCondition.js");
Object.defineProperty(exports, "getActiveClaimCondition", { enumerable: true, get: function () { return getActiveClaimCondition_js_1.getActiveClaimCondition; } });
var claimTo_js_1 = require("../../extensions/erc1155/drops/write/claimTo.js");
Object.defineProperty(exports, "claimTo", { enumerable: true, get: function () { return claimTo_js_1.claimTo; } });
