// thirdweb/dist/cjs/exports/extensions/erc20.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.burnFrom = exports.burn = exports.approve = exports.transferFrom = exports.transfer = exports.mintTo = exports.getCurrencyMetadata = exports.allowance = exports.totalSupply = exports.decimals = exports.balanceOf = exports.getBalance = void 0;
// read
var getBalance_js_1 = require("../../extensions/erc20/read/getBalance.js");
Object.defineProperty(exports, "getBalance", { enumerable: true, get: function () { return getBalance_js_1.getBalance; } });
var balanceOf_js_1 = require("../../extensions/erc20/__generated__/IERC20/read/balanceOf.js");
Object.defineProperty(exports, "balanceOf", { enumerable: true, get: function () { return balanceOf_js_1.balanceOf; } });
var decimals_js_1 = require("../../extensions/erc20/read/decimals.js");
Object.defineProperty(exports, "decimals", { enumerable: true, get: function () { return decimals_js_1.decimals; } });
var totalSupply_js_1 = require("../../extensions/erc20/__generated__/IERC20/read/totalSupply.js");
Object.defineProperty(exports, "totalSupply", { enumerable: true, get: function () { return totalSupply_js_1.totalSupply; } });
var allowance_js_1 = require("../../extensions/erc20/__generated__/IERC20/read/allowance.js");
Object.defineProperty(exports, "allowance", { enumerable: true, get: function () { return allowance_js_1.allowance; } });
var getCurrencyMetadata_js_1 = require("../../extensions/erc20/read/getCurrencyMetadata.js");
Object.defineProperty(exports, "getCurrencyMetadata", { enumerable: true, get: function () { return getCurrencyMetadata_js_1.getCurrencyMetadata; } });
// write
var mintTo_js_1 = require("../../extensions/erc20/write/mintTo.js");
Object.defineProperty(exports, "mintTo", { enumerable: true, get: function () { return mintTo_js_1.mintTo; } });
var transfer_js_1 = require("../../extensions/erc20/write/transfer.js");
Object.defineProperty(exports, "transfer", { enumerable: true, get: function () { return transfer_js_1.transfer; } });
var transferFrom_js_1 = require("../../extensions/erc20/write/transferFrom.js");
Object.defineProperty(exports, "transferFrom", { enumerable: true, get: function () { return transferFrom_js_1.transferFrom; } });
var approve_js_1 = require("../../extensions/erc20/write/approve.js");
Object.defineProperty(exports, "approve", { enumerable: true, get: function () { return approve_js_1.approve; } });
var burn_js_1 = require("../../extensions/erc20/__generated__/IBurnableERC20/write/burn.js");
Object.defineProperty(exports, "burn", { enumerable: true, get: function () { return burn_js_1.burn; } });
var burnFrom_js_1 = require("../../extensions/erc20/__generated__/IBurnableERC20/write/burnFrom.js");
Object.defineProperty(exports, "burnFrom", { enumerable: true, get: function () { return burnFrom_js_1.burnFrom; } });
