// thirdweb/dist/cjs/extensions/erc1155/__generated__/IPackVRFDirect/write/claimRewards.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.claimRewards = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const FN_SELECTOR = "0x372500ab";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "tuple[]",
        name: "rewardUnits",
        components: [
            {
                type: "address",
                name: "assetContract",
            },
            {
                type: "uint8",
                name: "tokenType",
            },
            {
                type: "uint256",
                name: "tokenId",
            },
            {
                type: "uint256",
                name: "totalAmount",
            },
        ],
    },
];
/**
 * Calls the "claimRewards" function on the contract.
 * @param options - The options for the "claimRewards" function.
 * @returns A prepared transaction object.
 * @extension ERC1155
 * @example
 * ```
 * import { claimRewards } from "thirdweb/extensions/erc1155";
 *
 * const transaction = claimRewards();
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function claimRewards(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
    });
}
exports.claimRewards = claimRewards;
