// thirdweb/dist/cjs/extensions/farcaster/__generated__/IStorageRegistry/write/batchRent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.batchRent = exports.encodeBatchRentParams = void 0;
const prepare_contract_call_js_1 = require("../../../../../transaction/prepare-contract-call.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const FN_SELECTOR = "0xa82c356e";
const FN_INPUTS = [
    {
        type: "uint256[]",
        name: "fids",
    },
    {
        type: "uint256[]",
        name: "units",
    },
];
const FN_OUTPUTS = [];
/**
 * Encodes the parameters for the "batchRent" function.
 * @param options - The options for the batchRent function.
 * @returns The encoded ABI parameters.
 * @extension FARCASTER
 * @example
 * ```
 * import { encodeBatchRentParams } "thirdweb/extensions/farcaster";
 * const result = encodeBatchRentParams({
 *  fids: ...,
 *  units: ...,
 * });
 * ```
 */
function encodeBatchRentParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.fids, options.units]);
}
exports.encodeBatchRentParams = encodeBatchRentParams;
/**
 * Calls the "batchRent" function on the contract.
 * @param options - The options for the "batchRent" function.
 * @returns A prepared transaction object.
 * @extension FARCASTER
 * @example
 * ```
 * import { batchRent } from "thirdweb/extensions/farcaster";
 *
 * const transaction = batchRent({
 *  fids: ...,
 *  units: ...,
 * });
 *
 * // Send the transaction
 * ...
 *
 * ```
 */
function batchRent(options) {
    return (0, prepare_contract_call_js_1.prepareContractCall)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: "asyncParams" in options
            ? async () => {
                const resolvedParams = await options.asyncParams();
                return [resolvedParams.fids, resolvedParams.units];
            }
            : [options.fids, options.units],
    });
}
exports.batchRent = batchRent;
