// thirdweb/dist/cjs/extensions/erc721/__generated__/IStaking721/read/getStakeInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStakeInfo = exports.decodeGetStakeInfoResult = exports.encodeGetStakeInfoParams = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
const FN_SELECTOR = "0xc3453153";
const FN_INPUTS = [
    {
        type: "address",
        name: "staker",
    },
];
const FN_OUTPUTS = [
    {
        type: "uint256[]",
        name: "_tokensStaked",
    },
    {
        type: "uint256",
        name: "_rewards",
    },
];
/**
 * Encodes the parameters for the "getStakeInfo" function.
 * @param options - The options for the getStakeInfo function.
 * @returns The encoded ABI parameters.
 * @extension ERC721
 * @example
 * ```ts
 * import { encodeGetStakeInfoParams } "thirdweb/extensions/erc721";
 * const result = encodeGetStakeInfoParams({
 *  staker: ...,
 * });
 * ```
 */
function encodeGetStakeInfoParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.staker]);
}
exports.encodeGetStakeInfoParams = encodeGetStakeInfoParams;
/**
 * Decodes the result of the getStakeInfo function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ERC721
 * @example
 * ```ts
 * import { decodeGetStakeInfoResult } from "thirdweb/extensions/erc721";
 * const result = decodeGetStakeInfoResult("...");
 * ```
 */
function decodeGetStakeInfoResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result);
}
exports.decodeGetStakeInfoResult = decodeGetStakeInfoResult;
/**
 * Calls the "getStakeInfo" function on the contract.
 * @param options - The options for the getStakeInfo function.
 * @returns The parsed result of the function call.
 * @extension ERC721
 * @example
 * ```ts
 * import { getStakeInfo } from "thirdweb/extensions/erc721";
 *
 * const result = await getStakeInfo({
 *  staker: ...,
 * });
 *
 * ```
 */
async function getStakeInfo(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.staker],
    });
}
exports.getStakeInfo = getStakeInfo;
