// thirdweb/dist/cjs/react/web/ui/ConnectWallet/screens/Buy/PaymentSelection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentSelection = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const styled_1 = require("@emotion/styled");
const Spacer_js_1 = require("../../../components/Spacer.js");
const basic_js_1 = require("../../../components/basic.js");
const buttons_js_1 = require("../../../components/buttons.js");
const text_js_1 = require("../../../components/text.js");
const CustomThemeProvider_js_1 = require("../../../design-system/CustomThemeProvider.js");
const elements_js_1 = require("../../../design-system/elements.js");
const index_js_1 = require("../../../design-system/index.js");
/**
 * @internal
 */
function PaymentSelection() {
    return ((0, jsx_runtime_1.jsxs)("div", { children: [(0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", children: "Pay with " }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xs" }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { style: {
                    display: "grid",
                    gridTemplateColumns: "1fr 1fr",
                    gridGap: index_js_1.spacing.sm,
                }, children: [(0, jsx_runtime_1.jsx)(CheckButton, { isChecked: true, variant: "outline", children: (0, jsx_runtime_1.jsx)(basic_js_1.Container, { gap: "xxs", flex: "row", center: "y", children: "Crypto" }) }), (0, jsx_runtime_1.jsxs)("div", { style: {
                            position: "relative",
                        }, children: [(0, jsx_runtime_1.jsx)(CheckButton, { variant: "outline", isChecked: false, style: {
                                    opacity: 0.5,
                                }, disabled: true, children: "Credit Card" }), (0, jsx_runtime_1.jsx)(FloatingBadge, { children: " Coming Soon " })] })] })] }));
}
exports.PaymentSelection = PaymentSelection;
const CheckButton = /* @__PURE__ */ (0, styled_1.default)(buttons_js_1.Button)((props) => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        fontSize: index_js_1.fontSize.sm,
        borderColor: props.isChecked
            ? theme.colors.accentText
            : theme.colors.borderColor,
        gap: index_js_1.spacing.xs,
        paddingInline: index_js_1.spacing.xxs,
        paddingBlock: index_js_1.spacing.sm,
        width: "100%",
    };
});
const FloatingBadge = /* @__PURE__ */ (0, elements_js_1.StyledDiv)(() => {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    return {
        position: "absolute",
        top: 0,
        right: 0,
        transform: "translate(10%, -60%)",
        backgroundColor: theme.colors.secondaryButtonBg,
        paddingBlock: "3px",
        paddingInline: index_js_1.spacing.xs,
        fontSize: index_js_1.fontSize.xs,
        borderRadius: index_js_1.radius.sm,
        color: theme.colors.accentText,
    };
});
