// thirdweb/dist/cjs/react/web/wallets/shared/WalletConnectConnection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletConnectConnection = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const ScanScreen_js_1 = require("./ScanScreen.js");
const react_1 = require("react");
const isMobile_js_1 = require("../../../core/utils/isMobile.js");
const handleWCSessionRequest_js_1 = require("../../../core/utils/handleWCSessionRequest.js");
const ConnectingScreen_js_1 = require("./ConnectingScreen.js");
const openWindow_js_1 = require("../../../core/utils/openWindow.js");
const useWalletConnectionCtx_js_1 = require("../../../core/hooks/others/useWalletConnectionCtx.js");
/**
 * QR Scan UI for connecting a specific wallet on desktop.
 * shows a "Connecting" screen and opens the app on mobile.
 * @internal
 */
const WalletConnectConnection = (props) => {
    const { onBack, onGetStarted, wallet, walletInfo, locale, done } = props;
    const [qrCodeUri, setQrCodeUri] = (0, react_1.useState)();
    const [errorConnecting, setErrorConnecting] = (0, react_1.useState)(false);
    const { chain, chains, client, walletConnect } = (0, useWalletConnectionCtx_js_1.useWalletConnectionCtx)();
    const connect = (0, react_1.useCallback)(() => {
        setErrorConnecting(false);
        const platformUris = {
            ios: walletInfo.mobile.native || "",
            android: walletInfo.mobile.universal || "",
            other: walletInfo.mobile.universal || "",
        };
        const onSessionRequestSent = (0, isMobile_js_1.isMobile)()
            ? () => (0, handleWCSessionRequest_js_1.handelWCSessionRequest)(platformUris)
            : undefined;
        wallet
            .connect({
            chain,
            client: client,
            walletConnect: {
                projectId: walletConnect?.projectId,
                showQrModal: false,
                onDisplayUri(uri) {
                    setQrCodeUri(uri);
                    if ((0, isMobile_js_1.isMobile)()) {
                        if ((0, isMobile_js_1.isAndroid)()) {
                            (0, openWindow_js_1.openWindow)(`${platformUris.android}wc?uri=${encodeURIComponent(uri)}`);
                        }
                        else if ((0, isMobile_js_1.isIOS)()) {
                            (0, openWindow_js_1.openWindow)(`${platformUris.ios}wc?uri=${encodeURIComponent(uri)}`);
                        }
                        else {
                            (0, openWindow_js_1.openWindow)(`${platformUris.other}wc?uri=${encodeURIComponent(uri)}`);
                        }
                    }
                },
                onSessionRequestSent,
                optionalChains: chains,
            },
        })
            .then(() => {
            done();
        })
            .catch((e) => {
            setErrorConnecting(true);
            console.error(e);
        });
    }, [
        walletConnect,
        walletInfo.mobile.native,
        walletInfo.mobile.universal,
        wallet,
        chain,
        client,
        chains,
        done,
    ]);
    const scanStarted = (0, react_1.useRef)(false);
    (0, react_1.useEffect)(() => {
        if (scanStarted.current) {
            return;
        }
        scanStarted.current = true;
        connect();
    }, [connect]);
    if ((0, isMobile_js_1.isMobile)()) {
        return ((0, jsx_runtime_1.jsx)(ConnectingScreen_js_1.ConnectingScreen, { locale: {
                getStartedLink: locale.getStartedLink,
                instruction: locale.connectionScreen.instruction,
                tryAgain: locale.connectionScreen.retry,
                inProgress: locale.connectionScreen.inProgress,
                failed: locale.connectionScreen.failed,
            }, onBack: onBack, walletName: walletInfo.name, walletId: wallet.id, errorConnecting: errorConnecting, onRetry: connect, onGetStarted: onGetStarted }));
    }
    return ((0, jsx_runtime_1.jsx)(ScanScreen_js_1.ScanScreen, { qrScanInstruction: locale.scanScreen.instruction, onBack: onBack, onGetStarted: onGetStarted, qrCodeUri: qrCodeUri, walletName: walletInfo.name, walletId: wallet.id, getStartedLink: locale.getStartedLink }));
};
exports.WalletConnectConnection = WalletConnectConnection;
