// thirdweb/dist/esm/extensions/uniswap/__generated__/IUniswapV3Factory/read/getPool.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
const FN_SELECTOR = "0x1698ee82";
const FN_INPUTS = [
    {
        type: "address",
        name: "tokenA",
    },
    {
        type: "address",
        name: "tokenB",
    },
    {
        type: "uint24",
        name: "fee",
    },
];
const FN_OUTPUTS = [
    {
        type: "address",
        name: "pool",
    },
];
/**
 * Encodes the parameters for the "getPool" function.
 * @param options - The options for the getPool function.
 * @returns The encoded ABI parameters.
 * @extension UNISWAP
 * @example
 * ```
 * import { encodeGetPoolParams } "thirdweb/extensions/uniswap";
 * const result = encodeGetPoolParams({
 *  tokenA: ...,
 *  tokenB: ...,
 *  fee: ...,
 * });
 * ```
 */
export function encodeGetPoolParams(options) {
    return encodeAbiParameters(FN_INPUTS, [
        options.tokenA,
        options.tokenB,
        options.fee,
    ]);
}
/**
 * Decodes the result of the getPool function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension UNISWAP
 * @example
 * ```
 * import { decodeGetPoolResult } from "thirdweb/extensions/uniswap";
 * const result = decodeGetPoolResult("...");
 * ```
 */
export function decodeGetPoolResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "getPool" function on the contract.
 * @param options - The options for the getPool function.
 * @returns The parsed result of the function call.
 * @extension UNISWAP
 * @example
 * ```
 * import { getPool } from "thirdweb/extensions/uniswap";
 *
 * const result = await getPool({
 *  tokenA: ...,
 *  tokenB: ...,
 *  fee: ...,
 * });
 *
 * ```
 */
export async function getPool(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.tokenA, options.tokenB, options.fee],
    });
}
